<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-Tutorials/genome-tracks-tutorial">
<head>
<meta charset="UTF-8">
<meta name="generator" content="Docusaurus v2.3.1">
<title data-rh="true">Making Genome Tracks Tutorial | ScriptManager</title><meta data-rh="true" name="viewport" content="width=device-width,initial-scale=1"><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://CEGRcode.github.io/scriptmanager-docs/docs/Tutorials/genome-tracks-tutorial"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Making Genome Tracks Tutorial | ScriptManager"><meta data-rh="true" name="description" content="How to turn your BAM files into genome track files that can be viewed in a genome browser"><meta data-rh="true" property="og:description" content="How to turn your BAM files into genome track files that can be viewed in a genome browser"><link data-rh="true" rel="icon" href="/scriptmanager-docs/img/SM_favicon.png"><link data-rh="true" rel="canonical" href="https://CEGRcode.github.io/scriptmanager-docs/docs/Tutorials/genome-tracks-tutorial"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/scriptmanager-docs/docs/Tutorials/genome-tracks-tutorial" hreflang="en"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/scriptmanager-docs/docs/Tutorials/genome-tracks-tutorial" hreflang="x-default"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.3.0/css/all.min.css" integrity="sha512-SzlrxWUlpfuzQ+pcUCosxcglQRNAq/DZjVsC0lE40xsADsfeQoEypE+enwcOiGjk/bSuGGKHEyjSoQ1zVisanQ==" crossorigin="anonymous"><link rel="stylesheet" href="/scriptmanager-docs/assets/css/styles.93e36019.css">
<link rel="preload" href="/scriptmanager-docs/assets/js/runtime~main.3250e9c7.js" as="script">
<link rel="preload" href="/scriptmanager-docs/assets/js/main.510075bb.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region" aria-label="Skip to main content"><a class="skipToContent_fXgn" href="#docusaurus_skipToContent_fallback">Skip to main content</a></div><nav aria-label="Main" class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Toggle navigation bar" aria-expanded="false" class="navbar__toggle clean-btn" type="button"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/scriptmanager-docs/"><div class="navbar__logo"><img src="/scriptmanager-docs/img/SM_favicon.png" alt="ScriptManager Logo" class="themedImage_ToTc themedImage--light_HNdA"><img src="/scriptmanager-docs/img/SM_favicon.png" alt="ScriptManager Logo" class="themedImage_ToTc themedImage--dark_i4oU"></div><b class="navbar__title text--truncate">ScriptManager</b></a><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/scriptmanager-docs/docs/">General</a><a class="navbar__item navbar__link" href="/scriptmanager-docs/docs/read-analysis/tag-pileup">Tools</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/CEGRcode/scriptmanager" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link"></a><div class="toggle_vylO colorModeToggle_DEke"><button class="clean-btn toggleButton_gllP toggleButtonDisabled_aARS" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)" aria-live="polite"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_pyhR"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_wfgR"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_ZlJk"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div id="docusaurus_skipToContent_fallback" class="main-wrapper mainWrapper_z2l0 docsWrapper_BCFX"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_sjWU" type="button"></button><div class="docPage__5DB"><aside class="theme-doc-sidebar-container docSidebarContainer_b6E3"><div class="sidebarViewport_Xe31"><div class="sidebar_njMd"><nav aria-label="Docs sidebar" class="menu thin-scrollbar menu_SIkG"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--sublist-caret" aria-expanded="false" href="/scriptmanager-docs/docs/">Guides</a></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--sublist-caret menu__link--active" aria-expanded="true" href="/scriptmanager-docs/docs/Tutorials/chipexo-tutorial">Tutorials</a></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/scriptmanager-docs/docs/Tutorials/chipexo-tutorial">ChIP-exo Tutorial</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/scriptmanager-docs/docs/Tutorials/genomic-features-tutorial">Genomic Features Tutorial</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/scriptmanager-docs/docs/Tutorials/genome-tracks-tutorial">Genome Tracks Tutorial</a></li></ul></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--sublist-caret" aria-expanded="false" href="/scriptmanager-docs/docs/References/file-formats">References</a></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--sublist-caret" aria-expanded="false" href="/scriptmanager-docs/docs/Contributing/developer-guidelines">Contributing</a></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--sublist-caret" aria-expanded="false" href="/scriptmanager-docs/docs/Community/code-of-conduct">Community</a></div></li></ul></nav></div></div></aside><main class="docMainContainer_gTbr"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_VOVn"><div class="docItemContainer_Djhp"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Z_bl" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a aria-label="Home page" class="breadcrumbs__link" href="/scriptmanager-docs/"><svg viewBox="0 0 24 24" class="breadcrumbHomeIcon_YNFT"><path d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z" fill="currentColor"></path></svg></a></li><li class="breadcrumbs__item"><span class="breadcrumbs__link">Tutorials</span><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">Genome Tracks Tutorial</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_ETCw theme-doc-toc-mobile tocMobile_ITEo"><button type="button" class="clean-btn tocCollapsibleButton_TO0P">On this page</button></div><div class="theme-doc-markdown markdown"><header><h1>Making Genome Tracks Tutorial</h1></header><p><em>How to turn your BAM files into genome track files that can be viewed in a genome browser</em></p><p><strong>Goal:</strong> This tutorial provides a guide to transforming and filtering your ChIP-exo alignment data into files that can be uploaded by a genome browser.</p><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/IGV_BrowserShot-1ff8f8b15497181fcb1800cc6fc746f5.png" style="width:100%" class="img_ev3q"></div><div class="theme-admonition theme-admonition-info alert alert--info admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 14 16"><path fill-rule="evenodd" d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg></span>info</div><div class="admonitionContent_S0QG"><p>❗ <strong> BigWig Warning </strong>❗
Be careful about performing analysis steps using BigWig files instead of BAM files. BigWigs are excellent for visualizing data pileups in a genome browser but if you are processing data, we recommend you process data directly from BAM formats. From a information content and reproducibility standpoint, BAM files have notable advantages over BigWigs. <a href="/scriptmanager-docs/docs/Contributing/developer-guidelines#bam-vs-bigwig-and-other-genome-track-formats">Read more about some of these advantages</a></p></div></div><p>This tutorial is based on the processing script from <a href="https://github.com/CEGRcode/2022-Mittal_SAGA/blob/main/03_Bulk_Processing/job/bulk_geo_tracks.pbs" target="_blank" rel="noopener noreferrer">Mittal et al, 2022</a> to turn BAM files into BigWig tracks <strong>with base-pair resolution for high resolution assays</strong>.</p><p>There will be a couple steps that need to be performed on the command line using UCSC binaries because we currently don&#x27;t have a parser for BigWig files (in the works but not ready yet). In the meantime, here are the step-by-step instructions for using as much of ScriptManager&#x27;s GUI as possible to generate the BigWig track.</p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="download-scriptmanager-v014">Download ScriptManager (v0.14):<a href="#download-scriptmanager-v014" class="hash-link" aria-label="Direct link to Download ScriptManager (v0.14):" title="Direct link to Download ScriptManager (v0.14):">​</a></h2><p>The current version of ScriptManager is available for download <strong><a href="https://github.com/CEGRcode/scriptmanager/releases/download/v0.14/ScriptManager-v0.14.jar" target="_blank" rel="noopener noreferrer">here</a></strong>. Make sure you have <a href="/scriptmanager-docs/docs/#java">Java installed</a>.</p><p>The file <code>ScriptManager-v0.14.jar</code> should be placed someplace locally accessible. For example on Mac OS on the Desktop (Permissions will need to be accepted) or someplace in your home directory (i.e. Macintosh HD/Users/userID/ScriptManager)</p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="download-uscs-binary-bedgraphtobigwig">Download USCS binary <code>bedGraphToBigWig</code><a href="#download-uscs-binary-bedgraphtobigwig" class="hash-link" aria-label="Direct link to download-uscs-binary-bedgraphtobigwig" title="Direct link to download-uscs-binary-bedgraphtobigwig">​</a></h2><p>You will need to download the precompiled binary for converting bedGraphs to BigWig files. Navigate to <a href="https://hgdownload.soe.ucsc.edu/admin/exe/" target="_blank" rel="noopener noreferrer">UCSC executables FTP</a>. Click the link to the OS that matches your machine and scroll down to download the <code>bedGraphToBigWig</code> binary.</p><div class="tabs-container tabList__CuJ"><ul role="tablist" aria-orientation="horizontal" class="tabs"><li role="tab" tabindex="0" aria-selected="true" class="tabs__item tabItem_LNqP tabs__item--active">Linux/MacOS</li></ul><div class="margin-top--md"><div role="tabpanel" class="tabItem_Ymn6"><p>Make sure you modify the permissions so that you can execute the binary. You can do this with <code>chmod</code></p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">chmod 755 /path/to/bedGraphToBigWig</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>For example, if your <code>bedGraphToBigWig</code> binary downloaded to your <code>Downloads</code> folder,</p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">chmod 755 ~/Downloads/bedGraphToBigWig</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div></div></div></div><h2 class="anchor anchorWithStickyNavbar_LWe7" id="download-data">Download data<a href="#download-data" class="hash-link" aria-label="Direct link to Download data" title="Direct link to Download data">​</a></h2><p>You need one file of sequencing data alignments (BAM) to complete this exercise and a file with the sacCer3 yeast reference genome chromosome sizes (<code>chrom.sizes</code>). <a href="/scriptmanager-docs/docs/References/file-formats">Read more about the BAM file format here.</a></p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="bam-file">BAM File<a href="#bam-file" class="hash-link" aria-label="Direct link to BAM File" title="Direct link to BAM File">​</a></h3><p>This is the set of Reb1 read alignments from the Yeast Epigenome Project (YEP). <a href="https://pubmed.ncbi.nlm.nih.gov/33692541/" target="_blank" rel="noopener noreferrer">See Rossi et al (2021)</a> for more details.</p><a href="ftp://data1.commons.psu.edu/pub/commons/eberly/pughlab/yeast-epigenome-project/12141_YEP.zip" target="_blank" rel="noopener noreferrer" class="button button--secondary">Download sample BAM file</a><p>OR</p><ol><li>Navigate to <a href="http://www.yeastepigenome.org" target="_blank" rel="noopener noreferrer">www.yeastepigenome.org</a> and search for Reb1</li><li>Select &quot;META DATA&quot;</li><li>Select &quot;Direct Download&quot;</li><li>Unzip the resulting file ‘12141_YEP.zip’ and inspect the contents of the new <code>12141_YEP</code> folder. It should contain a file called <code>12141_filtered.bam</code>.</li></ol><h3 class="anchor anchorWithStickyNavbar_LWe7" id="xxxxchromsizes-reference-file">XXXX.chrom.sizes Reference File<a href="#xxxxchromsizes-reference-file" class="hash-link" aria-label="Direct link to XXXX.chrom.sizes Reference File" title="Direct link to XXXX.chrom.sizes Reference File">​</a></h3><p>You will also need the chromosome sizes file of the yeast reference genome (sacCer3):</p><a href="https://hgdownload.soe.ucsc.edu/goldenPath/sacCer3/bigZips/sacCer3.chrom.sizes" target="_blank" rel="noopener noreferrer" class="button button--secondary">Download sacCer3.chrom.sizes (TXT)</a><br><br><div class="theme-admonition theme-admonition-warning alert alert--danger admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 12 16"><path fill-rule="evenodd" d="M5.05.31c.81 2.17.41 3.38-.52 4.31C3.55 5.67 1.98 6.45.9 7.98c-1.45 2.05-1.7 6.53 3.53 7.7-2.2-1.16-2.67-4.52-.3-6.61-.61 2.03.53 3.33 1.94 2.86 1.39-.47 2.3.53 2.27 1.67-.02.78-.31 1.44-1.13 1.81 3.42-.59 4.78-3.42 4.78-5.56 0-2.84-2.53-3.22-1.25-5.61-1.52.13-2.03 1.13-1.89 2.75.09 1.08-1.02 1.8-1.86 1.33-.67-.41-.66-1.19-.06-1.78C8.18 5.31 8.68 2.45 5.05.32L5.03.3l.02.01z"></path></svg></span>danger</div><div class="admonitionContent_S0QG"><p>The downloaded sacCer3.chrom.sizes file may be using the roman numeral naming system (<code>chrI chrII chrIII ...</code>). You will need to modify this to the arabic (<code>chr1 chr2 chr3 ...</code>) numeral naming system if you plan to use the BAM file linked above. You can do this in a simple text editor.</p><p>Also, if the file is missing the <code>2-micron</code> row, you will need to add it manually. Simply append the following tab-delimited row to the bottom of the <code>sacCer3.chrom.sizes</code> file.</p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">2-micron    6318</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div></div></div><h2 class="anchor anchorWithStickyNavbar_LWe7" id="generate-the-genome-tracks">Generate the Genome Tracks<a href="#generate-the-genome-tracks" class="hash-link" aria-label="Direct link to Generate the Genome Tracks" title="Direct link to Generate the Genome Tracks">​</a></h2><h3 class="anchor anchorWithStickyNavbar_LWe7" id="1-open-scriptmanager">1. Open ScriptManager<a href="#1-open-scriptmanager" class="hash-link" aria-label="Direct link to 1. Open ScriptManager" title="Direct link to 1. Open ScriptManager">​</a></h3><div class="tabs-container tabList__CuJ"><ul role="tablist" aria-orientation="horizontal" class="tabs"><li role="tab" tabindex="0" aria-selected="true" class="tabs__item tabItem_LNqP tabs__item--active">MacOS</li><li role="tab" tabindex="-1" aria-selected="false" class="tabs__item tabItem_LNqP">Linux</li><li role="tab" tabindex="-1" aria-selected="false" class="tabs__item tabItem_LNqP">Windows</li></ul><div class="margin-top--md"><div role="tabpanel" class="tabItem_Ymn6"><p>Depending on your system permissions, you may need to be an administrator to open this for the first time. On Mac systems, this can be done by right-clicking the file and selecting ‘Open’ at the top.</p><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/MacOpenSM-5f0508d59e500792ea199946818706f1.png" style="width:40%;border:solid 1px" class="img_ev3q"></div><div class="theme-admonition theme-admonition-caution alert alert--warning admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"></path></svg></span>caution</div><div class="admonitionContent_S0QG"><p>Some MacOS systems may not properly open the JAR by simply double-clicking on the JAR file so you may need to open your Terminal window and execute it from the command line by executing the jar file <em>without any arguments or flags</em>:</p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">java -jar /path/to/ScriptManager.jar</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>If you&#x27;re not sure how to type the path to ScriptManager, you can type <code>java -jar </code> (end with space) and then drag ScriptManager from Finder into your Terminal window and then press enter.</p></div></div></div><div role="tabpanel" class="tabItem_Ymn6" hidden=""><p>Double-click or right-click the ScriptManager JAR file to start the program.</p><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/LinuxOpenSM-1fe13f9ab9c369dd24c0a67c7ee86f41.png" style="width:40%;border:solid 1px" class="img_ev3q"></div></div><div role="tabpanel" class="tabItem_Ymn6" hidden=""><p>Double-click or right-click the ScriptManager JAR file to start the program.</p><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/WindowsOpenSM-ac32d3855e9f0cf632ab34c5c9fc983d.png" style="width:40%;border:solid 1px" class="img_ev3q"></div></div></div></div><p>Once you see the main tool selection window, you&#x27;re off to the races!</p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="2-create-a-whole-genome-pileup-using-bam-format-converter-tools">2. Create a whole-genome pileup using BAM Format Converter Tools<a href="#2-create-a-whole-genome-pileup-using-bam-format-converter-tools" class="hash-link" aria-label="Direct link to 2. Create a whole-genome pileup using BAM Format Converter Tools" title="Direct link to 2. Create a whole-genome pileup using BAM Format Converter Tools">​</a></h3><h4 class="anchor anchorWithStickyNavbar_LWe7" id="21-navigate-to-bam-format-converter-️-bam-to-scidx-to-do-a-genome-wide-pileup-of-your-bam">2.1. Navigate to BAM Format Converter ➡️ <a href="/scriptmanager-docs/docs/bam-format-converter/bam-to-scidx"><strong>BAM to scIDX</strong></a> to do a genome-wide pileup of your BAM<a href="#21-navigate-to-bam-format-converter-️-bam-to-scidx-to-do-a-genome-wide-pileup-of-your-bam" class="hash-link" aria-label="Direct link to 21-navigate-to-bam-format-converter-️-bam-to-scidx-to-do-a-genome-wide-pileup-of-your-bam" title="Direct link to 21-navigate-to-bam-format-converter-️-bam-to-scidx-to-do-a-genome-wide-pileup-of-your-bam">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/maingui-bamtoscidx-673594bf8e06d0b7074206cda67ec2fb.png" style="width:60%" class="img_ev3q"></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="22-select-encoding-information">2.2. Select Encoding information<a href="#22-select-encoding-information" class="hash-link" aria-label="Direct link to 2.2. Select Encoding information" title="Direct link to 2.2. Select Encoding information">​</a></h4><p>Since this is a ChIP-exo dataset, we will select Read 1 (5&#x27; end) for the encoding. You may select insert size filters if you wish.</p><h4 class="anchor anchorWithStickyNavbar_LWe7" id="23-execute-pileup">2.3. Execute pileup<a href="#23-execute-pileup" class="hash-link" aria-label="Direct link to 2.3. Execute pileup" title="Direct link to 2.3. Execute pileup">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-bamtoscidx-ad3c2db72cce5b6733b0e39637c7a997.png" style="width:60%" class="img_ev3q"><i class="fa-solid fa-arrow-right fa-lg"></i><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-bamtoscidx-output-ee4852e4db0dbbcab067e03bfe1f8537.png" style="width:40%" class="img_ev3q"></div><p>This should output a <code>12141_filtered_READ1.tab</code> file.</p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="3-optional-normalize-the-genome-track">3. (Optional) Normalize the genome track<a href="#3-optional-normalize-the-genome-track" class="hash-link" aria-label="Direct link to 3. (Optional) Normalize the genome track" title="Direct link to 3. (Optional) Normalize the genome track">​</a></h3><p><em>Skip this if you would like to make a track of the raw read counts.</em></p><p>ScriptManager includes a &quot;Calculate Scaling Factor&quot; tool under the &quot;Read Analysis&quot; tool group that can calculate scaling factors based on several methodologies including total tag normalization and the Normalization of ChIP-seq (NCIS) protocol. <a href="/scriptmanager-docs/docs/read-analysis/scaling-factor">Read more in the docs.</a> Please be sure to check that your <a href="/scriptmanager-docs/docs/read-analysis/scaling-factor">normalization method is appropriate</a> for your data.</p><h4 class="anchor anchorWithStickyNavbar_LWe7" id="31-navigate-to-read-analysis-️-calculate-scaling-factor-to-calculate-a-scaling-factor">3.1. Navigate to Read Analysis ➡️ <a href="/scriptmanager-docs/docs/read-analysis/scaling-factor"><strong>Calculate Scaling Factor</strong></a> to calculate a scaling factor<a href="#31-navigate-to-read-analysis-️-calculate-scaling-factor-to-calculate-a-scaling-factor" class="hash-link" aria-label="Direct link to 31-navigate-to-read-analysis-️-calculate-scaling-factor-to-calculate-a-scaling-factor" title="Direct link to 31-navigate-to-read-analysis-️-calculate-scaling-factor-to-calculate-a-scaling-factor">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/maingui-scalingfactor-b82954571a684d31b33b299dcfff78a3.png" style="width:60%" class="img_ev3q"></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="32-select-the-bam-file-you-want-to-create-the-data-track-from">3.2. Select the BAM file you want to create the data track from<a href="#32-select-the-bam-file-you-want-to-create-the-data-track-from" class="hash-link" aria-label="Direct link to 3.2. Select the BAM file you want to create the data track from" title="Direct link to 3.2. Select the BAM file you want to create the data track from">​</a></h4><h4 class="anchor anchorWithStickyNavbar_LWe7" id="33-select-a-normalization-method-and-execute">3.3. Select a normalization method and execute<a href="#33-select-a-normalization-method-and-execute" class="hash-link" aria-label="Direct link to 3.3. Select a normalization method and execute" title="Direct link to 3.3. Select a normalization method and execute">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-scalingfactor-a34b91e737de24602674d967d44e0901.png" style="width:60%" class="img_ev3q"><i class="fa-solid fa-arrow-right fa-lg"></i><div><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-scalingfactor-output-6a4014ce47e5493129f82fdef3a1bfaa.png" class="img_ev3q"><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-scalingfactor-dialog-ef95d930872de2663dbe5b30ee40d9a0.png" class="img_ev3q"></div></div><p>For this BAM file, the total tag method gives a scaling factor of <code>9.562259045426815</code>.</p><h4 class="anchor anchorWithStickyNavbar_LWe7" id="34-navigate-to-read-analysis-️-scale-matrix-to-apply-the-normalization-to-the-data-track">3.4. Navigate to Read Analysis ➡️ <a href="/scriptmanager-docs/docs/read-analysis/scale-matrix"><strong>Scale Matrix</strong></a> to apply the normalization to the data track<a href="#34-navigate-to-read-analysis-️-scale-matrix-to-apply-the-normalization-to-the-data-track" class="hash-link" aria-label="Direct link to 34-navigate-to-read-analysis-️-scale-matrix-to-apply-the-normalization-to-the-data-track" title="Direct link to 34-navigate-to-read-analysis-️-scale-matrix-to-apply-the-normalization-to-the-data-track">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/maingui-scalematrix-f62550542d34549096c181a3c9062f5c.png" style="width:60%" class="img_ev3q"></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="32-input-the-normalization-factor-and-adjust-the-start-row-to-2">3.2. Input the normalization factor and adjust the start row to 2<a href="#32-input-the-normalization-factor-and-adjust-the-start-row-to-2" class="hash-link" aria-label="Direct link to 3.2. Input the normalization factor and adjust the start row to 2" title="Direct link to 3.2. Input the normalization factor and adjust the start row to 2">​</a></h4><h4 class="anchor anchorWithStickyNavbar_LWe7" id="33-load-the-file-and-execute-matrix-scaling">3.3. Load the file and execute matrix scaling<a href="#33-load-the-file-and-execute-matrix-scaling" class="hash-link" aria-label="Direct link to 3.3. Load the file and execute matrix scaling" title="Direct link to 3.3. Load the file and execute matrix scaling">​</a></h4><p>You will be loading in the scIDX file (not the BAM).</p><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-scalematrix-02240129e3eefbd76eb286091e1c9d90.png" style="width:60%" class="img_ev3q"><i class="fa-solid fa-arrow-right fa-lg"></i><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-scalematrix-dialog-4df8e8f3d993fda3cd3422a4d575d7b8.png" style="width:40%" class="img_ev3q"></div><p>This should output a <code>12141_filtered_READ1_SCALE.tab</code> file.</p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="4-convert-scidx-to-bigwig">4. Convert scIDX to BigWig<a href="#4-convert-scidx-to-bigwig" class="hash-link" aria-label="Direct link to 4. Convert scIDX to BigWig" title="Direct link to 4. Convert scIDX to BigWig">​</a></h3><p>Since ScriptManager currently does not support converting BedGraph files to BigWigs, this step must be done in your terminal/on the command-prompt.</p><h4 class="anchor anchorWithStickyNavbar_LWe7" id="41-turn-your-scidx-file-into-a-bedgraph-file">4.1. Turn your scIDX file into a bedGraph file<a href="#41-turn-your-scidx-file-into-a-bedgraph-file" class="hash-link" aria-label="Direct link to 4.1. Turn your scIDX file into a bedGraph file" title="Direct link to 4.1. Turn your scIDX file into a bedGraph file">​</a></h4><div class="tabs-container tabList__CuJ"><ul role="tablist" aria-orientation="horizontal" class="tabs"><li role="tab" tabindex="0" aria-selected="true" class="tabs__item tabItem_LNqP tabs__item--active">Shell</li><li role="tab" tabindex="-1" aria-selected="false" class="tabs__item tabItem_LNqP">Excel</li></ul><div class="margin-top--md"><div role="tabpanel" class="tabItem_Ymn6"><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">grep -v -f sacCer3.chrom.sizes 12141_filtered_READ1_SCALE.tab &gt; 12141_filtered_READ1_SCALE.filtered.tab</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">sed &#x27;1d;2d&#x27; 12141_filtered_READ1_SCALE.filtered.tab | awk &#x27;{OFS=&quot;\t&quot;}{FS=&quot;\t&quot;}{print $1, $2, $2+1, $3}&#x27; |sort -k1,1 -k2,2n &gt; 12141_filtered_READ1_SCALE_forward.bedgraph</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">sed &#x27;1d;2d&#x27; 12141_filtered_READ1_SCALE.filtered.tab | awk &#x27;{OFS=&quot;\t&quot;}{FS=&quot;\t&quot;}{print $1, $2, $2+1, $4}&#x27; |sort -k1,1 -k2,2n &gt; 12141_filtered_READ1_SCALE_reverse.bedgraph</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div></div><div role="tabpanel" class="tabItem_Ymn6" hidden=""><p>Open your file in excel and strip the file of the first two rows:</p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">#2023-03-10 12:52:40.439;12141_filtered.bam;READ1</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">chrom   index   forward reverse value</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>You may also have read mapping to the very edges of the chromosome. These entries will also have to be removed.</p><p>Then rearrange the columns into BedGraph format by copying them in the following order (3rd column is the value of col2 plus one):</p><ul><li><code>12141_filtered_READ1_SCALE_forward.bedGraph</code>: col1, col2, (col2 +1), col3</li><li><code>12141_filtered_READ1_SCALE_forward.bedGraph</code>: col1, col2, (col2 +1), col4</li></ul><p>...and then sort each of them by the first and then the second columns. This format is tab-delimited (not comma-delimited!) so save as a tab-delimited text file. Do not save this as an Excel file format (<code>.xlsx</code>, <code>.xls</code>, etc). This should be a <code>.tsv</code> or <code>.tab</code> format.</p></div></div></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="42-turn-your-bedgraph-file-into-a-bigwig-file">4.2. Turn your bedGraph file into a BigWig file<a href="#42-turn-your-bedgraph-file-into-a-bigwig-file" class="hash-link" aria-label="Direct link to 4.2. Turn your bedGraph file into a BigWig file" title="Direct link to 4.2. Turn your bedGraph file into a BigWig file">​</a></h4><div class="tabs-container tabList__CuJ"><ul role="tablist" aria-orientation="horizontal" class="tabs"><li role="tab" tabindex="0" aria-selected="true" class="tabs__item tabItem_LNqP tabs__item--active">Shell</li></ul><div class="margin-top--md"><div role="tabpanel" class="tabItem_Ymn6"><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">./bedGraphToBigWig 12141_filtered_READ1_SCALE_forward.bedGraph sacCer3.chrom.sizes my-track-basename_forward.bw</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">./bedGraphToBigWig 12141_filtered_READ1_SCALE_reverse.bedGraph sacCer3.chrom.sizes my-track-basename_reverse.bw</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div></div></div></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="43-load-the-bigwig-into-your-favorite-genome-browser">4.3. Load the BigWig into your <a href="https://software.broadinstitute.org/software/igv/download" target="_blank" rel="noopener noreferrer">favorite genome browser</a><a href="#43-load-the-bigwig-into-your-favorite-genome-browser" class="hash-link" aria-label="Direct link to 43-load-the-bigwig-into-your-favorite-genome-browser" title="Direct link to 43-load-the-bigwig-into-your-favorite-genome-browser">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/IGV_BrowserShot-1ff8f8b15497181fcb1800cc6fc746f5.png" style="width:80%" class="img_ev3q"></div><p>Tah dah! You&#x27;ve made the genome tracks for a ChIP-exo dataset!</p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="command-line-shell-script">Command-Line shell script<a href="#command-line-shell-script" class="hash-link" aria-label="Direct link to Command-Line shell script" title="Direct link to Command-Line shell script">​</a></h2><p>The following shell commands mirror the manipulations described above to create BigWig genome tracks. This can serve as a template for you to write out your own workflows as bash scripts that execute command-line style ScriptManager.</p><div class="language-bash codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-bash codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token assign-left variable" style="color:#36acaa">SCRIPTMANAGER</span><span class="token operator" style="color:#393A34">=</span><span class="token plain">/path/to/ScriptManager.jar</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token assign-left variable" style="color:#36acaa">BGTOBW</span><span class="token operator" style="color:#393A34">=</span><span class="token plain">/path/to/bedGraphToBigWig</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token assign-left variable" style="color:#36acaa">CHRMSZ</span><span class="token operator" style="color:#393A34">=</span><span class="token plain">/path/to/sacCer3.chrom.sizes</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token assign-left variable" style="color:#36acaa">BAMFILE</span><span class="token operator" style="color:#393A34">=</span><span class="token plain">/path/to/12141_filtered.bam</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token assign-left variable" style="color:#36acaa">OUTPUT</span><span class="token operator" style="color:#393A34">=</span><span class="token plain">/path/to/my-track-basename</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token assign-left variable" style="color:#36acaa">FACTOR</span><span class="token operator" style="color:#393A34">=</span><span class="token number" style="color:#36acaa">9.562259045426815</span><span class="token plain">  </span><span class="token comment" style="color:#999988;font-style:italic"># calculate this separately</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Index BAM if not already indexed</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token punctuation" style="color:#393A34">[</span><span class="token plain"> -f </span><span class="token variable" style="color:#36acaa">$BAMFILE</span><span class="token plain">.bai </span><span class="token punctuation" style="color:#393A34">]</span><span class="token plain"> </span><span class="token operator" style="color:#393A34">||</span><span class="token plain"> samtools index </span><span class="token variable" style="color:#36acaa">$BAMFILE</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Pileup BAM along whole genome</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">java -jar </span><span class="token variable" style="color:#36acaa">$SCRIPTMANAGER</span><span class="token plain"> bam-format-converter bam-to-scidx </span><span class="token variable" style="color:#36acaa">$BAMFILE</span><span class="token plain"> -o </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token plain">.raw.tab</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Only include pileups from chromosomes in the sacCer3.chrom.sizes reference</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">grep</span><span class="token plain"> -v -f </span><span class="token variable" style="color:#36acaa">$CHRMSZ</span><span class="token plain"> </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token plain">.raw.tab </span><span class="token operator" style="color:#393A34">&gt;</span><span class="token plain"> </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token plain">.filtered.tab</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Scale SCIDX output (optional)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">java -jar </span><span class="token variable" style="color:#36acaa">$SCRIPTMANAGER</span><span class="token plain"> read-analysis scale-matrix </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token plain">.filtered.tab -s </span><span class="token variable" style="color:#36acaa">$FACTOR</span><span class="token plain"> -r </span><span class="token number" style="color:#36acaa">2</span><span class="token plain"> -o </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token plain">.scaled.tab</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Convert to forward/reverse Bedgraph tracks</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">sed</span><span class="token plain"> </span><span class="token string" style="color:#e3116c">&#x27;1d;2d&#x27;</span><span class="token plain"> </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token plain">.scaled.tab </span><span class="token operator" style="color:#393A34">|</span><span class="token plain"> </span><span class="token function" style="color:#d73a49">awk</span><span class="token plain"> </span><span class="token string" style="color:#e3116c">&#x27;{OFS=&quot;\t&quot;}{FS=&quot;\t&quot;}{print $1, $2, $2+1, $3}&#x27;</span><span class="token plain"> </span><span class="token operator" style="color:#393A34">|</span><span class="token function" style="color:#d73a49">sort</span><span class="token plain"> -k1,1 -k2,2n </span><span class="token operator" style="color:#393A34">&gt;</span><span class="token plain"> </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token plain">.forward.bedgraph</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">sed</span><span class="token plain"> </span><span class="token string" style="color:#e3116c">&#x27;1d;2d&#x27;</span><span class="token plain"> </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token plain">.scaled.tab </span><span class="token operator" style="color:#393A34">|</span><span class="token plain"> </span><span class="token function" style="color:#d73a49">awk</span><span class="token plain"> </span><span class="token string" style="color:#e3116c">&#x27;{OFS=&quot;\t&quot;}{FS=&quot;\t&quot;}{print $1, $2, $2+1, $4}&#x27;</span><span class="token plain"> </span><span class="token operator" style="color:#393A34">|</span><span class="token function" style="color:#d73a49">sort</span><span class="token plain"> -k1,1 -k2,2n </span><span class="token operator" style="color:#393A34">&gt;</span><span class="token plain"> </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token plain">.reverse.bedgraph</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Compress to BigWig format</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token variable" style="color:#36acaa">$BGTOBW</span><span class="token plain"> </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token plain">.forward.bedgraph </span><span class="token variable" style="color:#36acaa">$CHRMSZ</span><span class="token plain"> </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token punctuation" style="color:#393A34">\</span><span class="token plain">_forward.bw</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token variable" style="color:#36acaa">$BGTOBW</span><span class="token plain"> </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token plain">.reverse.bedgraph </span><span class="token variable" style="color:#36acaa">$CHRMSZ</span><span class="token plain"> </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token punctuation" style="color:#393A34">\</span><span class="token plain">_reverse.bw</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Clean-up</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">rm</span><span class="token plain"> </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token plain">.raw.tab </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token plain">.filtered.tab </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token plain">.scaled.tab </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token plain">.forward.bedgraph </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token plain">.reverse.bedgraph</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Output files:</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic">#  - /path/to/my-track-basename_forward.bw</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic">#  - /path/to/my-track-basename_reverse.bw</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"></div><div class="col lastUpdated_vwxv"><span class="theme-last-updated">Last updated<!-- --> on <b><time datetime="2023-03-10T22:22:46.000Z">Mar 10, 2023</time></b> by <b>Olivia Wen-Mei Lang</b></span></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--prev" href="/scriptmanager-docs/docs/Tutorials/genomic-features-tutorial"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Genomic Features Tutorial</div></a><a class="pagination-nav__link pagination-nav__link--next" href="/scriptmanager-docs/docs/References/file-formats"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">File Formats</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_bqdL thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#download-scriptmanager-v014" class="table-of-contents__link toc-highlight">Download ScriptManager (v0.14):</a></li><li><a href="#download-uscs-binary-bedgraphtobigwig" class="table-of-contents__link toc-highlight">Download USCS binary <code>bedGraphToBigWig</code></a></li><li><a href="#download-data" class="table-of-contents__link toc-highlight">Download data</a><ul><li><a href="#bam-file" class="table-of-contents__link toc-highlight">BAM File</a></li><li><a href="#xxxxchromsizes-reference-file" class="table-of-contents__link toc-highlight">XXXX.chrom.sizes Reference File</a></li></ul></li><li><a href="#generate-the-genome-tracks" class="table-of-contents__link toc-highlight">Generate the Genome Tracks</a><ul><li><a href="#1-open-scriptmanager" class="table-of-contents__link toc-highlight">1. Open ScriptManager</a></li><li><a href="#2-create-a-whole-genome-pileup-using-bam-format-converter-tools" class="table-of-contents__link toc-highlight">2. Create a whole-genome pileup using BAM Format Converter Tools</a></li><li><a href="#3-optional-normalize-the-genome-track" class="table-of-contents__link toc-highlight">3. (Optional) Normalize the genome track</a></li><li><a href="#4-convert-scidx-to-bigwig" class="table-of-contents__link toc-highlight">4. Convert scIDX to BigWig</a></li></ul></li><li><a href="#command-line-shell-script" class="table-of-contents__link toc-highlight">Command-Line shell script</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/scriptmanager-docs/docs/References/tool-index">Tool Index (A-Z)</a></li><li class="footer__item"><a class="footer__link-item" href="/scriptmanager-docs/docs/References/file-formats">File Formats</a></li></ul></div><div class="col footer__col"><div class="footer__title">Community</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://pughlab.mbg.cornell.edu" target="_blank" rel="noopener noreferrer" class="footer__link-item">Pugh Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://williamkmlai.github.io" target="_blank" rel="noopener noreferrer" class="footer__link-item">Lai Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div><div class="col footer__col"><div class="footer__title">Other Tools We Develop</div><ul class="footer__items clean-list"><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/scriptmanager-docs" target="_blank" rel="noopener noreferrer" class="footer__link-item">ScriptManager<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://github.com/seqcode/pegr" target="_blank" rel="noopener noreferrer" class="footer__link-item">PEGR<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/stencil" target="_blank" rel="noopener noreferrer" class="footer__link-item">STENCIL<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/GenoPipe-docs" target="_blank" rel="noopener noreferrer" class="footer__link-item">GenoPipe<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="margin-bottom--sm"><img src="/scriptmanager-docs/img/pughlab_logo.svg" alt="Pugh Lab Logo" class="themedImage_ToTc themedImage--light_HNdA footer__logo"><img src="/scriptmanager-docs/img/pughlab_logo.svg" alt="Pugh Lab Logo" class="themedImage_ToTc themedImage--dark_i4oU footer__logo"></div><div class="footer__copyright">Copyright © 2023 ScriptManager. Built with Docusaurus.</div></div></div></footer></div>
<script src="/scriptmanager-docs/assets/js/runtime~main.3250e9c7.js"></script>
<script src="/scriptmanager-docs/assets/js/main.510075bb.js"></script>
</body>
</html>