"use strict";(self.webpackChunkscriptmanager_docs=self.webpackChunkscriptmanager_docs||[]).push([[3763],{3905:(t,e,a)=>{a.d(e,{Zo:()=>m,kt:()=>f});var r=a(7294);function n(t,e,a){return e in t?Object.defineProperty(t,e,{value:a,enumerable:!0,configurable:!0,writable:!0}):t[e]=a,t}function l(t,e){var a=Object.keys(t);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(t);e&&(r=r.filter((function(e){return Object.getOwnPropertyDescriptor(t,e).enumerable}))),a.push.apply(a,r)}return a}function i(t){for(var e=1;e<arguments.length;e++){var a=null!=arguments[e]?arguments[e]:{};e%2?l(Object(a),!0).forEach((function(e){n(t,e,a[e])})):Object.getOwnPropertyDescriptors?Object.defineProperties(t,Object.getOwnPropertyDescriptors(a)):l(Object(a)).forEach((function(e){Object.defineProperty(t,e,Object.getOwnPropertyDescriptor(a,e))}))}return t}function o(t,e){if(null==t)return{};var a,r,n=function(t,e){if(null==t)return{};var a,r,n={},l=Object.keys(t);for(r=0;r<l.length;r++)a=l[r],e.indexOf(a)>=0||(n[a]=t[a]);return n}(t,e);if(Object.getOwnPropertySymbols){var l=Object.getOwnPropertySymbols(t);for(r=0;r<l.length;r++)a=l[r],e.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(t,a)&&(n[a]=t[a])}return n}var d=r.createContext({}),s=function(t){var e=r.useContext(d),a=e;return t&&(a="function"==typeof t?t(e):i(i({},e),t)),a},m=function(t){var e=s(t.components);return r.createElement(d.Provider,{value:e},t.children)},p={inlineCode:"code",wrapper:function(t){var e=t.children;return r.createElement(r.Fragment,{},e)}},c=r.forwardRef((function(t,e){var a=t.components,n=t.mdxType,l=t.originalType,d=t.parentName,m=o(t,["components","mdxType","originalType","parentName"]),c=s(a),f=n,k=c["".concat(d,".").concat(f)]||c[f]||p[f]||l;return a?r.createElement(k,i(i({ref:e},m),{},{components:a})):r.createElement(k,i({ref:e},m))}));function f(t,e){var a=arguments,n=e&&e.mdxType;if("string"==typeof t||n){var l=a.length,i=new Array(l);i[0]=c;var o={};for(var d in e)hasOwnProperty.call(e,d)&&(o[d]=e[d]);o.originalType=t,o.mdxType="string"==typeof t?t:n,i[1]=o;for(var s=2;s<l;s++)i[s]=a[s];return r.createElement.apply(null,i)}return r.createElement.apply(null,a)}c.displayName="MDXCreateElement"},5326:(t,e,a)=>{a.r(e),a.d(e,{assets:()=>d,contentTitle:()=>i,default:()=>p,frontMatter:()=>l,metadata:()=>o,toc:()=>s});var r=a(7462),n=(a(7294),a(3905));const l={id:"tool-group",title:"Tool Groups",sidebar_label:"Tool Groups"},i=void 0,o={unversionedId:"Guides/tool-group",id:"Guides/tool-group",title:"Tool Groups",description:"BAM Format Converter",source:"@site/docs/Guides/tool-group.md",sourceDirName:"Guides",slug:"/Guides/tool-group",permalink:"/scriptmanager-docs/docs/Guides/tool-group",draft:!1,tags:[],version:"current",lastUpdatedBy:"Olivia Wen-Mei Lang",lastUpdatedAt:1661543261,formattedLastUpdatedAt:"Aug 26, 2022",frontMatter:{id:"tool-group",title:"Tool Groups",sidebar_label:"Tool Groups"},sidebar:"someSidebar",previous:{title:"Command Line",permalink:"/scriptmanager-docs/docs/Guides/command-line"},next:{title:"Color Selection Guide",permalink:"/scriptmanager-docs/docs/Guides/color-guide"}},d={},s=[{value:"BAM Format Converter",id:"bam-format-converter",level:2},{value:"BAM Manipulation",id:"bam-manipulation",level:2},{value:"BAM Statistics",id:"bam-statistics",level:2},{value:"Coordinate Manipulation",id:"coordinate-manipulation",level:2},{value:"Figure Generation",id:"figure-generation",level:2},{value:"File Utilities",id:"file-utilities",level:2},{value:"Peak Analysis",id:"peak-analysis",level:2},{value:"Read Analysis",id:"read-analysis",level:2},{value:"Sequence Analysis",id:"sequence-analysis",level:2}],m={toc:s};function p(t){let{components:e,...a}=t;return(0,n.kt)("wrapper",(0,r.Z)({},m,a,{components:e,mdxType:"MDXLayout"}),(0,n.kt)("h2",{id:"bam-format-converter"},"BAM Format Converter"),(0,n.kt)("p",null,(0,n.kt)("em",{parentName:"p"},"Description of BAM-Format-Converter tool")),(0,n.kt)("table",null,(0,n.kt)("thead",{parentName:"table"},(0,n.kt)("tr",{parentName:"thead"},(0,n.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,n.kt)("th",{parentName:"tr",align:null},"Input"),(0,n.kt)("th",{parentName:"tr",align:null},"Output"),(0,n.kt)("th",{parentName:"tr",align:null},"Description"))),(0,n.kt)("tbody",{parentName:"table"},(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/bam-format-converter/bam-to-scidx"},(0,n.kt)("inlineCode",{parentName:"a"},"bam-to-scidx"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"scIDX")),(0,n.kt)("td",{parentName:"tr",align:null},"Convert BAM file to scIDX file.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/bam-format-converter/bam-to-gff"},(0,n.kt)("inlineCode",{parentName:"a"},"bam-to-gff"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"GFF")),(0,n.kt)("td",{parentName:"tr",align:null},"Convert BAM file to GFF file.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/bam-format-converter/bam-to-bed"},(0,n.kt)("inlineCode",{parentName:"a"},"bam-to-bed"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED")),(0,n.kt)("td",{parentName:"tr",align:null},"Convert BAM file to BED file.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/bam-format-converter/bam-to-bedgraph"},(0,n.kt)("inlineCode",{parentName:"a"},"bam-to-bedgraph"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"bedGraph")),(0,n.kt)("td",{parentName:"tr",align:null},"Convert BAM file to bedGraph file.")))),(0,n.kt)("h2",{id:"bam-manipulation"},"BAM Manipulation"),(0,n.kt)("p",null,(0,n.kt)("em",{parentName:"p"},"General description of BAM Manipulation")),(0,n.kt)("table",null,(0,n.kt)("thead",{parentName:"table"},(0,n.kt)("tr",{parentName:"thead"},(0,n.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,n.kt)("th",{parentName:"tr",align:null},"Input"),(0,n.kt)("th",{parentName:"tr",align:null},"Output"),(0,n.kt)("th",{parentName:"tr",align:null},"Description"))),(0,n.kt)("tbody",{parentName:"table"},(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/bam-manipulation/bam-indexer"},(0,n.kt)("inlineCode",{parentName:"a"},"bam-indexer"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAI")),(0,n.kt)("td",{parentName:"tr",align:null},"Generates BAI Index for input BAM files. Output BAI is in the same directory as input BAM file.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/bam-manipulation/sort-bam"},(0,n.kt)("inlineCode",{parentName:"a"},"sort-bam"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},"Sort BAM files in order to efficiently extract and manipulate.\\nRAM intensive process. If program freezes, increase JAVA heap size.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/bam-manipulation/remove-duplicates"},(0,n.kt)("inlineCode",{parentName:"a"},"remove-duplicates"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},"Removes duplicate reads in Paired-End sequencing given identical 5' read locations. RAM intensive process. If program freezes, increase JAVA heap size.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/bam-manipulation/merge-bam"},(0,n.kt)("inlineCode",{parentName:"a"},"merge-bam"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM"),"[","..","]"),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},"Merges Multiple BAM files into single BAM file. Sorting is performed automatically. RAM intensive process. If program freezes, increase JAVA heap size.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/bam-manipulation/filter-pip-seq"},(0,n.kt)("inlineCode",{parentName:"a"},"filter-pip-seq"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},"Filter BAM file by -1 nucleotide. Requires genome FASTA file. Note this program does not index the resulting BAM file and user must use appropriate samtools command to generate BAI.")))),(0,n.kt)("h2",{id:"bam-statistics"},"BAM Statistics"),(0,n.kt)("table",null,(0,n.kt)("thead",{parentName:"table"},(0,n.kt)("tr",{parentName:"thead"},(0,n.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,n.kt)("th",{parentName:"tr",align:null},"Input"),(0,n.kt)("th",{parentName:"tr",align:null},"Output"),(0,n.kt)("th",{parentName:"tr",align:null},"Description"))),(0,n.kt)("tbody",{parentName:"table"},(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/bam-statistics/se-stat"},"se-stat")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"TXT")),(0,n.kt)("td",{parentName:"tr",align:null},"Genome-Genome correlations for replicate comparisons given multiple sorted and indexed (BAI) BAM files.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/bam-statistics/pe-stat"},"pe-stat")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"TXT")),(0,n.kt)("td",{parentName:"tr",align:null},"Generates Insert-size Histogram statistic (GEO requirement) and outputs BAM Header including alignment statistics and parameters given a sorted and indexed (BAI) paired-end BAM File.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/bam-statistics/bam-correlation"},"bam-correlation"),"bam-correlation)"),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"TXT")),(0,n.kt)("td",{parentName:"tr",align:null},"Output BAM Header including alignment statistics and parameters given any indexed (BAI) BAM File.")))),(0,n.kt)("h2",{id:"coordinate-manipulation"},"Coordinate Manipulation"),(0,n.kt)("p",null,(0,n.kt)("em",{parentName:"p"},"Coordinate Manipulation tools description")),(0,n.kt)("table",null,(0,n.kt)("thead",{parentName:"table"},(0,n.kt)("tr",{parentName:"thead"},(0,n.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,n.kt)("th",{parentName:"tr",align:null},"Input"),(0,n.kt)("th",{parentName:"tr",align:null},"Output"),(0,n.kt)("th",{parentName:"tr",align:null},"Description"))),(0,n.kt)("tbody",{parentName:"table"},(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/coordinate-manipulation/expand-bed"},"expand-bed")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED")),(0,n.kt)("td",{parentName:"tr",align:null},"Expands input BED file by adding positions to the border or around the center.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/coordinate-manipulation/expand-gff"},"expand-gff")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"GFF")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"GFF")),(0,n.kt)("td",{parentName:"tr",align:null},"Expands input GFF file by adding positions to the border or around the center.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/coordinate-manipulation/bed-to-gff"},"bed-to-gff")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"GFF")),(0,n.kt)("td",{parentName:"tr",align:null},"Converts BED file format to GFF file format.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/coordinate-manipulation/gff-to-bed"},"gff-to-bed")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"GFF")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED")),(0,n.kt)("td",{parentName:"tr",align:null},"Converts GFF file format to BED file format.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/coordinate-manipulation/sort-bed"},"sort-bed")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED")),(0,n.kt)("td",{parentName:"tr",align:null},"Sort a CDT file and its corresponding BED file by the total score in the CDT file across the specified interval.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/coordinate-manipulation/sort-gff"},"sort-gff")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"GFF")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"GFF")),(0,n.kt)("td",{parentName:"tr",align:null},"Sort a CDT file and its corresponding GFF file by the total score in the CDT file across the specified interval.")))),(0,n.kt)("h2",{id:"figure-generation"},"Figure Generation"),(0,n.kt)("table",null,(0,n.kt)("thead",{parentName:"table"},(0,n.kt)("tr",{parentName:"thead"},(0,n.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,n.kt)("th",{parentName:"tr",align:null},"Input"),(0,n.kt)("th",{parentName:"tr",align:null},"Output"),(0,n.kt)("th",{parentName:"tr",align:null},"Description"))),(0,n.kt)("tbody",{parentName:"table"},(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/figure-generation/heatmap"},"heatmap")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"CDT")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"PNG")),(0,n.kt)("td",{parentName:"tr",align:null},"Generate heatmap using CDT files.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/figure-generation/merge-heatmap"},"merge-heatmap")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"PNG")," + ",(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"PNG")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"PNG")),(0,n.kt)("td",{parentName:"tr",align:null},"Merge Sense and Antisense png heatmaps.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/figure-generation/four-color"},"four-color")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"FASTA")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"PNG")),(0,n.kt)("td",{parentName:"tr",align:null},"Generate 4Color sequence plot given FASTA file and user-defined RGB colors.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/figure-generation/composite-plot"},"composite-plot")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"CDT")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"PNG")),(0,n.kt)("td",{parentName:"tr",align:null},"Generate a Composite Plot PNG from composite data like the output in TagPileup.")))),(0,n.kt)("h2",{id:"file-utilities"},"File Utilities"),(0,n.kt)("p",null,(0,n.kt)("em",{parentName:"p"},"File Utilities tools description")),(0,n.kt)("table",null,(0,n.kt)("thead",{parentName:"table"},(0,n.kt)("tr",{parentName:"thead"},(0,n.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,n.kt)("th",{parentName:"tr",align:null},"Input"),(0,n.kt)("th",{parentName:"tr",align:null},"Output"),(0,n.kt)("th",{parentName:"tr",align:null},"Description"))),(0,n.kt)("tbody",{parentName:"table"},(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-utilities/md5checksum"},"md5checksum")),(0,n.kt)("td",{parentName:"tr",align:null},"*"),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"TXT")),(0,n.kt)("td",{parentName:"tr",align:null},"Calculate MD5 checksum for an input file.")))),(0,n.kt)("h2",{id:"peak-analysis"},"Peak Analysis"),(0,n.kt)("p",null,(0,n.kt)("em",{parentName:"p"},"Peak Analysis tools description")),(0,n.kt)("table",null,(0,n.kt)("thead",{parentName:"table"},(0,n.kt)("tr",{parentName:"thead"},(0,n.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,n.kt)("th",{parentName:"tr",align:null},"Input"),(0,n.kt)("th",{parentName:"tr",align:null},"Output"),(0,n.kt)("th",{parentName:"tr",align:null},"Description"))),(0,n.kt)("tbody",{parentName:"table"},(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/peak-analysis/peak-align-ref"},(0,n.kt)("inlineCode",{parentName:"a"},"peak-align-ref"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED(ref)")," + ",(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED(peak)")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"CDT")),(0,n.kt)("td",{parentName:"tr",align:null},"Align BED peaks to Reference BED file creating CDT files for heatmap generation.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/peak-analysis/filter-bed"},(0,n.kt)("inlineCode",{parentName:"a"},"filter-bed"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED(cluster)")," + ",(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED(filter)")),(0,n.kt)("td",{parentName:"tr",align:null},"Filter BED file using user-specified exclusion zone using the score column to determine which peak to retain.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/peak-analysis/tile-genome"},(0,n.kt)("inlineCode",{parentName:"a"},"tile-genome"))),(0,n.kt)("td",{parentName:"tr",align:null},"genomeName"),(0,n.kt)("td",{parentName:"tr",align:null},"[",(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED"),"|",(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"GFF"),"]"),(0,n.kt)("td",{parentName:"tr",align:null},"Generate a coordinate file that tiles (non-overlapping) across an entire genome.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/peak-analysis/rand-coord"},(0,n.kt)("inlineCode",{parentName:"a"},"rand-coord"))),(0,n.kt)("td",{parentName:"tr",align:null},"genomeName"),(0,n.kt)("td",{parentName:"tr",align:null},"[",(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED"),"|",(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"GFF"),"]"),(0,n.kt)("td",{parentName:"tr",align:null},"Generate a coordinate file that tiles (non-overlapping) across an entire genome.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/peak-analysis/signal-dup"},(0,n.kt)("inlineCode",{parentName:"a"},"signal-dup"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"GFF")," + ",(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"TXT")),(0,n.kt)("td",{parentName:"tr",align:null},"Calculate duplication statistics at user-specified regions.")))),(0,n.kt)("h2",{id:"read-analysis"},"Read Analysis"),(0,n.kt)("p",null,"These Read Analysis tools are mostly tools that work with the ","[CDT or other tab-delimited Matrix file formats][file-format]","."),(0,n.kt)("p",null,"Most commonly used  is the TagPileup tool."),(0,n.kt)("p",null,"Others manipulate files to calculate normalization factors, normalize the data, or other manipulations and transformations of read info."),(0,n.kt)("table",null,(0,n.kt)("thead",{parentName:"table"},(0,n.kt)("tr",{parentName:"thead"},(0,n.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,n.kt)("th",{parentName:"tr",align:null},"Input"),(0,n.kt)("th",{parentName:"tr",align:null},"Output"),(0,n.kt)("th",{parentName:"tr",align:null},"Description"))),(0,n.kt)("tbody",{parentName:"table"},(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/read-analysis/tag-pileup"},(0,n.kt)("inlineCode",{parentName:"a"},"tag-pileup"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED")," + ",(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"CDT")),(0,n.kt)("td",{parentName:"tr",align:null},"Pileup 5' ends of aligned tags given BED and BAM files according to user-defined parameters.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/read-analysis/scaling-factor"},(0,n.kt)("inlineCode",{parentName:"a"},"scaling-factor"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BAM")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"TXT")),(0,n.kt)("td",{parentName:"tr",align:null},"Calculate the factor as either total tag normalization or normalization of ChIP-seq data with control. (PMID:22883957)")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/read-analysis/scale-matrix"},(0,n.kt)("inlineCode",{parentName:"a"},"scale-matrix"))),(0,n.kt)("td",{parentName:"tr",align:null},"[",(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"CDT"),"|",(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"TAB"),"]"),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"TAB"),"[","..","]"),(0,n.kt)("td",{parentName:"tr",align:null},"Apply a user-specified scaling factor to tab-delimited matrix data.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/read-analysis/aggregate-data"},(0,n.kt)("inlineCode",{parentName:"a"},"aggregate-data"))),(0,n.kt)("td",{parentName:"tr",align:null},"[",(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"CDT"),"|",(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"TAB"),"]","[","..","]"),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"TAB"),"[","..","]"),(0,n.kt)("td",{parentName:"tr",align:null},"The AggregateData tool is used to process a bunch of matrix files into one matrix file.")))),(0,n.kt)("h2",{id:"sequence-analysis"},"Sequence Analysis"),(0,n.kt)("p",null,(0,n.kt)("em",{parentName:"p"},"Sequence Analysis tools description")),(0,n.kt)("table",null,(0,n.kt)("thead",{parentName:"table"},(0,n.kt)("tr",{parentName:"thead"},(0,n.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,n.kt)("th",{parentName:"tr",align:null},"Input"),(0,n.kt)("th",{parentName:"tr",align:null},"Output"),(0,n.kt)("th",{parentName:"tr",align:null},"Description"))),(0,n.kt)("tbody",{parentName:"table"},(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/sequence-analysis/fasta-extract"},(0,n.kt)("inlineCode",{parentName:"a"},"fasta-extract"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED")," + ",(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"FASTA(genome)")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"FASTA")),(0,n.kt)("td",{parentName:"tr",align:null},"Generate FASTA file from indexed Genome FASTA file and BED file. Script will generate FAI index if not present in Genome FASTA folder.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/sequence-analysis/randomize-fasta"},(0,n.kt)("inlineCode",{parentName:"a"},"randomize-fasta"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"FASTA")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"FASTA")),(0,n.kt)("td",{parentName:"tr",align:null},"Randomizes FASTA sequence for each input entry")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/sequence-analysis/search-motif"},(0,n.kt)("inlineCode",{parentName:"a"},"search-motif"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"FASTA")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED")),(0,n.kt)("td",{parentName:"tr",align:null},"Search for an IUPAC DNA sequence motif in FASTA files with mismatches allowed.")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/sequence-analysis/dna-shape-bed"},(0,n.kt)("inlineCode",{parentName:"a"},"dna-shape-bed"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"BED")," + ",(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"FASTA(genome)")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"CDT")),(0,n.kt)("td",{parentName:"tr",align:null},"Calculate intrinsic DNA shape parameters given BED file and Genome FASTA file. Based on Roh's lab ",(0,n.kt)("a",{parentName:"td",href:"https://pubmed.ncbi.nlm.nih.gov/23703209"},"DNAshape server data"),".")),(0,n.kt)("tr",{parentName:"tbody"},(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/sequence-analysis/dna-shape-fasta"},(0,n.kt)("inlineCode",{parentName:"a"},"dna-shape-fasta"))),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"FASTA")),(0,n.kt)("td",{parentName:"tr",align:null},(0,n.kt)("a",{parentName:"td",href:"/scriptmanager-docs/docs/file-formats"},"CDT")),(0,n.kt)("td",{parentName:"tr",align:null},"Calculate intrinsic DNA shape parameters given input FASTA files. Based on Roh's lab ",(0,n.kt)("a",{parentName:"td",href:"https://pubmed.ncbi.nlm.nih.gov/23703209"},"DNAshape server data"),".")))))}p.isMDXComponent=!0}}]);