"use strict";(self.webpackChunkscriptmanager_docs=self.webpackChunkscriptmanager_docs||[]).push([[3763],{3905:(e,t,a)=>{a.d(t,{Zo:()=>s,kt:()=>c});var n=a(7294);function r(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function l(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function i(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?l(Object(a),!0).forEach((function(t){r(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):l(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function o(e,t){if(null==e)return{};var a,n,r=function(e,t){if(null==e)return{};var a,n,r={},l=Object.keys(e);for(n=0;n<l.length;n++)a=l[n],t.indexOf(a)>=0||(r[a]=e[a]);return r}(e,t);if(Object.getOwnPropertySymbols){var l=Object.getOwnPropertySymbols(e);for(n=0;n<l.length;n++)a=l[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(r[a]=e[a])}return r}var d=n.createContext({}),m=function(e){var t=n.useContext(d),a=t;return e&&(a="function"==typeof e?e(t):i(i({},t),e)),a},s=function(e){var t=m(e.components);return n.createElement(d.Provider,{value:t},e.children)},p={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},f=n.forwardRef((function(e,t){var a=e.components,r=e.mdxType,l=e.originalType,d=e.parentName,s=o(e,["components","mdxType","originalType","parentName"]),f=m(a),c=r,k=f["".concat(d,".").concat(c)]||f[c]||p[c]||l;return a?n.createElement(k,i(i({ref:t},s),{},{components:a})):n.createElement(k,i({ref:t},s))}));function c(e,t){var a=arguments,r=t&&t.mdxType;if("string"==typeof e||r){var l=a.length,i=new Array(l);i[0]=f;var o={};for(var d in t)hasOwnProperty.call(t,d)&&(o[d]=t[d]);o.originalType=e,o.mdxType="string"==typeof e?e:r,i[1]=o;for(var m=2;m<l;m++)i[m]=a[m];return n.createElement.apply(null,i)}return n.createElement.apply(null,a)}f.displayName="MDXCreateElement"},5326:(e,t,a)=>{a.r(t),a.d(t,{assets:()=>d,contentTitle:()=>i,default:()=>p,frontMatter:()=>l,metadata:()=>o,toc:()=>m});var n=a(7462),r=(a(7294),a(3905));const l={id:"tool-group",title:"Tool Groups",sidebar_label:"Tool Groups"},i=void 0,o={unversionedId:"Guides/tool-group",id:"Guides/tool-group",title:"Tool Groups",description:"BAM Format Converter",source:"@site/docs/Guides/tool-group.md",sourceDirName:"Guides",slug:"/Guides/tool-group",permalink:"/scriptmanager-docs/docs/Guides/tool-group",draft:!1,tags:[],version:"current",lastUpdatedBy:"Olivia Wen-Mei Lang",lastUpdatedAt:1674838098,formattedLastUpdatedAt:"Jan 27, 2023",frontMatter:{id:"tool-group",title:"Tool Groups",sidebar_label:"Tool Groups"},sidebar:"someSidebar",previous:{title:"Command Line",permalink:"/scriptmanager-docs/docs/Guides/command-line"},next:{title:"Color Selection Guide",permalink:"/scriptmanager-docs/docs/Guides/color-guide"}},d={},m=[{value:"BAM Format Converter",id:"bam-format-converter",level:2},{value:"BAM Manipulation",id:"bam-manipulation",level:2},{value:"BAM Statistics",id:"bam-statistics",level:2},{value:"Coordinate Manipulation",id:"coordinate-manipulation",level:2},{value:"Figure Generation",id:"figure-generation",level:2},{value:"File Utilities",id:"file-utilities",level:2},{value:"Peak Analysis",id:"peak-analysis",level:2},{value:"Read Analysis",id:"read-analysis",level:2},{value:"Sequence Analysis",id:"sequence-analysis",level:2}],s={toc:m};function p(e){let{components:t,...a}=e;return(0,r.kt)("wrapper",(0,n.Z)({},s,a,{components:t,mdxType:"MDXLayout"}),(0,r.kt)("h2",{id:"bam-format-converter"},"BAM Format Converter"),(0,r.kt)("p",null,(0,r.kt)("em",{parentName:"p"},"Description of BAM-Format-Converter tool")),(0,r.kt)("table",null,(0,r.kt)("thead",{parentName:"table"},(0,r.kt)("tr",{parentName:"thead"},(0,r.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,r.kt)("th",{parentName:"tr",align:null},"Input"),(0,r.kt)("th",{parentName:"tr",align:null},"Output"),(0,r.kt)("th",{parentName:"tr",align:null},"Description"))),(0,r.kt)("tbody",{parentName:"table"},(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/bam-format-converter/bam-to-scidx"},(0,r.kt)("inlineCode",{parentName:"a"},"bam-to-scidx"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#scidx"},"scIDX")),(0,r.kt)("td",{parentName:"tr",align:null},"Convert BAM file to scIDX file.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/bam-format-converter/bam-to-gff"},(0,r.kt)("inlineCode",{parentName:"a"},"bam-to-gff"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#gff"},"GFF")),(0,r.kt)("td",{parentName:"tr",align:null},"Convert BAM file to GFF file.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/bam-format-converter/bam-to-bed"},(0,r.kt)("inlineCode",{parentName:"a"},"bam-to-bed"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED")),(0,r.kt)("td",{parentName:"tr",align:null},"Convert BAM file to BED file.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/bam-format-converter/bam-to-bedgraph"},(0,r.kt)("inlineCode",{parentName:"a"},"bam-to-bedgraph"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bedgraph"},"bedGraph")),(0,r.kt)("td",{parentName:"tr",align:null},"Convert BAM file to bedGraph file.")))),(0,r.kt)("h2",{id:"bam-manipulation"},"BAM Manipulation"),(0,r.kt)("p",null,(0,r.kt)("em",{parentName:"p"},"General description of BAM Manipulation")),(0,r.kt)("table",null,(0,r.kt)("thead",{parentName:"table"},(0,r.kt)("tr",{parentName:"thead"},(0,r.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,r.kt)("th",{parentName:"tr",align:null},"Input"),(0,r.kt)("th",{parentName:"tr",align:null},"Output"),(0,r.kt)("th",{parentName:"tr",align:null},"Description"))),(0,r.kt)("tbody",{parentName:"table"},(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/bam-manipulation/bam-indexer"},(0,r.kt)("inlineCode",{parentName:"a"},"bam-indexer"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAI")),(0,r.kt)("td",{parentName:"tr",align:null},"Generates BAI Index for input BAM files. Output BAI is in the same directory as input BAM file.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/bam-manipulation/sort-bam"},(0,r.kt)("inlineCode",{parentName:"a"},"sort-bam"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},"Sort BAM files in order to efficiently extract and manipulate.\\nRAM intensive process. If program freezes, increase JAVA heap size.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/bam-manipulation/remove-duplicates"},(0,r.kt)("inlineCode",{parentName:"a"},"remove-duplicates"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},"Removes duplicate reads in Paired-End sequencing given identical 5' read locations. RAM intensive process. If program freezes, increase JAVA heap size.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/bam-manipulation/merge-bam"},(0,r.kt)("inlineCode",{parentName:"a"},"merge-bam"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM"),"[","..","]"),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},"Merges Multiple BAM files into single BAM file. Sorting is performed automatically. RAM intensive process. If program freezes, increase JAVA heap size.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/bam-manipulation/filter-pip-seq"},(0,r.kt)("inlineCode",{parentName:"a"},"filter-pip-seq"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},"Filter BAM file by -1 nucleotide. Requires genome FASTA file. Note this program does not index the resulting BAM file and user must use appropriate samtools command to generate BAI.")))),(0,r.kt)("h2",{id:"bam-statistics"},"BAM Statistics"),(0,r.kt)("table",null,(0,r.kt)("thead",{parentName:"table"},(0,r.kt)("tr",{parentName:"thead"},(0,r.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,r.kt)("th",{parentName:"tr",align:null},"Input"),(0,r.kt)("th",{parentName:"tr",align:null},"Output"),(0,r.kt)("th",{parentName:"tr",align:null},"Description"))),(0,r.kt)("tbody",{parentName:"table"},(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/bam-statistics/se-stat"},"se-stat")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#txt"},"TXT")),(0,r.kt)("td",{parentName:"tr",align:null},"Genome-Genome correlations for replicate comparisons given multiple sorted and indexed (BAI) BAM files.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/bam-statistics/pe-stat"},"pe-stat")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#txt"},"TXT")),(0,r.kt)("td",{parentName:"tr",align:null},"Generates Insert-size Histogram statistic (GEO requirement) and outputs BAM Header including alignment statistics and parameters given a sorted and indexed (BAI) paired-end BAM File.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/bam-statistics/bam-correlation"},"bam-correlation"),"bam-correlation)"),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#txt"},"TXT")),(0,r.kt)("td",{parentName:"tr",align:null},"Output BAM Header including alignment statistics and parameters given any indexed (BAI) BAM File.")))),(0,r.kt)("h2",{id:"coordinate-manipulation"},"Coordinate Manipulation"),(0,r.kt)("p",null,(0,r.kt)("em",{parentName:"p"},"Coordinate Manipulation tools description")),(0,r.kt)("table",null,(0,r.kt)("thead",{parentName:"table"},(0,r.kt)("tr",{parentName:"thead"},(0,r.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,r.kt)("th",{parentName:"tr",align:null},"Input"),(0,r.kt)("th",{parentName:"tr",align:null},"Output"),(0,r.kt)("th",{parentName:"tr",align:null},"Description"))),(0,r.kt)("tbody",{parentName:"table"},(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/coordinate-manipulation/expand-bed"},"expand-bed")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED")),(0,r.kt)("td",{parentName:"tr",align:null},"Expands input BED file by adding positions to the border or around the center.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/coordinate-manipulation/expand-gff"},"expand-gff")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#gff"},"GFF")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#gff"},"GFF")),(0,r.kt)("td",{parentName:"tr",align:null},"Expands input GFF file by adding positions to the border or around the center.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/coordinate-manipulation/bed-to-gff"},"bed-to-gff")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#gff"},"GFF")),(0,r.kt)("td",{parentName:"tr",align:null},"Converts BED file format to GFF file format.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/coordinate-manipulation/gff-to-bed"},"gff-to-bed")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#gff"},"GFF")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED")),(0,r.kt)("td",{parentName:"tr",align:null},"Converts GFF file format to BED file format.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/coordinate-manipulation/sort-bed"},"sort-bed")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED")),(0,r.kt)("td",{parentName:"tr",align:null},"Sort a CDT file and its corresponding BED file by the total score in the CDT file across the specified interval.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/coordinate-manipulation/sort-gff"},"sort-gff")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#gff"},"GFF")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#gff"},"GFF")),(0,r.kt)("td",{parentName:"tr",align:null},"Sort a CDT file and its corresponding GFF file by the total score in the CDT file across the specified interval.")))),(0,r.kt)("h2",{id:"figure-generation"},"Figure Generation"),(0,r.kt)("table",null,(0,r.kt)("thead",{parentName:"table"},(0,r.kt)("tr",{parentName:"thead"},(0,r.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,r.kt)("th",{parentName:"tr",align:null},"Input"),(0,r.kt)("th",{parentName:"tr",align:null},"Output"),(0,r.kt)("th",{parentName:"tr",align:null},"Description"))),(0,r.kt)("tbody",{parentName:"table"},(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/figure-generation/heatmap"},"heatmap")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#cdt"},"CDT")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#png"},"PNG")),(0,r.kt)("td",{parentName:"tr",align:null},"Generate heatmap using CDT files.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/figure-generation/merge-heatmap"},"merge-heatmap")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#png"},"PNG")," + ",(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#png"},"PNG")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#png"},"PNG")),(0,r.kt)("td",{parentName:"tr",align:null},"Merge Sense and Antisense png heatmaps.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/figure-generation/four-color"},"four-color")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#fasta"},"FASTA")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#png"},"PNG")),(0,r.kt)("td",{parentName:"tr",align:null},"Generate 4Color sequence plot given FASTA file and user-defined RGB colors.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/figure-generation/composite-plot"},"composite-plot")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#cdt"},"CDT")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#png"},"PNG")),(0,r.kt)("td",{parentName:"tr",align:null},"Generate a Composite Plot PNG from composite data like the output in TagPileup.")))),(0,r.kt)("h2",{id:"file-utilities"},"File Utilities"),(0,r.kt)("p",null,(0,r.kt)("em",{parentName:"p"},"File Utilities tools description")),(0,r.kt)("table",null,(0,r.kt)("thead",{parentName:"table"},(0,r.kt)("tr",{parentName:"thead"},(0,r.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,r.kt)("th",{parentName:"tr",align:null},"Input"),(0,r.kt)("th",{parentName:"tr",align:null},"Output"),(0,r.kt)("th",{parentName:"tr",align:null},"Description"))),(0,r.kt)("tbody",{parentName:"table"},(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/file-utilities/md5checksum"},"md5checksum")),(0,r.kt)("td",{parentName:"tr",align:null},"*"),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#txt"},"TXT")),(0,r.kt)("td",{parentName:"tr",align:null},"Calculate MD5 checksum for an input file.")))),(0,r.kt)("h2",{id:"peak-analysis"},"Peak Analysis"),(0,r.kt)("p",null,(0,r.kt)("em",{parentName:"p"},"Peak Analysis tools description")),(0,r.kt)("table",null,(0,r.kt)("thead",{parentName:"table"},(0,r.kt)("tr",{parentName:"thead"},(0,r.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,r.kt)("th",{parentName:"tr",align:null},"Input"),(0,r.kt)("th",{parentName:"tr",align:null},"Output"),(0,r.kt)("th",{parentName:"tr",align:null},"Description"))),(0,r.kt)("tbody",{parentName:"table"},(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/peak-analysis/peak-align-ref"},(0,r.kt)("inlineCode",{parentName:"a"},"peak-align-ref"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED(ref)")," + ",(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED(peak)")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#cdt"},"CDT")),(0,r.kt)("td",{parentName:"tr",align:null},"Align BED peaks to Reference BED file creating CDT files for heatmap generation.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/peak-analysis/filter-bed"},(0,r.kt)("inlineCode",{parentName:"a"},"filter-bed"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED(cluster)")," + ",(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED(filter)")),(0,r.kt)("td",{parentName:"tr",align:null},"Filter BED file using user-specified exclusion zone using the score column to determine which peak to retain.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/peak-analysis/tile-genome"},(0,r.kt)("inlineCode",{parentName:"a"},"tile-genome"))),(0,r.kt)("td",{parentName:"tr",align:null},"genomeName"),(0,r.kt)("td",{parentName:"tr",align:null},"[",(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED"),"|",(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#gff"},"GFF"),"]"),(0,r.kt)("td",{parentName:"tr",align:null},"Generate a coordinate file that tiles (non-overlapping) across an entire genome.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/peak-analysis/rand-coord"},(0,r.kt)("inlineCode",{parentName:"a"},"rand-coord"))),(0,r.kt)("td",{parentName:"tr",align:null},"genomeName"),(0,r.kt)("td",{parentName:"tr",align:null},"[",(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED"),"|",(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#gff"},"GFF"),"]"),(0,r.kt)("td",{parentName:"tr",align:null},"Generate a coordinate file that tiles (non-overlapping) across an entire genome.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/peak-analysis/signal-dup"},(0,r.kt)("inlineCode",{parentName:"a"},"signal-dup"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#gff"},"GFF")," + ",(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#txt"},"TXT")),(0,r.kt)("td",{parentName:"tr",align:null},"Calculate duplication statistics at user-specified regions.")))),(0,r.kt)("h2",{id:"read-analysis"},"Read Analysis"),(0,r.kt)("p",null,"These Read Analysis tools are mostly tools that work with the ","[CDT or other tab-delimited Matrix file formats][file-format]","."),(0,r.kt)("p",null,"Most commonly used  is the TagPileup tool."),(0,r.kt)("p",null,"Others manipulate files to calculate normalization factors, normalize the data, or other manipulations and transformations of read info."),(0,r.kt)("table",null,(0,r.kt)("thead",{parentName:"table"},(0,r.kt)("tr",{parentName:"thead"},(0,r.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,r.kt)("th",{parentName:"tr",align:null},"Input"),(0,r.kt)("th",{parentName:"tr",align:null},"Output"),(0,r.kt)("th",{parentName:"tr",align:null},"Description"))),(0,r.kt)("tbody",{parentName:"table"},(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/read-analysis/tag-pileup"},(0,r.kt)("inlineCode",{parentName:"a"},"tag-pileup"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED")," + ",(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#cdt"},"CDT")),(0,r.kt)("td",{parentName:"tr",align:null},"Pileup 5' ends of aligned tags given BED and BAM files according to user-defined parameters.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/read-analysis/scaling-factor"},(0,r.kt)("inlineCode",{parentName:"a"},"scaling-factor"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bam"},"BAM")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#txt"},"TXT")),(0,r.kt)("td",{parentName:"tr",align:null},"Calculate the factor as either total tag normalization or normalization of ChIP-seq data with control. (PMID:22883957)")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/read-analysis/scale-matrix"},(0,r.kt)("inlineCode",{parentName:"a"},"scale-matrix"))),(0,r.kt)("td",{parentName:"tr",align:null},"[",(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#cdt"},"CDT"),"|",(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#matrix-format-custom-for-these-tools"},"TAB"),"]"),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#matrix-format-custom-for-these-tools"},"TAB"),"[","..","]"),(0,r.kt)("td",{parentName:"tr",align:null},"Apply a user-specified scaling factor to tab-delimited matrix data.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/read-analysis/aggregate-data"},(0,r.kt)("inlineCode",{parentName:"a"},"aggregate-data"))),(0,r.kt)("td",{parentName:"tr",align:null},"[",(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#cdt"},"CDT"),"|",(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#matrix-format-custom-for-these-tools"},"TAB"),"]","[","..","]"),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#matrix-format-custom-for-these-tools"},"TAB"),"[","..","]"),(0,r.kt)("td",{parentName:"tr",align:null},"The AggregateData tool is used to process a bunch of matrix files into one matrix file.")))),(0,r.kt)("h2",{id:"sequence-analysis"},"Sequence Analysis"),(0,r.kt)("p",null,(0,r.kt)("em",{parentName:"p"},"Sequence Analysis tools description")),(0,r.kt)("table",null,(0,r.kt)("thead",{parentName:"table"},(0,r.kt)("tr",{parentName:"thead"},(0,r.kt)("th",{parentName:"tr",align:null},"Tool Name"),(0,r.kt)("th",{parentName:"tr",align:null},"Input"),(0,r.kt)("th",{parentName:"tr",align:null},"Output"),(0,r.kt)("th",{parentName:"tr",align:null},"Description"))),(0,r.kt)("tbody",{parentName:"table"},(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/sequence-analysis/fasta-extract"},(0,r.kt)("inlineCode",{parentName:"a"},"fasta-extract"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED")," + ",(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#fasta"},"FASTA(genome)")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#fasta"},"FASTA")),(0,r.kt)("td",{parentName:"tr",align:null},"Generate FASTA file from indexed Genome FASTA file and BED file. Script will generate FAI index if not present in Genome FASTA folder.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/sequence-analysis/randomize-fasta"},(0,r.kt)("inlineCode",{parentName:"a"},"randomize-fasta"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#fasta"},"FASTA")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#fasta"},"FASTA")),(0,r.kt)("td",{parentName:"tr",align:null},"Randomizes FASTA sequence for each input entry")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/sequence-analysis/search-motif"},(0,r.kt)("inlineCode",{parentName:"a"},"search-motif"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#fasta"},"FASTA")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED")),(0,r.kt)("td",{parentName:"tr",align:null},"Search for an IUPAC DNA sequence motif in FASTA files with mismatches allowed.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/sequence-analysis/dna-shape-bed"},(0,r.kt)("inlineCode",{parentName:"a"},"dna-shape-bed"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#bed"},"BED")," + ",(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#fasta"},"FASTA(genome)")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#cdt"},"CDT")),(0,r.kt)("td",{parentName:"tr",align:null},"Calculate intrinsic DNA shape parameters given BED file and Genome FASTA file. Based on Roh's lab ",(0,r.kt)("a",{parentName:"td",href:"https://pubmed.ncbi.nlm.nih.gov/23703209"},"DNAshape server data"),".")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/sequence-analysis/dna-shape-fasta"},(0,r.kt)("inlineCode",{parentName:"a"},"dna-shape-fasta"))),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#fasta"},"FASTA")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("a",{parentName:"td",href:"/docs/References/file-formats#cdt"},"CDT")),(0,r.kt)("td",{parentName:"tr",align:null},"Calculate intrinsic DNA shape parameters given input FASTA files. Based on Roh's lab ",(0,r.kt)("a",{parentName:"td",href:"https://pubmed.ncbi.nlm.nih.gov/23703209"},"DNAshape server data"),".")))))}p.isMDXComponent=!0}}]);