"use strict";(self.webpackChunkscriptmanager_docs=self.webpackChunkscriptmanager_docs||[]).push([[7327],{1498:(e,a,t)=>{t.d(a,{i:()=>i});var o=t(6540),r=t(2885);const n=()=>({}),i=e=>{let{columns:a,data:t,getHeaderProps:i=n,getColumnProps:s=n}=e;const{getTableProps:l,getTableBodyProps:d,headerGroups:c,rows:m,prepareRow:p}=(0,r.useTable)({columns:a,data:t},r.useSortBy);return o.createElement("table",l(),o.createElement("thead",null,c.map((e=>o.createElement("tr",e.getHeaderGroupProps(),e.headers.map((e=>o.createElement("th",e.getHeaderProps([{className:e.className},i(e),s(e),e.getSortByToggleProps()]),e.render("Header"),o.createElement("span",null,e.isSorted?e.isSortedDesc?" \u25bc":" \u25b2":"")))))))),o.createElement("tbody",d(),m.map(((e,a)=>(p(e),o.createElement("tr",e.getRowProps(),e.cells.map((e=>o.createElement("td",e.getCellProps([{className:e.column.className,style:e.column.style},s(e.column)]),e.render("Cell"))))))))))}},4216:(e,a,t)=>{t.r(a),t.d(a,{assets:()=>u,contentTitle:()=>p,default:()=>h,frontMatter:()=>m,metadata:()=>g,toc:()=>f});var o=t(8168),r=t(6540),n=t(5680),i=t(1498),s=t(6025);const l={name:"MyTool",group:"MyGroup",jfree:!1,picard:!1,gui_only:!1},d=[{...l,name:"Aggregate Data",group:"RA",description:"Compile data from tab-delimited file into matrix according to user-specified metric.",command:"-",url:"docs/Tools/read-analysis/aggregate-data"},{...l,name:"BAM Correlation",group:"BS",description:"Genome-Genome correlations for replicate comparisons given multiple sorted and indexed (BAI) BAM files.",command:"-",url:"docs/Tools/bam-statistics/bam-correlation",jfree:!0},{...l,name:"BAM Indexer",group:"BM",description:"For most tools using BAM inputs (both within and without ScriptManager), a BAM index file (.bai) file is required so that the tool can efficiently query the file for alignment records.",command:"-",url:"docs/Tools/bam-manipulation/bam-indexer",picard:!0},{...l,name:"BAM to bedGraph",group:"BF",description:"Convert BAM file to bedGraph file",command:"-",url:"docs/Tools/bam-format-converter/bam-to-bedgraph"},{...l,name:"BAM to BED",group:"BF",description:"Convert BAM file to BED file",command:"-",url:"docs/Tools/bam-format-converter/bam-to-bed"},{...l,name:"BAM to GFF",group:"BF",description:"Convert BAM file to GFF file",command:"-",url:"docs/Tools/bam-format-converter/bam-to-gff"},{...l,name:"BAM to scIdx",group:"BF",description:"Convert BAM file to scIDX file",command:"-",url:"docs/Tools/bam-format-converter/bam-to-scidx"},{...l,name:"BED to GFF",group:"CM",description:"Converts BED file to GFF file",command:"-",url:"docs/Tools/coordinate-manipulation/bed-to-gff"},{...l,name:"Composite Line Plot",group:"FG",description:"Generate a Composite Plot PNG from composite data like the output in TagPileup",command:"-",url:"docs/Tools/figure-generation/composite-plot",jfree:!0},{...l,name:"Cross Correlation",group:"BS",description:"Calculate optimal tag shift based on ArchTEx implementation from (PMID:22302569)",command:"-",url:"docs/Tools/bam-statistics/cross-corr",jfree:!0},{...l,name:"DNA shape from BED",group:"SA",description:"Calculate intrinsic DNA shape parameters given BED file and Genome FASTA file.",command:"-",url:"docs/Tools/sequence-analysis/dna-shape-bed",jfree:!0},{...l,name:"DNA shape from FASTA",group:"SA",description:"Calculate intrinsic DNA shape parameters given input FASTA files. Based on Roh's lab DNAshape server data",command:"-",url:"docs/Tools/sequence-analysis/dna-shape-fasta",jfree:!0},{...l,name:"Expand BED",group:"CM",description:"Expands input BED file by adding positions to the border or around the center.",command:"-",url:"docs/Tools/coordinate-manipulation/expand-bed"},{...l,name:"Expand GFF",group:"CM",description:"Expands input GFF file by adding positions to the border or around the center",command:"-",url:"docs/Tools/coordinate-manipulation/expand-gff"},{...l,name:"Extract FASTA",group:"SA",description:"Generate FASTA file from indexed Genome FASTA file and BED file. Script will generate FAI index if not present in Genome FASTA folder.",command:"-",url:"docs/Tools/sequence-analysis/fasta-extract"},{...l,name:"Filter BED by Proximity",group:"PA",description:"Filter BED file using user-specified exclusion zone using the score column to determine which peak to retain.",command:"-",url:"docs/Tools/peak-analysis/filter-bed"},{...l,name:"Filter PIP-seq",group:"BM",description:"Filter BAM file by -1 nucleotide. Requires genome FASTA file. Note this program does not index the resulting BAM file and user must use appropriate samtools command to generate BAI.",command:"-",url:"docs/Tools/bam-manipulation/filter-pip-seq"},{...l,name:"Four Color Plot",group:"FG",description:"Generate 4Color sequence plot given FASTA file and user-defined RGB colors.",command:"-",url:"docs/Tools/figure-generation/four-color",jfree:!0},{...l,name:"GFF to BED",group:"CM",description:"Converts GFF file format to BED file format",command:"-",url:"docs/Tools/coordinate-manipulation/gff-to-bed"},{...l,name:"Two Color Heatmap Plot",group:"FG",description:"This tool generates a heatmap from a tab-delimited matrix input of numeric values.",command:"-",url:"docs/Tools/figure-generation/heatmap",jfree:!0},{...l,name:"MD5 Checksum",group:"FU",description:"A common quality control and security step that everyone should use when downloading files from another source is to compare MD5 checksum values to ensure that the file that was downloaded exactly matches the reported MD5checksum value for the file.",command:"-",url:"docs/Tools/file-utilities/md5checksum"},{...l,name:"Merge BAM Replicates",group:"BM",description:"Merge multiple sorted alignment files, producing a single sorted output file that contains all the input records and maintains the existing sort order.",command:"-",url:"docs/Tools/bam-manipulation/merge-bam",picard:!0},{...l,name:"Merge Heatmaps",group:"FG",description:"This tool merges two PNG files into a third PNG file that is an average of each corresponding pair of pixels from the input files.",command:"-",url:"docs/Tools/figure-generation/merge-heatmap",jfree:!0},{...l,name:"Paired-end Statistics",group:"BS",description:"Generates Insert-size Histogram statistic (GEO requirement) and outputs BAM Header including alignment statistics and parameters given a sorted and indexed (BAI) paired-end BAM File.",command:"-",url:"docs/Tools/bam-statistics/pe-stat",jfree:!0},{...l,name:"Align BED to Reference",group:"PA",description:"Align BED peaks to Reference BED file creating CDT files for heatmap generation",command:"-",url:"docs/Tools/peak-analysis/peak-align-ref"},{...l,name:"Random Coordinates",group:"PA",description:"Generate a coordinate file that tiles (non-overlapping) across an entire genome.",command:"-",url:"docs/Tools/peak-analysis/rand-coord"},{...l,name:"Randomize FASTA",group:"SA",description:"Randomizes FASTA sequence for each input entry",command:"-",url:"docs/Tools/sequence-analysis/randomize-fasta"},{...l,name:"Mark (Remove) Duplicates",group:"BM",description:"This tool locates and tags duplicate reads in a BAM or SAM file, where duplicate reads are defined as originating from a single fragment of DNA.",command:"Picard",url:"docs/Tools/bam-manipulation/mark-duplicates",picard:!0},{...l,name:"Scale Matrix",group:"RA",description:"Apply a user-specified scaling factor to tab-delimited matrix data",command:"-",url:"docs/Tools/read-analysis/scale-matrix"},{...l,name:"Calculate Scaling Factor",group:"RA",description:"Calculate the factor as either total tag normalization or normalization of ChIP-seq data with control (PMID:22883957)",command:"-",url:"docs/Tools/read-analysis/scaling-factor",jfree:!0},{...l,name:"Single-end Statistics",group:"BS",description:"Output BAM Header including alignment statistics and parameters given any indexed (BAI) BAM File.",command:"-",url:"docs/Tools/bam-statistics/se-stat"},{...l,name:"Search Motif",group:"SA",description:"Search for an IUPAC DNA sequence motif in FASTA files with mismatches allowed",command:"-",url:"docs/Tools/sequence-analysis/search-motif"},{...l,name:"Signal Duplication",group:"PA",description:"Calculate duplication statistics at user-specified regions.",command:"-",url:"docs/Tools/peak-analysis/signal-dup"},{...l,name:"Sort BAM",group:"BM",description:"Sort alignments by leftmost coordinates, or by read name when -n is used. An appropriate @HD-SO sort order header tag will be added or an existing one updated if necessary.",command:"-",url:"docs/Tools/bam-manipulation/sort-bam",picard:!0},{...l,name:"Sort BED",group:"CM",description:"Sort a CDT file and its corresponding BED file by the total score in the CDT file across the specified interval",command:"-",url:"docs/Tools/coordinate-manipulation/sort-bed"},{...l,name:"Sort BED By Distance",group:"PA",description:"Sort BED/GFF by distance to another BED/GFF (e.g. peak annotations)",command:"-",url:"docs/Tools/peak-analysis/sort-by-dis"},{...l,name:"Sort GFF",group:"CM",description:"Sort a CDT file and its corresponding GFF file by the total score in the CDT file across the specified interval",command:"-",url:"docs/Tools/coordinate-manipulation/sort-gff"},{...l,name:"Tag Pileup",group:"RA",description:"Pileup 5' ends of aligned tags given BED and BAM files according to user-defined parameters",command:"-",url:"docs/Tools/read-analysis/tag-pileup",jfree:!0},{...l,name:"Tile Genome",group:"PA",description:"Generate a coordinate file that tiles (non-overlapping) across an entire genome.",command:"-",url:"docs/Tools/peak-analysis/tile-genome"},{...l,name:"Transpose Matrix",group:"RA",description:"Interchange the rows and columns of a matrix.",command:"-",url:"docs/Tools/read-analysis/transpose-matrix"}];const c=[{Header:"Group",accessor:"group",className:"pester-data-table left",Cell:e=>{let{cell:{value:a},row:{original:t}}=e;return r.createElement("b",null," ",a," ")}},{Header:"Tool Name",accessor:"name",className:"pester-data-table left",Cell:e=>{let{cell:{value:a},row:{original:t}}=e;return r.createElement("a",{href:(0,s.A)(`${t.url}`),target:"blank",rel:"noreferrer noopener"},a)}},{Header:"Description",accessor:"description",className:"pester-data-table"}],m={id:"tool-index",title:"Tool Index (A-Z)",sidebar_label:"Tool Index (A-Z)"},p=void 0,g={unversionedId:"Guides/tool-index",id:"Guides/tool-index",title:"Tool Index (A-Z)",description:"Tool Index",source:"@site/docs/Guides/tool-index.md",sourceDirName:"Guides",slug:"/Guides/tool-index",permalink:"/scriptmanager-docs/docs/Guides/tool-index",draft:!1,tags:[],version:"current",lastUpdatedBy:"Olivia Wen-Mei Lang",lastUpdatedAt:1725643186,formattedLastUpdatedAt:"Sep 6, 2024",frontMatter:{id:"tool-index",title:"Tool Index (A-Z)",sidebar_label:"Tool Index (A-Z)"},sidebar:"generalSidebar",previous:{title:"Logging Manager Guide",permalink:"/scriptmanager-docs/docs/Guides/Getting-Started/logging-manager"},next:{title:"ChIP-exo",permalink:"/scriptmanager-docs/docs/Guides/Tutorials/chipexo-tutorial"}},u={},f=[{value:"Tool Index",id:"tool-index",level:3}],y={toc:f},A="wrapper";function h(e){let{components:a,...t}=e;return(0,n.yg)(A,(0,o.A)({},y,t,{components:a,mdxType:"MDXLayout"}),(0,n.yg)("h3",{id:"tool-index"},"Tool Index"),(0,n.yg)("p",null,"The full list of ScriptManager tools. It's sortable! Click the headers!"),(0,n.yg)("details",null," ",(0,n.yg)("summary",null," Click to see group codes explained "),(0,n.yg)("table",null,(0,n.yg)("thead",{parentName:"table"},(0,n.yg)("tr",{parentName:"thead"},(0,n.yg)("th",{parentName:"tr",align:"center"},"Code"),(0,n.yg)("th",{parentName:"tr",align:"center"},"Tool Group"))),(0,n.yg)("tbody",{parentName:"table"},(0,n.yg)("tr",{parentName:"tbody"},(0,n.yg)("td",{parentName:"tr",align:"center"},(0,n.yg)("strong",{parentName:"td"},"BF")),(0,n.yg)("td",{parentName:"tr",align:"center"},"BAM Format Converter")),(0,n.yg)("tr",{parentName:"tbody"},(0,n.yg)("td",{parentName:"tr",align:"center"},(0,n.yg)("strong",{parentName:"td"},"BM")),(0,n.yg)("td",{parentName:"tr",align:"center"},"BAM Manipulation")),(0,n.yg)("tr",{parentName:"tbody"},(0,n.yg)("td",{parentName:"tr",align:"center"},(0,n.yg)("strong",{parentName:"td"},"BS")),(0,n.yg)("td",{parentName:"tr",align:"center"},"BAM Statistics")),(0,n.yg)("tr",{parentName:"tbody"},(0,n.yg)("td",{parentName:"tr",align:"center"},(0,n.yg)("strong",{parentName:"td"},"CM")),(0,n.yg)("td",{parentName:"tr",align:"center"},"Coordinate Manipulation")),(0,n.yg)("tr",{parentName:"tbody"},(0,n.yg)("td",{parentName:"tr",align:"center"},(0,n.yg)("strong",{parentName:"td"},"FG")),(0,n.yg)("td",{parentName:"tr",align:"center"},"Figure Generation")),(0,n.yg)("tr",{parentName:"tbody"},(0,n.yg)("td",{parentName:"tr",align:"center"},(0,n.yg)("strong",{parentName:"td"},"FU")),(0,n.yg)("td",{parentName:"tr",align:"center"},"File Utilities")),(0,n.yg)("tr",{parentName:"tbody"},(0,n.yg)("td",{parentName:"tr",align:"center"},(0,n.yg)("strong",{parentName:"td"},"PA")),(0,n.yg)("td",{parentName:"tr",align:"center"},"Peak Analysis")),(0,n.yg)("tr",{parentName:"tbody"},(0,n.yg)("td",{parentName:"tr",align:"center"},(0,n.yg)("strong",{parentName:"td"},"RA")),(0,n.yg)("td",{parentName:"tr",align:"center"},"Read Analysis")),(0,n.yg)("tr",{parentName:"tbody"},(0,n.yg)("td",{parentName:"tr",align:"center"},(0,n.yg)("strong",{parentName:"td"},"SA")),(0,n.yg)("td",{parentName:"tr",align:"center"},"Sequence Analysis"))))),(0,n.yg)(i.i,{columns:c,data:d,mdxType:"PesterDataTable"}))}h.isMDXComponent=!0}}]);