"use strict";(self.webpackChunkscriptmanager_docs=self.webpackChunkscriptmanager_docs||[]).push([[3537],{5680:(e,t,a)=>{a.d(t,{xA:()=>c,yg:()=>m});var i=a(6540);function n(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function o(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);t&&(i=i.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,i)}return a}function s(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?o(Object(a),!0).forEach((function(t){n(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):o(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function r(e,t){if(null==e)return{};var a,i,n=function(e,t){if(null==e)return{};var a,i,n={},o=Object.keys(e);for(i=0;i<o.length;i++)a=o[i],t.indexOf(a)>=0||(n[a]=e[a]);return n}(e,t);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);for(i=0;i<o.length;i++)a=o[i],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(n[a]=e[a])}return n}var l=i.createContext({}),d=function(e){var t=i.useContext(l),a=t;return e&&(a="function"==typeof e?e(t):s(s({},t),e)),a},c=function(e){var t=d(e.components);return i.createElement(l.Provider,{value:t},e.children)},u="mdxType",p={inlineCode:"code",wrapper:function(e){var t=e.children;return i.createElement(i.Fragment,{},t)}},g=i.forwardRef((function(e,t){var a=e.components,n=e.mdxType,o=e.originalType,l=e.parentName,c=r(e,["components","mdxType","originalType","parentName"]),u=d(a),g=n,m=u["".concat(l,".").concat(g)]||u[g]||p[g]||o;return a?i.createElement(m,s(s({ref:t},c),{},{components:a})):i.createElement(m,s({ref:t},c))}));function m(e,t){var a=arguments,n=t&&t.mdxType;if("string"==typeof e||n){var o=a.length,s=new Array(o);s[0]=g;var r={};for(var l in t)hasOwnProperty.call(t,l)&&(r[l]=t[l]);r.originalType=e,r[u]="string"==typeof e?e:n,s[1]=r;for(var d=2;d<o;d++)s[d]=a[d];return i.createElement.apply(null,s)}return i.createElement.apply(null,a)}g.displayName="MDXCreateElement"},9365:(e,t,a)=>{a.d(t,{A:()=>s});var i=a(6540),n=a(53);const o={tabItem:"tabItem_Ymn6"};function s(e){let{children:t,hidden:a,className:s}=e;return i.createElement("div",{role:"tabpanel",className:(0,n.A)(o.tabItem,s),hidden:a},t)}},1470:(e,t,a)=>{a.d(t,{A:()=>w});var i=a(8168),n=a(6540),o=a(53),s=a(3104),r=a(6347),l=a(7485),d=a(1682),c=a(9466);function u(e){return function(e){return n.Children.map(e,(e=>{if((0,n.isValidElement)(e)&&"value"in e.props)return e;throw new Error(`Docusaurus error: Bad <Tabs> child <${"string"==typeof e.type?e.type:e.type.name}>: all children of the <Tabs> component should be <TabItem>, and every <TabItem> should have a unique "value" prop.`)}))}(e).map((e=>{let{props:{value:t,label:a,attributes:i,default:n}}=e;return{value:t,label:a,attributes:i,default:n}}))}function p(e){const{values:t,children:a}=e;return(0,n.useMemo)((()=>{const e=t??u(a);return function(e){const t=(0,d.X)(e,((e,t)=>e.value===t.value));if(t.length>0)throw new Error(`Docusaurus error: Duplicate values "${t.map((e=>e.value)).join(", ")}" found in <Tabs>. Every value needs to be unique.`)}(e),e}),[t,a])}function g(e){let{value:t,tabValues:a}=e;return a.some((e=>e.value===t))}function m(e){let{queryString:t=!1,groupId:a}=e;const i=(0,r.W6)(),o=function(e){let{queryString:t=!1,groupId:a}=e;if("string"==typeof t)return t;if(!1===t)return null;if(!0===t&&!a)throw new Error('Docusaurus error: The <Tabs> component groupId prop is required if queryString=true, because this value is used as the search param name. You can also provide an explicit value such as queryString="my-search-param".');return a??null}({queryString:t,groupId:a});return[(0,l.aZ)(o),(0,n.useCallback)((e=>{if(!o)return;const t=new URLSearchParams(i.location.search);t.set(o,e),i.replace({...i.location,search:t.toString()})}),[o,i])]}function h(e){const{defaultValue:t,queryString:a=!1,groupId:i}=e,o=p(e),[s,r]=(0,n.useState)((()=>function(e){let{defaultValue:t,tabValues:a}=e;if(0===a.length)throw new Error("Docusaurus error: the <Tabs> component requires at least one <TabItem> children component");if(t){if(!g({value:t,tabValues:a}))throw new Error(`Docusaurus error: The <Tabs> has a defaultValue "${t}" but none of its children has the corresponding value. Available values are: ${a.map((e=>e.value)).join(", ")}. If you intend to show no default tab, use defaultValue={null} instead.`);return t}const i=a.find((e=>e.default))??a[0];if(!i)throw new Error("Unexpected error: 0 tabValues");return i.value}({defaultValue:t,tabValues:o}))),[l,d]=m({queryString:a,groupId:i}),[u,h]=function(e){let{groupId:t}=e;const a=function(e){return e?`docusaurus.tab.${e}`:null}(t),[i,o]=(0,c.Dv)(a);return[i,(0,n.useCallback)((e=>{a&&o.set(e)}),[a,o])]}({groupId:i}),f=(()=>{const e=l??u;return g({value:e,tabValues:o})?e:null})();(0,n.useLayoutEffect)((()=>{f&&r(f)}),[f]);return{selectedValue:s,selectValue:(0,n.useCallback)((e=>{if(!g({value:e,tabValues:o}))throw new Error(`Can't select invalid tab value=${e}`);r(e),d(e),h(e)}),[d,h,o]),tabValues:o}}var f=a(2303);const y={tabList:"tabList__CuJ",tabItem:"tabItem_LNqP"};function b(e){let{className:t,block:a,selectedValue:r,selectValue:l,tabValues:d}=e;const c=[],{blockElementScrollPositionUntilNextRender:u}=(0,s.a_)(),p=e=>{const t=e.currentTarget,a=c.indexOf(t),i=d[a].value;i!==r&&(u(t),l(i))},g=e=>{let t=null;switch(e.key){case"Enter":p(e);break;case"ArrowRight":{const a=c.indexOf(e.currentTarget)+1;t=c[a]??c[0];break}case"ArrowLeft":{const a=c.indexOf(e.currentTarget)-1;t=c[a]??c[c.length-1];break}}t?.focus()};return n.createElement("ul",{role:"tablist","aria-orientation":"horizontal",className:(0,o.A)("tabs",{"tabs--block":a},t)},d.map((e=>{let{value:t,label:a,attributes:s}=e;return n.createElement("li",(0,i.A)({role:"tab",tabIndex:r===t?0:-1,"aria-selected":r===t,key:t,ref:e=>c.push(e),onKeyDown:g,onClick:p},s,{className:(0,o.A)("tabs__item",y.tabItem,s?.className,{"tabs__item--active":r===t})}),a??t)})))}function v(e){let{lazy:t,children:a,selectedValue:i}=e;if(a=Array.isArray(a)?a:[a],t){const e=a.find((e=>e.props.value===i));return e?(0,n.cloneElement)(e,{className:"margin-top--md"}):null}return n.createElement("div",{className:"margin-top--md"},a.map(((e,t)=>(0,n.cloneElement)(e,{key:t,hidden:e.props.value!==i}))))}function A(e){const t=h(e);return n.createElement("div",{className:(0,o.A)("tabs-container",y.tabList)},n.createElement(b,(0,i.A)({},e,t)),n.createElement(v,(0,i.A)({},e,t)))}function w(e){const t=(0,f.A)();return n.createElement(A,(0,i.A)({key:String(t)},e))}},918:(e,t,a)=>{a.r(t),a.d(t,{assets:()=>u,contentTitle:()=>d,default:()=>h,frontMatter:()=>l,metadata:()=>c,toc:()=>p});var i=a(8168),n=(a(6540),a(5680)),o=a(1470),s=a(9365),r=a(5489);const l={id:"atac-seq-tutorial",title:"ATAC-seq Tutorial",sidebar_label:"ATAC-seq"},d=void 0,c={unversionedId:"Guides/Tutorials/atac-seq-tutorial",id:"Guides/Tutorials/atac-seq-tutorial",title:"ATAC-seq Tutorial",description:"Generating two basic sequence-specific ATAC-Seq plots: composite and heatmap",source:"@site/docs/Guides/Tutorials/atac-seq-tutorial.md",sourceDirName:"Guides/Tutorials",slug:"/Guides/Tutorials/atac-seq-tutorial",permalink:"/scriptmanager-docs/docs/Guides/Tutorials/atac-seq-tutorial",draft:!1,tags:[],version:"current",lastUpdatedBy:"Olivia Wen-Mei Lang",lastUpdatedAt:1724168054,formattedLastUpdatedAt:"Aug 20, 2024",frontMatter:{id:"atac-seq-tutorial",title:"ATAC-seq Tutorial",sidebar_label:"ATAC-seq"},sidebar:"generalSidebar",previous:{title:"Genome Tracks (BigWig)",permalink:"/scriptmanager-docs/docs/Guides/Tutorials/genome-tracks-tutorial"},next:{title:"General Developer Notes",permalink:"/scriptmanager-docs/docs/Guides/Contributing/developer-guidelines"}},u={},p=[{value:"Download ScriptManager (v0.14):",id:"download-scriptmanager-v014",level:2},{value:"Download Data",id:"download-data",level:2},{value:"BED File",id:"bed-file",level:3},{value:"BAM File",id:"bam-file",level:3},{value:"Generate the Plots",id:"generate-the-plots",level:2},{value:"1. Open ScriptManager",id:"1-open-scriptmanager",level:3},{value:"2. Generate BAI index file",id:"2-generate-bai-index-file",level:3},{value:"2.1. Navigate to BAM Manipulation \u27a1\ufe0f BAM-BAI Indexer",id:"21-navigate-to-bam-manipulation-\ufe0f-bam-bai-indexer",level:4},{value:"2.2. Generate BAI index files for each BAM file of interest by loading your BAM file and clicking &quot;Index.&quot; (228,000,000 tags processed)",id:"22-generate-bai-index-files-for-each-bam-file-of-interest-by-loading-your-bam-file-and-clicking-index-228000000-tags-processed",level:4},{value:"3. BAM File Statistics",id:"3-bam-file-statistics",level:3},{value:"3.1. Navigate to BAM Statistics \u27a1\ufe0f Paired-End Statistics",id:"31-navigate-to-bam-statistics-\ufe0f-paired-end-statistics",level:4},{value:"3.2. Select Output Statistics",id:"32-select-output-statistics",level:4},{value:"4. Resize the GRCh38 BED file",id:"4-resize-the-grch38-bed-file",level:3},{value:"4.1. Navigate to Coordinate File Manipulation \u27a1\ufe0f Expand BED File",id:"41-navigate-to-coordinate-file-manipulation-\ufe0f-expand-bed-file",level:4},{value:"4.2. For this tutorial, use the 5000bp expansion and select &quot;Expand from Center&quot;.",id:"42-for-this-tutorial-use-the-5000bp-expansion-and-select-expand-from-center",level:4},{value:"5. Generate the tag pileup",id:"5-generate-the-tag-pileup",level:3},{value:"5.1. Navigate to Sequence Read Analysis \u27a1\ufe0f Tag Pileup",id:"51-navigate-to-sequence-read-analysis-\ufe0f-tag-pileup",level:4},{value:"5.2. Load the BED and BAM files",id:"52-load-the-bed-and-bam-files",level:4},{value:"5.3. Use &quot;Full Fragment&quot; in the top right, then select &quot;Combined&quot; color , &quot;Output Composite&quot;, &quot;CDT&quot;, and &quot;Output GZIP&quot;. Then, proceed to &quot;Output Directory&quot;.",id:"53-use-full-fragment-in-the-top-right-then-select-combined-color--output-composite-cdt-and-output-gzip-then-proceed-to-output-directory",level:4},{value:"5.4. When ready, select \u2018Pile Tags\u2019 to execute",id:"54-when-ready-select-pile-tags-to-execute",level:4},{value:"5.5. Save composite results as PNG",id:"55-save-composite-results-as-png",level:4},{value:"6. Sort BED file by CDT file",id:"6-sort-bed-file-by-cdt-file",level:3},{value:"6.1. Navigate to Coordinate File Manipulation \u27a1\ufe0f Sort BED.",id:"61-navigate-to-coordinate-file-manipulation-\ufe0f-sort-bed",level:4},{value:"6.2. Load BED &amp; CDT File from Tag Pileup",id:"62-load-bed--cdt-file-from-tag-pileup",level:4},{value:"6.3. Change Expansion Size to 1000 from Center",id:"63-change-expansion-size-to-1000-from-center",level:4},{value:"6.4. Select Base Output File Name for sorted BED &amp; CDT Files",id:"64-select-base-output-file-name-for-sorted-bed--cdt-files",level:4},{value:"7. Generate Heatmaps",id:"7-generate-heatmaps",level:3},{value:"7.1. Navigate to Figure Generation \u27a1\ufe0f Heat Map.",id:"71-navigate-to-figure-generation-\ufe0f-heat-map",level:4},{value:"7.2. Load the sorted CDT file",id:"72-load-the-sorted-cdt-file",level:4},{value:"7.3. Select &quot;Custom&quot; for color, and change the Contrast Threshold to &quot;Percentile Value&quot;.The heatmaps for this dataset show the best contrast when using the default &quot;Percentile Threshold&quot; value (.95 or 95%).",id:"73-select-custom-for-color-and-change-the-contrast-threshold-to-percentile-valuethe-heatmaps-for-this-dataset-show-the-best-contrast-when-using-the-default-percentile-threshold-value-95-or-95",level:4},{value:"7.4. Click the &quot;Output Heatmap&quot; checkbox. The Heatmap generator does not save the produced PNG by default.",id:"74-click-the-output-heatmap-checkbox-the-heatmap-generator-does-not-save-the-produced-png-by-default",level:4},{value:"7.5. Click &quot;Generate&quot; to save your PNG heatmap!",id:"75-click-generate-to-save-your-png-heatmap",level:4},{value:"Command-Line shell script",id:"command-line-shell-script",level:2}],g={toc:p},m="wrapper";function h(e){let{components:t,...l}=e;return(0,n.yg)(m,(0,i.A)({},g,l,{components:t,mdxType:"MDXLayout"}),(0,n.yg)("p",null,(0,n.yg)("em",{parentName:"p"},"Generating two basic sequence-specific ATAC-Seq plots: composite and heatmap")),(0,n.yg)("p",null,(0,n.yg)("strong",{parentName:"p"},"Goal:")," This tutorial provides a guide to generating 2 basic plots (Composite plot and Heatmap) using the ScriptManager platform and data generated by the ",(0,n.yg)("a",{parentName:"p",href:"https://www.encodeproject.org/experiments/ENCSR868FGK/"},"Stanford Encode Project"),"."),(0,n.yg)("div",{class:"tutorial-img-flow-container",style:{textAlign:"center"}},(0,n.yg)("img",{src:a(1440).A,style:{width:"20%"}}),(0,n.yg)("img",{src:a(3069).A,style:{width:"70%"}})),(0,n.yg)("h2",{id:"download-scriptmanager-v014"},"Download ScriptManager (v0.14):"),(0,n.yg)("p",null,"The current version of ScriptManager is available for download ",(0,n.yg)("strong",{parentName:"p"},(0,n.yg)("a",{parentName:"strong",href:"https://github.com/CEGRcode/scriptmanager/releases/download/v0.14/ScriptManager-v0.14.jar"},"here")),". Make sure you have ",(0,n.yg)("a",{parentName:"p",href:"/docs/#java"},"Java installed"),"."),(0,n.yg)("p",null,"The file ",(0,n.yg)("inlineCode",{parentName:"p"},"ScriptManager-v0.14.jar")," should be placed someplace locally accessible. For example on Mac OS on the Desktop (Permissions will need to be accepted) or someplace in your home directory (",(0,n.yg)("inlineCode",{parentName:"p"},"Macintosh_HD:Users/userID/ScriptManager"),")"),(0,n.yg)("h2",{id:"download-data"},"Download Data"),(0,n.yg)("p",null,"You need one set of genomic coordinate regions to investigate (BED) and one file of sequencing data alignments (BAM) to complete this exercise. ",(0,n.yg)("a",{parentName:"p",href:"/docs/Guides/Getting-Started/file-formats"},"Read more about the BED/BAM file formats here.")),(0,n.yg)("h3",{id:"bed-file"},"BED File"),(0,n.yg)("p",null,"This is the set of transcription start site (TSS) annotations for the GRCh38 genome build that has been subsampled down to 2000 sites for quick tutorial purposes. A more formal analysis may use a more complete set of annotations (many more sites)."),(0,n.yg)(r.A,{className:"button button--secondary",target:"\\_blank",href:"/sample_data/atac-seq-tutorial/UCSC_GRCh38_knownGene_GENCODEV3_TSS_2000bp_SUBSAMPLE-2000-Sites.bed",mdxType:"Link"},"Download sample BED file"),(0,n.yg)("br",null),(0,n.yg)("br",null),(0,n.yg)("admonition",{type:"caution"},(0,n.yg)("p",{parentName:"admonition"},"If your BED file downloads with a ",(0,n.yg)("inlineCode",{parentName:"p"},".txt")," extension, make sure to change the filename to a ",(0,n.yg)("inlineCode",{parentName:"p"},".bed")," extension. For this tutorial, the BED file is named ",(0,n.yg)("inlineCode",{parentName:"p"},"UCSC_GRCh38_knownGene_GENCODEV3_TSS_2000bp_SUBSAMPLE-2000-Sites.bed"),".")),(0,n.yg)("h3",{id:"bam-file"},"BAM File"),(0,n.yg)("p",null,"This is the set of read alignments from the ",(0,n.yg)("a",{parentName:"p",href:"https://www.encodeproject.org/experiments/ENCSR868FGK/"},"ENCODE Project")," (",(0,n.yg)("inlineCode",{parentName:"p"},"ENCFF128WZG.bam"),"). "),(0,n.yg)(r.A,{className:"button button--secondary",href:"https://www.encodeproject.org/files/ENCFF128WZG/@@download/ENCFF128WZG.bam",mdxType:"Link"},"Download sample BAM file"),(0,n.yg)("br",null),(0,n.yg)("br",null),(0,n.yg)("admonition",{type:"caution"},(0,n.yg)("p",{parentName:"admonition"},"The BAM file is ~6.5GB large so make sure you have enough space on your machine before downloading.")),(0,n.yg)("h2",{id:"generate-the-plots"},"Generate the Plots"),(0,n.yg)("h3",{id:"1-open-scriptmanager"},"1. Open ScriptManager"),(0,n.yg)(o.A,{mdxType:"Tabs"},(0,n.yg)(s.A,{value:"macos",label:"MacOS",default:!0,mdxType:"TabItem"},(0,n.yg)("p",null,"Depending on your system permissions, you may need to be an administrator to open this for the first time. On Mac systems, this can be done by right-clicking the file and selecting \u2018Open\u2019 at the top."),(0,n.yg)("div",{class:"tutorial-img-flow-container"},(0,n.yg)("img",{src:a(6176).A,style:{width:"40%",border:"solid 1px"}})),(0,n.yg)("admonition",{type:"caution"},(0,n.yg)("p",{parentName:"admonition"},"Some MacOS systems may not properly open the JAR by simply double-clicking on the JAR file so you may need to open your Terminal window and execute it from the command line by executing the jar file ",(0,n.yg)("em",{parentName:"p"},"without any arguments or flags"),":"),(0,n.yg)("pre",{parentName:"admonition"},(0,n.yg)("code",{parentName:"pre"},"java -jar /path/to/ScriptManager.jar\n")),(0,n.yg)("p",{parentName:"admonition"},"If you're not sure how to type the path to ScriptManager, you can type ",(0,n.yg)("inlineCode",{parentName:"p"},"java -jar ")," (end with space) and then drag ScriptManager from Finder into your Terminal window and then press enter."))),(0,n.yg)(s.A,{value:"linux",label:"Linux",mdxType:"TabItem"},(0,n.yg)("p",null,"Double-click or right-click the ScriptManager JAR file to start the program."),(0,n.yg)("div",{class:"tutorial-img-flow-container"},(0,n.yg)("img",{src:a(2692).A,style:{width:"40%",border:"solid 1px"}}))),(0,n.yg)(s.A,{value:"windows",label:"Windows",mdxType:"TabItem"},(0,n.yg)("p",null,"Double-click or right-click the ScriptManager JAR file to start the program."),(0,n.yg)("div",{class:"tutorial-img-flow-container"},(0,n.yg)("img",{src:a(3630).A,style:{width:"40%",border:"solid 1px"}})))),(0,n.yg)("p",null,"Once you see the main tool selection window, you're off to the races!"),(0,n.yg)("h3",{id:"2-generate-bai-index-file"},"2. Generate BAI index file"),(0,n.yg)("p",null,"A BAI index file is required for each BAM file of interest (i.e., the tag occupancy data you want to plot). This file allows for rapid access of the sorted and aligned sequence reads (BAM file)."),(0,n.yg)("admonition",{type:"tip"},(0,n.yg)("p",{parentName:"admonition"},"SAM/BAM standard is to keep BAI file in same directory as BAM file with the ScriptManager-generated filename."),(0,n.yg)("pre",{parentName:"admonition"},(0,n.yg)("code",{parentName:"pre",className:"language-bash"},"ENCFF534DCE.bam\nENCFF534DCE.bam.bai  # Need to generate this file to proceed.\n"))),(0,n.yg)("h4",{id:"21-navigate-to-bam-manipulation-\ufe0f-bam-bai-indexer"},"2.1. Navigate to BAM Manipulation \u27a1\ufe0f ",(0,n.yg)("a",{parentName:"h4",href:"/docs/Tools/bam-manipulation/bam-indexer"},(0,n.yg)("strong",{parentName:"a"},"BAM-BAI Indexer"))),(0,n.yg)("div",{class:"tutorial-img-flow-container"},(0,n.yg)("img",{src:a(1483).A,style:{width:"60%"}})),(0,n.yg)("h4",{id:"22-generate-bai-index-files-for-each-bam-file-of-interest-by-loading-your-bam-file-and-clicking-index-228000000-tags-processed"},'2.2. Generate BAI index files for each BAM file of interest by loading your BAM file and clicking "Index." (228,000,000 tags processed)'),(0,n.yg)("div",{class:"tutorial-img-flow-container"},(0,n.yg)("img",{src:a(9693).A,style:{width:"60%"}}),(0,n.yg)("i",{class:"fa-solid fa-arrow-right fa-lg"}),(0,n.yg)("img",{src:a(9270).A,style:{width:"30%"}})),(0,n.yg)("admonition",{type:"note"},(0,n.yg)("p",{parentName:"admonition"},"The speed of this step scales with the size of the BAM file. Generally this step 30 sec for a 100 MB BAM file but may take 1-2 min for a multi-GB BAM file.")),(0,n.yg)("h3",{id:"3-bam-file-statistics"},"3. BAM File Statistics"),(0,n.yg)("p",null,"The BAM File Statistics is a Quality Control feature that includes alignment statistics that can be used to check what the data looks like. Paired-end Statistics is a common quality control step used in many ATAC-seq analyses.  "),(0,n.yg)("h4",{id:"31-navigate-to-bam-statistics-\ufe0f-paired-end-statistics"},"3.1. Navigate to BAM Statistics \u27a1\ufe0f ",(0,n.yg)("a",{parentName:"h4",href:"/docs/Tools/bam-statistics/pe-stat"},(0,n.yg)("strong",{parentName:"a"},"Paired-End Statistics"))),(0,n.yg)("div",{class:"tutorial-img_atacseq-flow-container",style:{textAlign:"center"}},(0,n.yg)("img",{src:a(8333).A,style:{width:"70%"}})),(0,n.yg)("h4",{id:"32-select-output-statistics"},"3.2. Select Output Statistics"),(0,n.yg)("div",{className:"tutorial-img_atacseq-flow-container",style:{textAlign:"center"}},(0,n.yg)("img",{src:a(3286).A,style:{width:"60%"}}),(0,n.yg)("img",{src:a(7423).A,style:{width:"80%"}})),(0,n.yg)("h3",{id:"4-resize-the-grch38-bed-file"},"4. Resize the GRCh38 BED file"),(0,n.yg)("p",null,"The BED file is the set of reference coordinates that your heatmap and composite plots will be aligned to, but you\u2019ll need to specify how far upstream and downstream you want your data to be plotted; i.e., \u201cSize of Expansion (bp). If you bed file is defined by more than a 1 bp interval AND you want to add to limits of that interval, then select  \u201cAdd to Border\u201d)."),(0,n.yg)("h4",{id:"41-navigate-to-coordinate-file-manipulation-\ufe0f-expand-bed-file"},"4.1. Navigate to Coordinate File Manipulation \u27a1\ufe0f ",(0,n.yg)("a",{parentName:"h4",href:"/docs/Tools/coordinate-manipulation/expand-bed"},(0,n.yg)("strong",{parentName:"a"},"Expand BED File"))),(0,n.yg)("div",{class:"tutorial-img-flow-container"},(0,n.yg)("img",{src:a(5721).A,style:{width:"60%"}})),(0,n.yg)("h4",{id:"42-for-this-tutorial-use-the-5000bp-expansion-and-select-expand-from-center"},'4.2. For this tutorial, use the 5000bp expansion and select "Expand from Center".'),(0,n.yg)("div",{class:"tutorial-img-flow-container"},(0,n.yg)("img",{src:a(7979).A,style:{width:"60%"}}),(0,n.yg)("i",{class:"fa-solid fa-arrow-right fa-lg"}),(0,n.yg)("img",{src:a(3254).A,style:{width:"30%"}})),(0,n.yg)("h3",{id:"5-generate-the-tag-pileup"},"5. Generate the tag pileup"),(0,n.yg)("p",null,"Use TagPileup to pileup the BAM data within a set of BED coordinate windows to generate the composite plot and the matrix(CDT) files that will be used to generate the heatmaps."),(0,n.yg)("h4",{id:"51-navigate-to-sequence-read-analysis-\ufe0f-tag-pileup"},"5.1. Navigate to Sequence Read Analysis \u27a1\ufe0f ",(0,n.yg)("a",{parentName:"h4",href:"/docs/Tools/read-analysis/tag-pileup"},(0,n.yg)("strong",{parentName:"a"},"Tag Pileup"))),(0,n.yg)("div",{class:"tutorial-img-flow-container"},(0,n.yg)("img",{src:a(3731).A,style:{width:"60%"}})),(0,n.yg)("h4",{id:"52-load-the-bed-and-bam-files"},"5.2. Load the BED and BAM files"),(0,n.yg)("h4",{id:"53-use-full-fragment-in-the-top-right-then-select-combined-color--output-composite-cdt-and-output-gzip-then-proceed-to-output-directory"},'5.3. Use "Full Fragment" in the top right, then select "Combined" color , "Output Composite", "CDT", and "Output GZIP". Then, proceed to "Output Directory".'),(0,n.yg)("div",{class:"tutorial-img-flow-container"},(0,n.yg)("img",{src:a(7237).A,style:{width:"100%"}})),(0,n.yg)("h4",{id:"54-when-ready-select-pile-tags-to-execute"},"5.4. When ready, select \u2018Pile Tags\u2019 to execute"),(0,n.yg)("admonition",{type:"tip"},(0,n.yg)("p",{parentName:"admonition"},"The default parameters Tag Pileup is set to expect is a sequence-specific strand separated Atac-seq dataset. Modifications to these parameters are needed for more specific analysis or when using data generated from other assays.")),(0,n.yg)("h4",{id:"55-save-composite-results-as-png"},"5.5. Save composite results as PNG"),(0,n.yg)("div",{class:"tutorial-img-flow-container"},(0,n.yg)("img",{src:a(334).A,style:{width:"100%"}})),(0,n.yg)("ul",null,(0,n.yg)("li",{parentName:"ul"},(0,n.yg)("p",{parentName:"li"},"The displayed composite plot can be modified by right-clicking and selecting properties. Things such as axis labels, axis range, and colors can be modified here.")),(0,n.yg)("li",{parentName:"ul"},(0,n.yg)("p",{parentName:"li"},"The final image can then be saved by right-clicking and selecting \u2018Save as\u2019. PNG is fine for most cases, but SVG is strongly recommended if this composite plot will be used in Adobe Illustrator later."))),(0,n.yg)("admonition",{type:"note"},(0,n.yg)("p",{parentName:"admonition"},"Besides the composite plot image, ScriptManager has saved the matrix ",(0,n.yg)("inlineCode",{parentName:"p"},"*.cdt")," files to your Output Directory together with the composite plot values file (If you didn't change the name it would be called ",(0,n.yg)("inlineCode",{parentName:"p"},"composite_average.out"),"). These CDT files will be used as the input for generating heatmaps in the next step.")),(0,n.yg)("h3",{id:"6-sort-bed-file-by-cdt-file"},"6. Sort BED file by CDT file"),(0,n.yg)("p",null,"Use Sort BED files by CDT to sort the entries based on chromosome name (C), followed by the start position (D), and then by the end position (T)."),(0,n.yg)("h4",{id:"61-navigate-to-coordinate-file-manipulation-\ufe0f-sort-bed"},"6.1. Navigate to Coordinate File Manipulation \u27a1\ufe0f ",(0,n.yg)("a",{parentName:"h4",href:"/docs/Tools/coordinate-manipulation/sort-bed"},(0,n.yg)("strong",{parentName:"a"},"Sort BED")),"."),(0,n.yg)("div",{class:"tutorial-img-flow-container"},(0,n.yg)("img",{src:a(1292).A,style:{width:"60%"}})),(0,n.yg)("h4",{id:"62-load-bed--cdt-file-from-tag-pileup"},"6.2. Load BED & CDT File from Tag Pileup"),(0,n.yg)("div",{class:"tutorial-img-flow-container"},(0,n.yg)("img",{src:a(2589).A,style:{width:"80%"}})),(0,n.yg)("h4",{id:"63-change-expansion-size-to-1000-from-center"},"6.3. Change Expansion Size to 1000 from Center"),(0,n.yg)("div",{class:"tutorial-img-flow-container"},(0,n.yg)("img",{src:a(5271).A,style:{width:"80%"}})),(0,n.yg)("h4",{id:"64-select-base-output-file-name-for-sorted-bed--cdt-files"},"6.4. Select Base Output File Name for sorted BED & CDT Files"),(0,n.yg)("div",{class:"tutorial-img-flow-container"},(0,n.yg)("img",{src:a(3553).A,style:{width:"80%"}})),(0,n.yg)("h3",{id:"7-generate-heatmaps"},"7. Generate Heatmaps"),(0,n.yg)("h4",{id:"71-navigate-to-figure-generation-\ufe0f-heat-map"},"7.1. Navigate to Figure Generation \u27a1\ufe0f ",(0,n.yg)("a",{parentName:"h4",href:"/docs/Tools/figure-generation/heatmap"},(0,n.yg)("strong",{parentName:"a"},"Heat Map")),"."),(0,n.yg)("div",{class:"tutorial-img-flow-container"},(0,n.yg)("img",{src:a(5112).A,style:{width:"80%"}})),(0,n.yg)("h4",{id:"72-load-the-sorted-cdt-file"},"7.2. Load the sorted CDT file"),(0,n.yg)("h4",{id:"73-select-custom-for-color-and-change-the-contrast-threshold-to-percentile-valuethe-heatmaps-for-this-dataset-show-the-best-contrast-when-using-the-default-percentile-threshold-value-95-or-95"},'7.3. Select "Custom" for color, and change the Contrast Threshold to "Percentile Value".The heatmaps for this dataset show the best contrast when using the default "Percentile Threshold" value (.95 or 95%).'),(0,n.yg)("h4",{id:"74-click-the-output-heatmap-checkbox-the-heatmap-generator-does-not-save-the-produced-png-by-default"},'7.4. Click the "Output Heatmap" checkbox. The Heatmap generator does not save the produced PNG by default.'),(0,n.yg)("h4",{id:"75-click-generate-to-save-your-png-heatmap"},'7.5. Click "Generate" to save your PNG heatmap!'),(0,n.yg)("div",{class:"tutorial-img-flow-container"},(0,n.yg)("img",{src:a(9911).A,style:{width:"55%"}}),(0,n.yg)("i",{class:"fa-solid fa-arrow-right fa-lg"}),(0,n.yg)("img",{src:a(9703).A,style:{width:"45%"}})),(0,n.yg)("h2",{id:"command-line-shell-script"},"Command-Line shell script"),(0,n.yg)("p",null,"The following shell commands records the locations for a BED file, a BAM file, and the anticipated OUTPUT basename as environmental variables to derive the corresponding composite plot values and heatmaps. This can serve as a template for you to write out your own workflows as bash scripts that execute command-line style ScriptManager."),(0,n.yg)("pre",null,(0,n.yg)("code",{parentName:"pre",className:"language-bash"},"SCRIPTMANAGER=/path/to/ScriptManager.jar\nBEDFILE=/path/to/UCSC_GRCh38_knownGene_GENCODEV3_ALL-TSS_2000bp.bed\nBAMFILE=/path/to/ENCFF128WZG.bam\nOUTPUT=/path/to/myoutput\n\nsamtools index $BAMFILE\n\njava -jar $SCRIPTMANAGER bam-statistics pe-stat $BAMFILE -o $OUTPUT\njava -jar $SCRIPTMANAGER coordinate-manipulation expand-bed -c 5000 $BEDFILE -o $OUTPUT\\_5000bp.bed\njava -jar $SCRIPTMANAGER read-analysis tag-pileup --combined --full-fragment $OUTPUT\\_5000bp.bed $BAMFILE -o $OUTPUT\\_composite.out -M $OUTPUT\\_matrix\njava -jar $SCRIPTMANAGER coordinate-manipulation sort-bed -c 1000 $OUTPUT\\_5000bp.bed $OUTPUT\\_matrix_combined.cdt -o $OUTPUT\\_SORT\njava -jar $SCRIPTMANAGER figure-generation heatmap -p .95 --black $OUTPUT\\_SORT.cdt -o $OUTPUT\\_heatmap.png\n\n# Output files:\n#  - /path/to/myoutput_InsertHistogram.out\n#  - /path/to/myoutput_PE.png\n#  - /path/to/myoutput_5000bp.bed\n#  - /path/to/myoutput_composite.out\n#  - /path/to/myoutput_matrix_combined.cdt\n#  - /path/to/myoutput_SORT.bed\n#  - /path/to/myoutput_SORT.cdt\n#  - /path/to/myoutput_heatmap.png\n")))}h.isMDXComponent=!0},9270:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/gui-bamidx-complete-e925a25e686f6c6813848dd104f8cfd5.png"},3254:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/gui-expandbed-complete-031750880fd32f9fc0d7002c35bd022d.png"},1483:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/maingui-bamidx-916174344aa9670cd3fe1bb3f0287a74.png"},5721:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/maingui-expandbed-c34fe4ec13cf11c4c61fa108f2a8d5a2.png"},5112:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/maingui-heatmap-9bd8cf5b3e862579d99b4ffb1b913ca5.png"},3731:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/maingui-tagpileup-a595f39070628d9d700325cc9d77895d.png"},1440:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/Atac-Seq_heatmap-a75289a9c820c8412fe87fe307e89165.png"},3069:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/Atac-seq_composite-845f021ab408c88dab9988fb1938fd72.png"},8333:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/bamstatistics(1)-dae4bd6e3bee1e1120f78b910c608921.png"},3286:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/bamstats(2)-076eca6cdb7600135fe9db789588cbd5.png"},7423:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/bamstatsfile-6f83953d558cfe1f2ed4c4fa717354f6.png"},5271:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/bed+cdt_expansion-e98e1f78c4c1ae6fb3cc2ce5a5a3ae68.png"},1292:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/bedtocdt_coordinate-190006f2fb8bbb8420430898acf2de4d.png"},9693:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/gui-bamidx2-1d5c74c923dcf82e75a6a12a937aecac.png"},7979:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/gui-expandbed2-cfc8169c7c7047827eda0c1b3ed43734.png"},9703:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/gui-heatmap-out-1406d501dfb7af1decb4cda0d71dc6cd.png"},9911:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/gui-heatmp-cee997f1f5c2a5f9aba471e38fb6104b.png"},7237:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/gui-tagpileup2-3c1bc5c28a8356b5877937152b0ab83e.png"},2589:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/loadbed:bam-dee5e90a295534d4d59d3d5f02c4b617.png"},3553:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/outputfile-1aedb78a0b28373859523573a090a834.png"},334:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/tagpileupsave-7b5a85d9779cc8d2658af266703e0c15.png"},2692:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/LinuxOpenSM-1fe13f9ab9c369dd24c0a67c7ee86f41.png"},6176:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/MacOpenSM-5f0508d59e500792ea199946818706f1.png"},3630:(e,t,a)=>{a.d(t,{A:()=>i});const i=a.p+"assets/images/WindowsOpenSM-ac32d3855e9f0cf632ab34c5c9fc983d.png"}}]);