"use strict";(self.webpackChunkscriptmanager_docs=self.webpackChunkscriptmanager_docs||[]).push([[5750],{5680:(e,t,a)=>{a.d(t,{xA:()=>c,yg:()=>y});var n=a(6540);function r(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function i(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function o(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?i(Object(a),!0).forEach((function(t){r(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):i(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function l(e,t){if(null==e)return{};var a,n,r=function(e,t){if(null==e)return{};var a,n,r={},i=Object.keys(e);for(n=0;n<i.length;n++)a=i[n],t.indexOf(a)>=0||(r[a]=e[a]);return r}(e,t);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(n=0;n<i.length;n++)a=i[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(r[a]=e[a])}return r}var s=n.createContext({}),p=function(e){var t=n.useContext(s),a=t;return e&&(a="function"==typeof e?e(t):o(o({},t),e)),a},c=function(e){var t=p(e.components);return n.createElement(s.Provider,{value:t},e.children)},u="mdxType",m={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},d=n.forwardRef((function(e,t){var a=e.components,r=e.mdxType,i=e.originalType,s=e.parentName,c=l(e,["components","mdxType","originalType","parentName"]),u=p(a),d=r,y=u["".concat(s,".").concat(d)]||u[d]||m[d]||i;return a?n.createElement(y,o(o({ref:t},c),{},{components:a})):n.createElement(y,o({ref:t},c))}));function y(e,t){var a=arguments,r=t&&t.mdxType;if("string"==typeof e||r){var i=a.length,o=new Array(i);o[0]=d;var l={};for(var s in t)hasOwnProperty.call(t,s)&&(l[s]=t[s]);l.originalType=e,l[u]="string"==typeof e?e:r,o[1]=l;for(var p=2;p<i;p++)o[p]=a[p];return n.createElement.apply(null,o)}return n.createElement.apply(null,a)}d.displayName="MDXCreateElement"},1125:(e,t,a)=>{a.r(t),a.d(t,{assets:()=>s,contentTitle:()=>o,default:()=>m,frontMatter:()=>i,metadata:()=>l,toc:()=>p});var n=a(8168),r=(a(6540),a(5680));const i={id:"search-motif",title:"Search Motif",sidebar_label:"search-motif"},o=void 0,l={unversionedId:"Tools/sequence-analysis/search-motif",id:"Tools/sequence-analysis/search-motif",title:"Search Motif",description:"search-motif",source:"@site/docs/Tools/sequence-analysis/search-motif.md",sourceDirName:"Tools/sequence-analysis",slug:"/Tools/sequence-analysis/search-motif",permalink:"/scriptmanager-docs/docs/Tools/sequence-analysis/search-motif",draft:!1,tags:[],version:"current",lastUpdatedBy:"Olivia Wen-Mei Lang",lastUpdatedAt:1724193973,formattedLastUpdatedAt:"Aug 20, 2024",frontMatter:{id:"search-motif",title:"Search Motif",sidebar_label:"search-motif"},sidebar:"toolSidebar",previous:{title:"randomize-fasta",permalink:"/scriptmanager-docs/docs/Tools/sequence-analysis/randomize-fasta"}},s={},p=[{value:"File inputs (FASTA)",id:"file-inputs-fasta",level:3},{value:"Search Options",id:"search-options",level:3},{value:"Positional Input",id:"positional-input",level:3},{value:"Output Options",id:"output-options",level:3},{value:"Search Options",id:"search-options-1",level:3}],c={toc:p},u="wrapper";function m(e){let{components:t,...i}=e;return(0,r.yg)(u,(0,n.A)({},c,i,{components:t,mdxType:"MDXLayout"}),(0,r.yg)("p",null,(0,r.yg)("img",{alt:"search-motif",src:a(8877).A,width:"103",height:"103"})),(0,r.yg)("p",null,"Search for an IUPAC DNA sequence motif in FASTA files with mismatches allowed"),(0,r.yg)("img",{src:a(800).A,style:{width:"70%"}}),(0,r.yg)("h3",{id:"file-inputs-fasta"},"File inputs (FASTA)"),(0,r.yg)("p",null,"Each input FASTA-formatted set of sequences has an average shape score series calculated for it. Because the shape score is a series corresponding to the bp position, the FASTA sequences input should be positionally linked to some feature and of the same length."),(0,r.yg)("h3",{id:"search-options"},"Search Options"),(0,r.yg)("ol",null,(0,r.yg)("li",{parentName:"ol"},"The IUPAC Motif (International Union of Pure and Applied Chemistry) is a code that represents nucletide sequences with ambiguty code:")),(0,r.yg)("ul",null,(0,r.yg)("li",{parentName:"ul"},"'A': Adenine"),(0,r.yg)("li",{parentName:"ul"},"'T': Thymine"),(0,r.yg)("li",{parentName:"ul"},"'C': Cytosine"),(0,r.yg)("li",{parentName:"ul"},"'G': Guanine"),(0,r.yg)("li",{parentName:"ul"},"'R': Purine (A or G)"),(0,r.yg)("li",{parentName:"ul"},"'Y': Pyrimidine (C or T)"),(0,r.yg)("li",{parentName:"ul"},"'N': Any Nucleotide (A, T, C, or G)")),(0,r.yg)("p",null,"When you 'Enter an IUPAC motif', you provide a sequence that can include these IUPAC codes to define a DNA pattern you want to search for. "),(0,r.yg)("ol",{start:2},(0,r.yg)("li",{parentName:"ol"},"The 'Enter Mismatches Allowed' option species the  number of mismatched nucleotides that can be tolerated when searching for the motif in the FASTA sequences. Mismatches are positions in the sequence where the nucleotides do not exactly match the motif but are still considered a valid match.")),(0,r.yg)("h1",{id:"command-line-interface"},"Command Line Interface"),(0,r.yg)("p",null,"Usage:"),(0,r.yg)("pre",null,(0,r.yg)("code",{parentName:"pre",className:"language-bash"},"java -jar ScriptManager.jar sequence-analysis search-motif [-hV] -m=<motif>\n[-n=<ALLOWED_MISMATCH>] [-o=<output>] <fastaFile>\n")),(0,r.yg)("h3",{id:"positional-input"},"Positional Input"),(0,r.yg)("table",null,(0,r.yg)("thead",{parentName:"table"},(0,r.yg)("tr",{parentName:"thead"},(0,r.yg)("th",{parentName:"tr",align:null},"Option"),(0,r.yg)("th",{parentName:"tr",align:null},"Description"))),(0,r.yg)("tbody",{parentName:"table"},(0,r.yg)("tr",{parentName:"tbody"},(0,r.yg)("td",{parentName:"tr",align:null},(0,r.yg)("inlineCode",{parentName:"td"},"<fastaFile>")),(0,r.yg)("td",{parentName:"tr",align:null},"reference genome FASTA file")))),(0,r.yg)("h3",{id:"output-options"},"Output Options"),(0,r.yg)("table",null,(0,r.yg)("thead",{parentName:"table"},(0,r.yg)("tr",{parentName:"thead"},(0,r.yg)("th",{parentName:"tr",align:null},"Option"),(0,r.yg)("th",{parentName:"tr",align:null},"Description"))),(0,r.yg)("tbody",{parentName:"table"},(0,r.yg)("tr",{parentName:"tbody"},(0,r.yg)("td",{parentName:"tr",align:null},(0,r.yg)("inlineCode",{parentName:"td"},"-o, --output=<output>")),(0,r.yg)("td",{parentName:"tr",align:null},"specify output file")),(0,r.yg)("tr",{parentName:"tbody"},(0,r.yg)("td",{parentName:"tr",align:null},(0,r.yg)("inlineCode",{parentName:"td"},"-z, --gzip")),(0,r.yg)("td",{parentName:"tr",align:null},"gzip output (default=false)")))),(0,r.yg)("h3",{id:"search-options-1"},"Search Options"),(0,r.yg)("table",null,(0,r.yg)("thead",{parentName:"table"},(0,r.yg)("tr",{parentName:"thead"},(0,r.yg)("th",{parentName:"tr",align:null},"Option"),(0,r.yg)("th",{parentName:"tr",align:null},"Description"))),(0,r.yg)("tbody",{parentName:"table"},(0,r.yg)("tr",{parentName:"tbody"},(0,r.yg)("td",{parentName:"tr",align:null},(0,r.yg)("inlineCode",{parentName:"td"},"-m, --motif=<motif>")),(0,r.yg)("td",{parentName:"tr",align:null},"the IUPAC motif to search for")),(0,r.yg)("tr",{parentName:"tbody"},(0,r.yg)("td",{parentName:"tr",align:null},(0,r.yg)("inlineCode",{parentName:"td"},"-n, --mismatches=<ALLOWED_MISMATCH>")),(0,r.yg)("td",{parentName:"tr",align:null},"the number of mismatches allowed (default=0)")))))}m.isMDXComponent=!0},800:(e,t,a)=>{a.d(t,{A:()=>n});const n=a.p+"assets/images/SearchMotifWindow-f91eb9ef073eb34e97dd1420af0d64f1.png"},8877:(e,t,a)=>{a.d(t,{A:()=>n});const n=a.p+"assets/images/SearchMotif_square-873eb5425360d560bd63902f49b93e1b.svg"}}]);