"use strict";(self.webpackChunkscriptmanager_docs=self.webpackChunkscriptmanager_docs||[]).push([[8581],{5610:e=>{e.exports=JSON.parse('{"pluginId":"default","version":"current","label":"Next","banner":null,"badge":false,"noIndex":false,"className":"docs-version-current","isLast":true,"docsSidebars":{"generalSidebar":[{"type":"category","label":"Guides","items":[{"type":"link","label":"File Formats","href":"/scriptmanager-docs/docs/Guides/Getting-Started/file-formats","docId":"Guides/Getting-Started/file-formats"},{"type":"link","label":"Command Line","href":"/scriptmanager-docs/docs/Guides/Getting-Started/command-line","docId":"Guides/Getting-Started/command-line"},{"type":"link","label":"Color Selection Guide","href":"/scriptmanager-docs/docs/Guides/Getting-Started/color-guide","docId":"Guides/Getting-Started/color-guide"},{"type":"link","label":"Logging Manager Guide","href":"/scriptmanager-docs/docs/Guides/Getting-Started/logging-manager","docId":"Guides/Getting-Started/logging-manager"}],"collapsed":true,"collapsible":true,"href":"/scriptmanager-docs/docs/"},{"type":"link","label":"Tool Index (A-Z)","href":"/scriptmanager-docs/docs/Guides/tool-index","docId":"Guides/tool-index"},{"type":"category","label":"Tutorials","items":[{"type":"link","label":"ChIP-exo","href":"/scriptmanager-docs/docs/Guides/Tutorials/chipexo-tutorial","docId":"Guides/Tutorials/chipexo-tutorial"},{"type":"link","label":"Genomic Features","href":"/scriptmanager-docs/docs/Guides/Tutorials/genomic-features-tutorial","docId":"Guides/Tutorials/genomic-features-tutorial"},{"type":"link","label":"Genome Tracks (BigWig)","href":"/scriptmanager-docs/docs/Guides/Tutorials/genome-tracks-tutorial","docId":"Guides/Tutorials/genome-tracks-tutorial"},{"type":"link","label":"ATAC-seq","href":"/scriptmanager-docs/docs/Guides/Tutorials/atac-seq-tutorial","docId":"Guides/Tutorials/atac-seq-tutorial"}],"collapsed":true,"collapsible":true},{"type":"category","label":"Contributing","items":[{"type":"link","label":"General Developer Notes","href":"/scriptmanager-docs/docs/Guides/Contributing/developer-guidelines","docId":"Guides/Contributing/developer-guidelines"},{"type":"link","label":"Useful Code Snippets","href":"/scriptmanager-docs/docs/Guides/Contributing/code-snippets","docId":"Guides/Contributing/code-snippets"},{"type":"link","label":"Testing","href":"/scriptmanager-docs/docs/Guides/Contributing/testing","docId":"Guides/Contributing/testing"}],"collapsed":true,"collapsible":true},{"type":"category","label":"Community","items":[{"type":"link","label":"Code of Conduct","href":"/scriptmanager-docs/docs/Guides/Community/code-of-conduct","docId":"Guides/Community/code-of-conduct"}],"collapsed":true,"collapsible":true},{"type":"link","label":"Citing Us","href":"/scriptmanager-docs/docs/Guides/citing-us","docId":"Guides/citing-us"}],"toolSidebar":[{"type":"category","label":"BAM Format Converter","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"BAM to BED","href":"/scriptmanager-docs/docs/Tools/bam-format-converter/bam-to-bed","docId":"Tools/bam-format-converter/bam-to-bed"},{"type":"link","label":"BAM to bedGraph","href":"/scriptmanager-docs/docs/Tools/bam-format-converter/bam-to-bedgraph","docId":"Tools/bam-format-converter/bam-to-bedgraph"},{"type":"link","label":"BAM to GFF","href":"/scriptmanager-docs/docs/Tools/bam-format-converter/bam-to-gff","docId":"Tools/bam-format-converter/bam-to-gff"},{"type":"link","label":"BAM to scIDX","href":"/scriptmanager-docs/docs/Tools/bam-format-converter/bam-to-scidx","docId":"Tools/bam-format-converter/bam-to-scidx"}],"href":"/scriptmanager-docs/docs/Tools/bam-format-converter/"},{"type":"category","label":"BAM Manipulation","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"BAM Indexer","href":"/scriptmanager-docs/docs/Tools/bam-manipulation/bam-indexer","docId":"Tools/bam-manipulation/bam-indexer"},{"type":"link","label":"Filter PIP-seq","href":"/scriptmanager-docs/docs/Tools/bam-manipulation/filter-pip-seq","docId":"Tools/bam-manipulation/filter-pip-seq"},{"type":"link","label":"Mark Duplicates","href":"/scriptmanager-docs/docs/Tools/bam-manipulation/mark-duplicates","docId":"Tools/bam-manipulation/mark-duplicates"},{"type":"link","label":"Merge BAM","href":"/scriptmanager-docs/docs/Tools/bam-manipulation/merge-bam","docId":"Tools/bam-manipulation/merge-bam"},{"type":"link","label":"Sort BAM","href":"/scriptmanager-docs/docs/Tools/bam-manipulation/sort-bam","docId":"Tools/bam-manipulation/sort-bam"}],"href":"/scriptmanager-docs/docs/Tools/bam-manipulation/"},{"type":"category","label":"BAM Statistics","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"bam-correlation","href":"/scriptmanager-docs/docs/Tools/bam-statistics/bam-correlation","docId":"Tools/bam-statistics/bam-correlation"},{"type":"link","label":"BAM Cross Correlation (ArchTEx)","href":"/scriptmanager-docs/docs/Tools/bam-statistics/cross-corr","docId":"Tools/bam-statistics/cross-corr"},{"type":"link","label":"Paired-End Statistics","href":"/scriptmanager-docs/docs/Tools/bam-statistics/pe-stat","docId":"Tools/bam-statistics/pe-stat"},{"type":"link","label":"Single-End Statistics","href":"/scriptmanager-docs/docs/Tools/bam-statistics/se-stat","docId":"Tools/bam-statistics/se-stat"}],"href":"/scriptmanager-docs/docs/Tools/bam-statistics/"},{"type":"category","label":"Coordinate Manipulation","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"BED to GFF","href":"/scriptmanager-docs/docs/Tools/coordinate-manipulation/bed-to-gff","docId":"Tools/coordinate-manipulation/bed-to-gff"},{"type":"link","label":"Expand BED","href":"/scriptmanager-docs/docs/Tools/coordinate-manipulation/expand-bed","docId":"Tools/coordinate-manipulation/expand-bed"},{"type":"link","label":"Expand GFF","href":"/scriptmanager-docs/docs/Tools/coordinate-manipulation/expand-gff","docId":"Tools/coordinate-manipulation/expand-gff"},{"type":"link","label":"GFF to BED","href":"/scriptmanager-docs/docs/Tools/coordinate-manipulation/gff-to-bed","docId":"Tools/coordinate-manipulation/gff-to-bed"},{"type":"link","label":"Shift BED/GFF","href":"/scriptmanager-docs/docs/Tools/coordinate-manipulation/shift-coord","docId":"Tools/coordinate-manipulation/shift-coord"},{"type":"link","label":"sort-bed","href":"/scriptmanager-docs/docs/Tools/coordinate-manipulation/sort-bed","docId":"Tools/coordinate-manipulation/sort-bed"},{"type":"link","label":"sort-gff","href":"/scriptmanager-docs/docs/Tools/coordinate-manipulation/sort-gff","docId":"Tools/coordinate-manipulation/sort-gff"}],"href":"/scriptmanager-docs/docs/Tools/coordinate-manipulation/"},{"type":"category","label":"Figure Generation","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Composite Plot","href":"/scriptmanager-docs/docs/Tools/figure-generation/composite-plot","docId":"Tools/figure-generation/composite-plot"},{"type":"link","label":"Four Color Plot","href":"/scriptmanager-docs/docs/Tools/figure-generation/four-color","docId":"Tools/figure-generation/four-color"},{"type":"link","label":"Label Heatmap","href":"/scriptmanager-docs/docs/Tools/figure-generation/heatmap-labeler","docId":"Tools/figure-generation/heatmap-labeler"},{"type":"link","label":"Two-color Heatmap","href":"/scriptmanager-docs/docs/Tools/figure-generation/heatmap","docId":"Tools/figure-generation/heatmap"},{"type":"link","label":"Merge Heatmap","href":"/scriptmanager-docs/docs/Tools/figure-generation/merge-heatmap","docId":"Tools/figure-generation/merge-heatmap"},{"type":"link","label":"three-color-heatmap","href":"/scriptmanager-docs/docs/Tools/figure-generation/three-color-heatmap","docId":"Tools/figure-generation/three-color-heatmap"}],"href":"/scriptmanager-docs/docs/Tools/figure-generation/"},{"type":"category","label":"File Utilities","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Chr Name Converter","href":"/scriptmanager-docs/docs/Tools/file-utilities/chrname-converter","docId":"Tools/file-utilities/chrname-converter"},{"type":"link","label":"Compress/Decompress Files (Gzip)","href":"/scriptmanager-docs/docs/Tools/file-utilities/compress-files","docId":"Tools/file-utilities/compress-files"},{"type":"link","label":"MD5 Checksum","href":"/scriptmanager-docs/docs/Tools/file-utilities/md5checksum","docId":"Tools/file-utilities/md5checksum"}],"href":"/scriptmanager-docs/docs/Tools/file-utilities/"},{"type":"category","label":"Peak Analysis","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Filter BED by Proximity","href":"/scriptmanager-docs/docs/Tools/peak-analysis/filter-bed","docId":"Tools/peak-analysis/filter-bed"},{"type":"link","label":"FRiX Calculator","href":"/scriptmanager-docs/docs/Tools/peak-analysis/frix-score","docId":"Tools/peak-analysis/frix-score"},{"type":"link","label":"Align BED to Reference","href":"/scriptmanager-docs/docs/Tools/peak-analysis/peak-align-ref","docId":"Tools/peak-analysis/peak-align-ref"},{"type":"link","label":"Random Coordinate Generator","href":"/scriptmanager-docs/docs/Tools/peak-analysis/rand-coord","docId":"Tools/peak-analysis/rand-coord"},{"type":"link","label":"Signal Duplication (suspended)","href":"/scriptmanager-docs/docs/Tools/peak-analysis/signal-dup","docId":"Tools/peak-analysis/signal-dup"},{"type":"link","label":"sort-by-dist","href":"/scriptmanager-docs/docs/Tools/peak-analysis/sort-by-dist","docId":"Tools/peak-analysis/sort-by-dist"},{"type":"link","label":"Genomic Coordinate Tile","href":"/scriptmanager-docs/docs/Tools/peak-analysis/tile-genome","docId":"Tools/peak-analysis/tile-genome"}],"href":"/scriptmanager-docs/docs/Tools/peak-analysis/"},{"type":"category","label":"Read Analysis","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"aggregate-data","href":"/scriptmanager-docs/docs/Tools/read-analysis/aggregate-data","docId":"Tools/read-analysis/aggregate-data"},{"type":"link","label":"Scale Matrix","href":"/scriptmanager-docs/docs/Tools/read-analysis/scale-matrix","docId":"Tools/read-analysis/scale-matrix"},{"type":"link","label":"Scaling Factor","href":"/scriptmanager-docs/docs/Tools/read-analysis/scaling-factor","docId":"Tools/read-analysis/scaling-factor"},{"type":"link","label":"Similarity Matrix (suspended)","href":"/scriptmanager-docs/docs/Tools/read-analysis/similarity-matrix","docId":"Tools/read-analysis/similarity-matrix"},{"type":"link","label":"Tag Pileup","href":"/scriptmanager-docs/docs/Tools/read-analysis/tag-pileup","docId":"Tools/read-analysis/tag-pileup"},{"type":"link","label":"transpose-matrix","href":"/scriptmanager-docs/docs/Tools/read-analysis/transpose-matrix","docId":"Tools/read-analysis/transpose-matrix"}],"href":"/scriptmanager-docs/docs/Tools/read-analysis/"},{"type":"category","label":"Sequence Analysis","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"DNA Shape from BED","href":"/scriptmanager-docs/docs/Tools/sequence-analysis/dna-shape-bed","docId":"Tools/sequence-analysis/dna-shape-bed"},{"type":"link","label":"DNA Shape from FASTA","href":"/scriptmanager-docs/docs/Tools/sequence-analysis/dna-shape-fasta","docId":"Tools/sequence-analysis/dna-shape-fasta"},{"type":"link","label":"fasta-extract","href":"/scriptmanager-docs/docs/Tools/sequence-analysis/fasta-extract","docId":"Tools/sequence-analysis/fasta-extract"},{"type":"link","label":"randomize-fasta","href":"/scriptmanager-docs/docs/Tools/sequence-analysis/randomize-fasta","docId":"Tools/sequence-analysis/randomize-fasta"},{"type":"link","label":"search-motif","href":"/scriptmanager-docs/docs/Tools/sequence-analysis/search-motif","docId":"Tools/sequence-analysis/search-motif"}],"href":"/scriptmanager-docs/docs/Tools/sequence-analysis/"}]},"docs":{"Guides/citing-us":{"id":"Guides/citing-us","title":"Citing Us","description":"If you use ScriptManager in your work, you can use the Lang et al (2022) publication or use ScriptManager\'s unique RRID:SCR_021797.","sidebar":"generalSidebar"},"Guides/Community/code-of-conduct":{"id":"Guides/Community/code-of-conduct","title":"Community Code of Conduct","description":"This code of conduct outlines our expectations for participants within the","sidebar":"generalSidebar"},"Guides/Contributing/code-snippets":{"id":"Guides/Contributing/code-snippets","title":"Useful Code Snippets","description":"scripts package","sidebar":"generalSidebar"},"Guides/Contributing/developer-guidelines":{"id":"Guides/Contributing/developer-guidelines","title":"General Developer Notes","description":"Welcome to the ScriptManager developer wiki!","sidebar":"generalSidebar"},"Guides/Contributing/testing":{"id":"Guides/Contributing/testing","title":"Testing","description":"Automatic CLI Testing","sidebar":"generalSidebar"},"Guides/Getting-Started/color-guide":{"id":"Guides/Getting-Started/color-guide","title":"Color Selection Guide","description":"Some tools allow you to customize colors used in the output, specifically among the figure-generation tools. The following guide introduces the user to color customization in ScriptManager.","sidebar":"generalSidebar"},"Guides/Getting-Started/command-line":{"id":"Guides/Getting-Started/command-line","title":"Command Line","description":"The Command Line Interface (CLI) tools from ScriptManager were written to mirror the tools wrapped in the Graphical User Interface (GUI) version.","sidebar":"generalSidebar"},"Guides/Getting-Started/file-formats":{"id":"Guides/Getting-Started/file-formats","title":"File Formats","description":"A variety of standard file formats including BAM, GFF, BED, and CDT are used by the ScriptManager tools along with some custom file formats. The purpose of this guide is to help users understand what types of information these formats store and find tools in ScriptManager based on the format their data exists in.","sidebar":"generalSidebar"},"Guides/Getting-Started/getting-started":{"id":"Guides/Getting-Started/getting-started","title":"Getting Started","description":"Welcome to the ScriptManager wiki!","sidebar":"generalSidebar"},"Guides/Getting-Started/logging-manager":{"id":"Guides/Getting-Started/logging-manager","title":"Logging Manager Guide","description":"This guide introduces users to the logging feature in ScriptManager, which stores a record of a user\'s ScriptManager activity for a given graphical interface session.","sidebar":"generalSidebar"},"Guides/tool-index":{"id":"Guides/tool-index","title":"Tool Index (A-Z)","description":"Tool Index","sidebar":"generalSidebar"},"Guides/Tutorials/atac-seq-tutorial":{"id":"Guides/Tutorials/atac-seq-tutorial","title":"ATAC-seq Tutorial","description":"Generating two basic sequence-specific ATAC-Seq plots: composite and heatmap","sidebar":"generalSidebar"},"Guides/Tutorials/chipexo-tutorial":{"id":"Guides/Tutorials/chipexo-tutorial","title":"ChIP-exo Tutorial","description":"Generating two basic sequence-specific ChIP-exo plots: composite and heatmap","sidebar":"generalSidebar"},"Guides/Tutorials/genome-tracks-tutorial":{"id":"Guides/Tutorials/genome-tracks-tutorial","title":"Making Genome Tracks Tutorial","description":"How to turn your BAM files into genome track files that can be viewed in a genome browser","sidebar":"generalSidebar"},"Guides/Tutorials/genomic-features-tutorial":{"id":"Guides/Tutorials/genomic-features-tutorial","title":"Genomic Features Tutorial","description":"Generating four-color plots to compare positional sequence content across genomic sites","sidebar":"generalSidebar"},"tool-template":{"id":"tool-template","title":"Tool Name","description":"Brief (1-sentence) description (see tool description used in JAR)."},"Tools/bam-format-converter/bam-to-bed":{"id":"Tools/bam-format-converter/bam-to-bed","title":"BAM to BED","description":"bam-to-bed","sidebar":"toolSidebar"},"Tools/bam-format-converter/bam-to-bedgraph":{"id":"Tools/bam-format-converter/bam-to-bedgraph","title":"BAM to bedGraph","description":"bam-to-bedgraph","sidebar":"toolSidebar"},"Tools/bam-format-converter/bam-to-gff":{"id":"Tools/bam-format-converter/bam-to-gff","title":"BAM to GFF","description":"bam-to-gff","sidebar":"toolSidebar"},"Tools/bam-format-converter/bam-to-scidx":{"id":"Tools/bam-format-converter/bam-to-scidx","title":"BAM to scIDX","description":"bam-to-scidx","sidebar":"toolSidebar"},"Tools/bam-format-converter/index":{"id":"Tools/bam-format-converter/index","title":"BAM Format Converter","description":"Extract genome-wide pileup information from a BAM file.","sidebar":"toolSidebar"},"Tools/bam-manipulation/bam-indexer":{"id":"Tools/bam-manipulation/bam-indexer","title":"BAM Indexer","description":"Picard - BuildBamIndex","sidebar":"toolSidebar"},"Tools/bam-manipulation/filter-pip-seq":{"id":"Tools/bam-manipulation/filter-pip-seq","title":"Filter PIP-seq","description":"filter-pip-seq","sidebar":"toolSidebar"},"Tools/bam-manipulation/index":{"id":"Tools/bam-manipulation/index","title":"BAM Manipulation","description":"Perform various utility functions on BAM files (e.g. index, sort, remove duplicates, merge, filter).","sidebar":"toolSidebar"},"Tools/bam-manipulation/mark-duplicates":{"id":"Tools/bam-manipulation/mark-duplicates","title":"Mark Duplicates (Picard)","description":"Picard - MarkDuplicates","sidebar":"toolSidebar"},"Tools/bam-manipulation/merge-bam":{"id":"Tools/bam-manipulation/merge-bam","title":"Merge BAM","description":"Picard - MergeBamAlignment","sidebar":"toolSidebar"},"Tools/bam-manipulation/sort-bam":{"id":"Tools/bam-manipulation/sort-bam","title":"Sort BAM","description":"Picard - SortSam","sidebar":"toolSidebar"},"Tools/bam-statistics/bam-correlation":{"id":"Tools/bam-statistics/bam-correlation","title":"BAM Correlation","description":"bam-correlation","sidebar":"toolSidebar"},"Tools/bam-statistics/cross-corr":{"id":"Tools/bam-statistics/cross-corr","title":"BAM Cross Correlation (ArchTEx)","description":"Calculate optimal tag shift based on ArchTEx implementation from Lai et al, 2012 (PMID:22302569).","sidebar":"toolSidebar"},"Tools/bam-statistics/index":{"id":"Tools/bam-statistics/index","title":"BAM Statistics","description":"Perform various statistical checks on BAM files (e.g. insert size distributions and coverage correlations).","sidebar":"toolSidebar"},"Tools/bam-statistics/pe-stat":{"id":"Tools/bam-statistics/pe-stat","title":"Paired-End Statistics","description":"pe-stat","sidebar":"toolSidebar"},"Tools/bam-statistics/se-stat":{"id":"Tools/bam-statistics/se-stat","title":"Single-End Statistics","description":"se-stat","sidebar":"toolSidebar"},"Tools/coordinate-manipulation/bed-to-gff":{"id":"Tools/coordinate-manipulation/bed-to-gff","title":"Convert BED to GFF","description":"BEDtoGFF","sidebar":"toolSidebar"},"Tools/coordinate-manipulation/expand-bed":{"id":"Tools/coordinate-manipulation/expand-bed","title":"Expand BED File","description":"ExpandBED","sidebar":"toolSidebar"},"Tools/coordinate-manipulation/expand-gff":{"id":"Tools/coordinate-manipulation/expand-gff","title":"Expand GFF File","description":"expand-gff","sidebar":"toolSidebar"},"Tools/coordinate-manipulation/gff-to-bed":{"id":"Tools/coordinate-manipulation/gff-to-bed","title":"Convert GFF to BED","description":"gff-to-bed","sidebar":"toolSidebar"},"Tools/coordinate-manipulation/index":{"id":"Tools/coordinate-manipulation/index","title":"Coordinate Manipulation","description":"Perform manipulations andd transformations of coordinate interval files (e.g. expansion, sort, filter, convert).","sidebar":"toolSidebar"},"Tools/coordinate-manipulation/shift-coord":{"id":"Tools/coordinate-manipulation/shift-coord","title":"Shift BED/GFF","description":"Shift coordinate intervals up/downstream.","sidebar":"toolSidebar"},"Tools/coordinate-manipulation/sort-bed":{"id":"Tools/coordinate-manipulation/sort-bed","title":"Sort BED by CDT","description":"sort-bed","sidebar":"toolSidebar"},"Tools/coordinate-manipulation/sort-gff":{"id":"Tools/coordinate-manipulation/sort-gff","title":"Sort GFF by CDT","description":"sort-gff","sidebar":"toolSidebar"},"Tools/figure-generation/composite-plot":{"id":"Tools/figure-generation/composite-plot","title":"Composite Plot","description":"Composite Plot","sidebar":"toolSidebar"},"Tools/figure-generation/four-color":{"id":"Tools/figure-generation/four-color","title":"Four Color Sequence Plot","description":"four-color","sidebar":"toolSidebar"},"Tools/figure-generation/heatmap":{"id":"Tools/figure-generation/heatmap","title":"Two-color Heatmap","description":"Two-colorheatmap","sidebar":"toolSidebar"},"Tools/figure-generation/heatmap-labeler":{"id":"Tools/figure-generation/heatmap-labeler","title":"Label Heatmap","description":"Label Heatmap","sidebar":"toolSidebar"},"Tools/figure-generation/index":{"id":"Tools/figure-generation/index","title":"Figure Generation","description":"Make publication quality figures from heatmap matrix data and sequences.","sidebar":"toolSidebar"},"Tools/figure-generation/merge-heatmap":{"id":"Tools/figure-generation/merge-heatmap","title":"Merge Heatmap","description":"Merge Heatmap","sidebar":"toolSidebar"},"Tools/figure-generation/three-color-heatmap":{"id":"Tools/figure-generation/three-color-heatmap","title":"Three Color Heatmap","description":"Three Color Heatmap","sidebar":"toolSidebar"},"Tools/file-utilities/chrname-converter":{"id":"Tools/file-utilities/chrname-converter","title":"Chromosome Name Converter","description":"ChrNameConverter","sidebar":"toolSidebar"},"Tools/file-utilities/compress-files":{"id":"Tools/file-utilities/compress-files","title":"Compress/Decompress Files","description":"Compress files to help with limited storage issues and faster reading/writing of files.","sidebar":"toolSidebar"},"Tools/file-utilities/index":{"id":"Tools/file-utilities/index","title":"File Utilities","description":"Miscellaneous GUI wrappers for common file utilities (e.g. checksums, compression, and yeast genome conversion)","sidebar":"toolSidebar"},"Tools/file-utilities/md5checksum":{"id":"Tools/file-utilities/md5checksum","title":"MD5 Checksum","description":"MD5Checksum","sidebar":"toolSidebar"},"Tools/peak-analysis/filter-bed":{"id":"Tools/peak-analysis/filter-bed","title":"Filter BED by Proximity","description":"filter-bed","sidebar":"toolSidebar"},"Tools/peak-analysis/frix-score":{"id":"Tools/peak-analysis/frix-score","title":"FRiX Score Calculator","description":"Quantify library enrichment around a set of Reference points (Fraction of Reads in Peak/Motif, aka FRiP/FRiM).","sidebar":"toolSidebar"},"Tools/peak-analysis/index":{"id":"Tools/peak-analysis/index","title":"Peak Analysis","description":"Useful peak manipulation (filter, comparison, and simulation) utilities.","sidebar":"toolSidebar"},"Tools/peak-analysis/peak-align-ref":{"id":"Tools/peak-analysis/peak-align-ref","title":"Align BED to Reference","description":"peak-align-ref","sidebar":"toolSidebar"},"Tools/peak-analysis/rand-coord":{"id":"Tools/peak-analysis/rand-coord","title":"Random Coordinate Generator","description":"rand-coord","sidebar":"toolSidebar"},"Tools/peak-analysis/signal-dup":{"id":"Tools/peak-analysis/signal-dup","title":"Signal Duplication (suspended)","description":"Signal Duplication is still under development and not yet actively supported.","sidebar":"toolSidebar"},"Tools/peak-analysis/sort-by-dist":{"id":"Tools/peak-analysis/sort-by-dist","title":"Sort Coordinate By Reference","description":"Sort BED/GFF by distance to another BED/GFF (e.g. peak annotations).","sidebar":"toolSidebar"},"Tools/peak-analysis/tile-genome":{"id":"Tools/peak-analysis/tile-genome","title":"Genomic Coordinate Tile","description":"tile-genome","sidebar":"toolSidebar"},"Tools/read-analysis/aggregate-data":{"id":"Tools/read-analysis/aggregate-data","title":"Aggregate Data","description":"aggregate-data","sidebar":"toolSidebar"},"Tools/read-analysis/index":{"id":"Tools/read-analysis/index","title":"Read Analysis","description":"These Read Analysis tools are mostly tools that work with the CDT or other tab-delimited Matrix file formats.","sidebar":"toolSidebar"},"Tools/read-analysis/scale-matrix":{"id":"Tools/read-analysis/scale-matrix","title":"Scale Matrix","description":"scale-matrix","sidebar":"toolSidebar"},"Tools/read-analysis/scaling-factor":{"id":"Tools/read-analysis/scaling-factor","title":"Scaling Factor","description":"scaling-factor","sidebar":"toolSidebar"},"Tools/read-analysis/similarity-matrix":{"id":"Tools/read-analysis/similarity-matrix","title":"Similarity Matrix (suspended)","description":"Similarity Matrix is still under development and not yet actively supported.","sidebar":"toolSidebar"},"Tools/read-analysis/tag-pileup":{"id":"Tools/read-analysis/tag-pileup","title":"Tag Pileup","description":"tag-pileup","sidebar":"toolSidebar"},"Tools/read-analysis/transpose-matrix":{"id":"Tools/read-analysis/transpose-matrix","title":"Transpose Matrix","description":"transpose-matrix","sidebar":"toolSidebar"},"Tools/sequence-analysis/dna-shape-bed":{"id":"Tools/sequence-analysis/dna-shape-bed","title":"DNA Shape from BED File","description":"dna-shape-bed","sidebar":"toolSidebar"},"Tools/sequence-analysis/dna-shape-fasta":{"id":"Tools/sequence-analysis/dna-shape-fasta","title":"DNA Shape from FASTA File","description":"dna-shape-fasta","sidebar":"toolSidebar"},"Tools/sequence-analysis/fasta-extract":{"id":"Tools/sequence-analysis/fasta-extract","title":"Extract FASTA","description":"fasta-extract","sidebar":"toolSidebar"},"Tools/sequence-analysis/index":{"id":"Tools/sequence-analysis/index","title":"Sequence Analysis","description":"Tools for analyzing sequence files (e.g. shape analysis, motif searches, randomization, and extractions).","sidebar":"toolSidebar"},"Tools/sequence-analysis/randomize-fasta":{"id":"Tools/sequence-analysis/randomize-fasta","title":"Randomize FASTA","description":"randomize-fasta","sidebar":"toolSidebar"},"Tools/sequence-analysis/search-motif":{"id":"Tools/sequence-analysis/search-motif","title":"Search Motif","description":"search-motif","sidebar":"toolSidebar"}}}')}}]);