"use strict";(self.webpackChunkscriptmanager_docs=self.webpackChunkscriptmanager_docs||[]).push([[1811],{3905:(e,t,a)=>{a.d(t,{Zo:()=>p,kt:()=>m});var o=a(7294);function n(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function i(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);t&&(o=o.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,o)}return a}function r(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?i(Object(a),!0).forEach((function(t){n(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):i(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function s(e,t){if(null==e)return{};var a,o,n=function(e,t){if(null==e)return{};var a,o,n={},i=Object.keys(e);for(o=0;o<i.length;o++)a=i[o],t.indexOf(a)>=0||(n[a]=e[a]);return n}(e,t);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(o=0;o<i.length;o++)a=i[o],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(n[a]=e[a])}return n}var l=o.createContext({}),c=function(e){var t=o.useContext(l),a=t;return e&&(a="function"==typeof e?e(t):r(r({},t),e)),a},p=function(e){var t=c(e.components);return o.createElement(l.Provider,{value:t},e.children)},d={inlineCode:"code",wrapper:function(e){var t=e.children;return o.createElement(o.Fragment,{},t)}},u=o.forwardRef((function(e,t){var a=e.components,n=e.mdxType,i=e.originalType,l=e.parentName,p=s(e,["components","mdxType","originalType","parentName"]),u=c(a),m=n,f=u["".concat(l,".").concat(m)]||u[m]||d[m]||i;return a?o.createElement(f,r(r({ref:t},p),{},{components:a})):o.createElement(f,r({ref:t},p))}));function m(e,t){var a=arguments,n=t&&t.mdxType;if("string"==typeof e||n){var i=a.length,r=new Array(i);r[0]=u;var s={};for(var l in t)hasOwnProperty.call(t,l)&&(s[l]=t[l]);s.originalType=e,s.mdxType="string"==typeof e?e:n,r[1]=s;for(var c=2;c<i;c++)r[c]=a[c];return o.createElement.apply(null,r)}return o.createElement.apply(null,a)}u.displayName="MDXCreateElement"},433:(e,t,a)=>{a.r(t),a.d(t,{assets:()=>l,contentTitle:()=>r,default:()=>d,frontMatter:()=>i,metadata:()=>s,toc:()=>c});var o=a(7462),n=(a(7294),a(3905));const i={id:"genomic-features-tutorial",title:"Genomic Features Tutorial",sidebar_label:"Genomic Features Tutorial"},r=void 0,s={unversionedId:"Tutorials/genomic-features-tutorial",id:"Tutorials/genomic-features-tutorial",title:"Genomic Features Tutorial",description:"Generating four-color plots to compare positional sequence content across genomic sites",source:"@site/docs/Tutorials/genomic-features-tutorial.md",sourceDirName:"Tutorials",slug:"/Tutorials/genomic-features-tutorial",permalink:"/scriptmanager-docs/docs/Tutorials/genomic-features-tutorial",draft:!1,tags:[],version:"current",lastUpdatedBy:"Olivia Wen-Mei Lang",lastUpdatedAt:1674679815,formattedLastUpdatedAt:"Jan 25, 2023",frontMatter:{id:"genomic-features-tutorial",title:"Genomic Features Tutorial",sidebar_label:"Genomic Features Tutorial"},sidebar:"someSidebar",previous:{title:"ChIP-exo Tutorial",permalink:"/scriptmanager-docs/docs/Tutorials/chipexo-tutorial"},next:{title:"File Formats",permalink:"/scriptmanager-docs/docs/file-formats"}},l={},c=[{value:"Download ScriptManager (v0.14):",id:"download-scriptmanager-v014",level:2},{value:"Download data files",id:"download-data-files",level:2},{value:"Generate the Four-color Plot",id:"generate-the-four-color-plot",level:2},{value:"1. Open ScriptManager",id:"1-open-scriptmanager",level:3},{value:"2. Resize the Reb1 motif-aligned BED file",id:"2-resize-the-reb1-motif-aligned-bed-file",level:3},{value:"2.1. Navigate to Coordinate File Manipulation \u27a1\ufe0f Expand BED File",id:"21-navigate-to-coordinate-file-manipulation-\ufe0f-expand-bed-file",level:4},{value:"2.2. For this tutorial, use the 50bp expansion and select &quot;Expand from Center&quot;.",id:"22-for-this-tutorial-use-the-50bp-expansion-and-select-expand-from-center",level:4},{value:"3. Generate the FASTA sequence input",id:"3-generate-the-fasta-sequence-input",level:3},{value:"3.1. Navigate to DNA Sequence Analysis \u27a1\ufe0f FASTA from BED to create the input for generating a Four Color plot.",id:"31-navigate-to-dna-sequence-analysis-\ufe0f-fasta-from-bed-to-create-the-input-for-generating-a-four-color-plot",level:4},{value:"3.2. Load the FASTA file containing the Genome FASTA",id:"32-load-the-fasta-file-containing-the-genome-fasta",level:4},{value:"3.3. Load appropriate BED file for sequence FASTA generation.",id:"33-load-appropriate-bed-file-for-sequence-fasta-generation",level:4},{value:"3.4. Click &quot;Calculate&quot; to start the extraction which outputs a FASTA file",id:"34-click-calculate-to-start-the-extraction-which-outputs-a-fasta-file",level:4},{value:"4. Generate the Four color sequence image",id:"4-generate-the-four-color-sequence-image",level:3},{value:"4.1. Navigate to Figure Generation \u27a1\ufe0f Four Color Plot to generate the plot once you have generated the FASTA file of the sequences within the BED regions.",id:"41-navigate-to-figure-generation-\ufe0f-four-color-plot-to-generate-the-plot-once-you-have-generated-the-fasta-file-of-the-sequences-within-the-bed-regions",level:4},{value:"4.2. Load the FASTA file containing the FASTA sequences (generated in step 2).",id:"42-load-the-fasta-file-containing-the-fasta-sequences-generated-in-step-2",level:4},{value:"4.3. Click &quot;Generate&quot; to execute the script.",id:"43-click-generate-to-execute-the-script",level:4},{value:"Command-Line shell script",id:"command-line-shell-script",level:2}],p={toc:c};function d(e){let{components:t,...i}=e;return(0,n.kt)("wrapper",(0,o.Z)({},p,i,{components:t,mdxType:"MDXLayout"}),(0,n.kt)("link",{rel:"stylesheet",href:"//maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css"}),(0,n.kt)("p",null,(0,n.kt)("em",{parentName:"p"},"Generating four-color plots to compare positional sequence content across genomic sites")),(0,n.kt)("p",null,(0,n.kt)("strong",{parentName:"p"},"Goal:")," This tutorial provides a guide to generating a four-color plot using the ScriptManager platform and the data generated by the Yeast Epigenome project. These plots are especially great for showing the binding motifs within the ChIP-exo peaks of sequence-specific transcription factors."),(0,n.kt)("h2",{id:"download-scriptmanager-v014"},"Download ScriptManager (v0.14):"),(0,n.kt)("p",null,"The current version of ScriptManager is available for download ",(0,n.kt)("strong",{parentName:"p"},(0,n.kt)("a",{parentName:"strong",href:"https://github.com/CEGRcode/scriptmanager/releases/download/v0.14/ScriptManager-v0.14.jar"},"here")),". Make sure you have ",(0,n.kt)("a",{parentName:"p",href:"/docs/#java"},"Java installed"),"."),(0,n.kt)("p",null,"The file ",(0,n.kt)("inlineCode",{parentName:"p"},"ScriptManager-v0.14.jar")," should be placed someplace locally accessible. For example on Mac OS on the Desktop (Permissions will need to be accepted) or someplace in your home directory (i.e. Macintosh HD/Users/userID/ScriptManager)"),(0,n.kt)("h2",{id:"download-data-files"},"Download data files"),(0,n.kt)("ul",null,(0,n.kt)("li",{parentName:"ul"},"You need the full reference genome(FASTA) and the Reb1 peak coordinates (BED) to complete this exercise.",(0,n.kt)("ul",{parentName:"li"},(0,n.kt)("li",{parentName:"ul"},(0,n.kt)("a",{parentName:"li",href:"https://github.com/CEGRcode/2018-Rossi_GenomeResearch/blob/master/Fig1_Reb1/A.Reb1_Rhee_primary_sites_975.bed"},"Sample BED file for Reb1")),(0,n.kt)("li",{parentName:"ul"},"Reference Yeast Genome - sacCer3 build (see instructions below)")))),(0,n.kt)("admonition",{type:"warning"},(0,n.kt)("p",{parentName:"admonition"},"\ud83d\udea7 \ud83d\udc77\u200d\u2640\ufe0f ",(0,n.kt)("strong",{parentName:"p"},"UNDER CONSTRUCTION")," \ud83d\udc77\u200d \ud83d\udea7"),(0,n.kt)("p",{parentName:"admonition"},"Turn ",(0,n.kt)("a",{parentName:"p",href:"https://github.com/CEGRcode/GenoPipe/blob/master/EpitopeID/utility_scripts/genome_data/download_sacCer3_Genome.sh"},"this script")," into an easier-to-run way to get the reference genome with arabic numerals. Your own sacCer3.fa genome should work in this tutorial if you use the ",(0,n.kt)("inlineCode",{parentName:"p"},"chr1 chr2 chr3 ...")," naming system, not the ",(0,n.kt)("inlineCode",{parentName:"p"},"chrI chrII chrIII ..."),".")),(0,n.kt)("admonition",{type:"caution"},(0,n.kt)("p",{parentName:"admonition"},"If your BED file downloads with a ",(0,n.kt)("inlineCode",{parentName:"p"},".txt")," extension, make sure to change the filename to a ",(0,n.kt)("inlineCode",{parentName:"p"},".bed")," extension. For this tutorial, the BED file is named ",(0,n.kt)("inlineCode",{parentName:"p"},"Reb1_Rhee_primary_sites_975.bed"),".")),(0,n.kt)("h2",{id:"generate-the-four-color-plot"},"Generate the Four-color Plot"),(0,n.kt)("h3",{id:"1-open-scriptmanager"},"1. Open ScriptManager"),(0,n.kt)("p",null,"Depending on your system permissions, you may need to be an administrator to open this for the first time."),(0,n.kt)("ul",null,(0,n.kt)("li",{parentName:"ul"},"On Mac systems, this can be done by right-clicking the file and selecting \u2018Open\u2019 at the top.")),(0,n.kt)("div",{class:"tutorial-img-flow-container"},(0,n.kt)("img",{src:a(2140).Z,style:{width:"40%"}})),(0,n.kt)("admonition",{type:"caution"},(0,n.kt)("p",{parentName:"admonition"},"Some of the newer MacOS systems may not properly open the JAR by simply double-clicking on the JAR file so you may need to open your Terminal window and execute it from the command line by typing"),(0,n.kt)("pre",{parentName:"admonition"},(0,n.kt)("code",{parentName:"pre"},"java -jar /path/to/ScriptManager.jar\n")),(0,n.kt)("p",{parentName:"admonition"},"If you're not sure how to type the path to ScriptManager, you can type ",(0,n.kt)("inlineCode",{parentName:"p"},"java -jar ")," (end with space) and then drag ScriptManager from Finder into your Terminal window and then press enter.")),(0,n.kt)("h3",{id:"2-resize-the-reb1-motif-aligned-bed-file"},"2. Resize the Reb1 motif-aligned BED file"),(0,n.kt)("p",null,"The BED file is the set of reference coordinates that your heatmap and composite plots will be aligned to, but you\u2019ll need to specify how far upstream and downstream you want your data to be plotted; i.e., \u201cSize of Expansion (bp). If you bed file is defined by more than a 1 bp interval AND you want to add to limits of that interval, then select  \u201cAdd to Border\u201d)."),(0,n.kt)("h4",{id:"21-navigate-to-coordinate-file-manipulation-\ufe0f-expand-bed-file"},"2.1. Navigate to Coordinate File Manipulation \u27a1\ufe0f ",(0,n.kt)("a",{parentName:"h4",href:"/scriptmanager-docs/docs/coordinate-manipulation/expand-bed"},(0,n.kt)("strong",{parentName:"a"},"Expand BED File"))),(0,n.kt)("div",{class:"tutorial-img-flow-container"},(0,n.kt)("img",{src:a(6386).Z,style:{width:"60%"}})),(0,n.kt)("h4",{id:"22-for-this-tutorial-use-the-50bp-expansion-and-select-expand-from-center"},'2.2. For this tutorial, use the 50bp expansion and select "Expand from Center".'),(0,n.kt)("div",{class:"tutorial-img-flow-container"},(0,n.kt)("img",{src:a(3397).Z,style:{width:"60%"}}),(0,n.kt)("i",{class:"fa fa-arrow-right fa-lg"}),(0,n.kt)("img",{src:a(4567).Z,style:{width:"30%"}})),(0,n.kt)("admonition",{type:"tip"},(0,n.kt)("p",{parentName:"admonition"},"BED file coordinates often need to be resized for more informative tag pileups. As a factor that binds a short motif, Reb1 does not require a large window size to visualize the motif sequence. In fact, a wider window will make it harder to visualize the stripes of color around the motif.")),(0,n.kt)("h3",{id:"3-generate-the-fasta-sequence-input"},"3. Generate the FASTA sequence input"),(0,n.kt)("h4",{id:"31-navigate-to-dna-sequence-analysis-\ufe0f-fasta-from-bed-to-create-the-input-for-generating-a-four-color-plot"},"3.1. Navigate to DNA Sequence Analysis \u27a1\ufe0f ",(0,n.kt)("a",{parentName:"h4",href:"/scriptmanager-docs/docs/sequence-analysis/fasta-extract"},(0,n.kt)("strong",{parentName:"a"},"FASTA from BED"))," to create the input for generating a Four Color plot."),(0,n.kt)("div",{class:"tutorial-img-flow-container"},(0,n.kt)("img",{src:a(7688).Z,style:{width:"60%"}})),(0,n.kt)("h4",{id:"32-load-the-fasta-file-containing-the-genome-fasta"},"3.2. Load the FASTA file containing the Genome FASTA"),(0,n.kt)("ul",null,(0,n.kt)("li",{parentName:"ul"},"A ","*",".fai file will be generated for the genome file the first time it is used. If the Genome FASTA file is NOT in proper FASTA format the script will fail")),(0,n.kt)("h4",{id:"33-load-appropriate-bed-file-for-sequence-fasta-generation"},"3.3. Load appropriate BED file for sequence FASTA generation."),(0,n.kt)("div",{class:"tutorial-img-flow-container"},(0,n.kt)("img",{src:a(589).Z,style:{width:"60%"}}),(0,n.kt)("i",{class:"fa fa-arrow-right fa-lg"}),(0,n.kt)("img",{src:a(2138).Z,style:{width:"40%"}})),(0,n.kt)("h4",{id:"34-click-calculate-to-start-the-extraction-which-outputs-a-fasta-file"},'3.4. Click "Calculate" to start the extraction which outputs a FASTA file'),(0,n.kt)("h3",{id:"4-generate-the-four-color-sequence-image"},"4. Generate the Four color sequence image"),(0,n.kt)("h4",{id:"41-navigate-to-figure-generation-\ufe0f-four-color-plot-to-generate-the-plot-once-you-have-generated-the-fasta-file-of-the-sequences-within-the-bed-regions"},"4.1. Navigate to Figure Generation \u27a1\ufe0f ",(0,n.kt)("a",{parentName:"h4",href:"/scriptmanager-docs/docs/figure-generation/four-color"},(0,n.kt)("strong",{parentName:"a"},"Four Color Plot"))," to generate the plot once you have generated the FASTA file of the sequences within the BED regions."),(0,n.kt)("div",{class:"tutorial-img-flow-container"},(0,n.kt)("img",{src:a(5315).Z,style:{width:"60%"}})),(0,n.kt)("h4",{id:"42-load-the-fasta-file-containing-the-fasta-sequences-generated-in-step-2"},"4.2. Load the FASTA file containing the FASTA sequences (generated in step 2)."),(0,n.kt)("admonition",{type:"note"},(0,n.kt)("p",{parentName:"admonition"},"At this point you may opt to resize the pixel dimensions of each nucleotide rectangle or customize the colors corresponding to each nucleic acid base.")),(0,n.kt)("h4",{id:"43-click-generate-to-execute-the-script"},'4.3. Click "Generate" to execute the script.'),(0,n.kt)("div",{class:"tutorial-img-flow-container"},(0,n.kt)("img",{src:a(4161).Z,style:{width:"60%"}}),(0,n.kt)("i",{class:"fa fa-arrow-right fa-lg"}),(0,n.kt)("img",{src:a(4679).Z,style:{width:"15%"}})),(0,n.kt)("p",null,"Tah dah! You've made the four color plot! It's kind of tall but you can resize it in your favorite image editing software."),(0,n.kt)("h2",{id:"command-line-shell-script"},"Command-Line shell script"),(0,n.kt)("p",null,"The following shell commands takes a BED file and a FASTA file of the full genomic sequences to generate a four-color sequence plot of the center 50bp nucleotides in each BED coordinate interval."),(0,n.kt)("p",null," and the anticipated OUTPUT basename as environmental variables to derive the corresponding composite plot values and heatmaps. This can serve as a template for you to write out your own workflows as bash scripts that execute command-line style ScriptManager."),(0,n.kt)("pre",null,(0,n.kt)("code",{parentName:"pre",className:"language-bash"},"SCRIPTMANAGER=/path/to/ScriptManager.jar\nBEDFILE=/path/to/Reb1_Rhee_primary_sites_975.bed\nBAMFILE=/path/to/12141_filtered.bam\nOUTPUT=/path/to/myfourcolorplot.png\n\nsamtools index $BAMFILE\njava -jar $SCRIPTMANAGER coordinate-manipulation expand-bed -c 250 $BEDFILE -o BED_250bp.bed\njava -jar $SCRIPTMANAGER read-analysis tag-pileup $BEDFILE $BAMFILE -o $OUTPUT\\_composite.out -M $OUTPUT\\_matrix\njava -jar $SCRIPTMANAGER figure-generation heatmap -p .95 --blue $OUTPUT\\_matrix_sense.cdt -o SENSE.png\njava -jar $SCRIPTMANAGER figure-generation heatmap -p .95 --red $OUTPUT\\_matrix_sense.cdt -o ANTI.png\njava -jar $SCRIPTMANAGER figure-generation merge-heatmap SENSE.png ANTI.png -o $OUTPUT\\_heatmap.png\n\nrm BED_250bp.bed SENSE.png ANTI.png\n\n# Output files:\n#  - /path/to/myoutput_composite.out\n#  - /path/to/myoutput_matrix_sense.cdt\n#  - /path/to/myoutput_matrix_anti.cdt\n#  - /path/to/myoutput_heatmap.png\n")))}d.isMDXComponent=!0},4679:(e,t,a)=>{a.d(t,{Z:()=>o});const o=a.p+"assets/images/Reb1_Rhee_primary_sites_975_50bp-f462ef2800b4567b1c4f67b3e231cc2a.png"},4567:(e,t,a)=>{a.d(t,{Z:()=>o});const o=a.p+"assets/images/gui-expandbed-complete-031750880fd32f9fc0d7002c35bd022d.png"},3397:(e,t,a)=>{a.d(t,{Z:()=>o});const o=a.p+"assets/images/gui-expandbed50-f4f4192ce03a9a5fcb5b6e4f7d92ba97.png"},2138:(e,t,a)=>{a.d(t,{Z:()=>o});const o=a.p+"assets/images/gui-extractfasta-complete-d2a886c116f60c8df4bc432f0f9c3829.png"},589:(e,t,a)=>{a.d(t,{Z:()=>o});const o=a.p+"assets/images/gui-extractfasta-e75ed1a1d8cebfd2e3df015fe4040b34.png"},4161:(e,t,a)=>{a.d(t,{Z:()=>o});const o=a.p+"assets/images/gui-fourcolor-7bad2334ef61cf1ac0128232661591b4.png"},6386:(e,t,a)=>{a.d(t,{Z:()=>o});const o=a.p+"assets/images/maingui-expandbed-c34fe4ec13cf11c4c61fa108f2a8d5a2.png"},7688:(e,t,a)=>{a.d(t,{Z:()=>o});const o=a.p+"assets/images/maingui-extractfasta-8ee7d7bf09fa7a2c91ba40c5600423a5.png"},5315:(e,t,a)=>{a.d(t,{Z:()=>o});const o=a.p+"assets/images/maingui-fourcolor-b8c8e7c12aac0b3ad071819a748cfd27.png"},2140:(e,t,a)=>{a.d(t,{Z:()=>o});const o=a.p+"assets/images/open-sm-5f0508d59e500792ea199946818706f1.png"}}]);