"use strict";(self.webpackChunkscriptmanager_docs=self.webpackChunkscriptmanager_docs||[]).push([[7264],{3905:(e,t,a)=>{a.d(t,{Zo:()=>p,kt:()=>m});var n=a(7294);function i(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function o(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function r(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?o(Object(a),!0).forEach((function(t){i(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):o(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function s(e,t){if(null==e)return{};var a,n,i=function(e,t){if(null==e)return{};var a,n,i={},o=Object.keys(e);for(n=0;n<o.length;n++)a=o[n],t.indexOf(a)>=0||(i[a]=e[a]);return i}(e,t);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);for(n=0;n<o.length;n++)a=o[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(i[a]=e[a])}return i}var l=n.createContext({}),d=function(e){var t=n.useContext(l),a=t;return e&&(a="function"==typeof e?e(t):r(r({},t),e)),a},p=function(e){var t=d(e.components);return n.createElement(l.Provider,{value:t},e.children)},c="mdxType",u={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},h=n.forwardRef((function(e,t){var a=e.components,i=e.mdxType,o=e.originalType,l=e.parentName,p=s(e,["components","mdxType","originalType","parentName"]),c=d(a),h=i,m=c["".concat(l,".").concat(h)]||c[h]||u[h]||o;return a?n.createElement(m,r(r({ref:t},p),{},{components:a})):n.createElement(m,r({ref:t},p))}));function m(e,t){var a=arguments,i=t&&t.mdxType;if("string"==typeof e||i){var o=a.length,r=new Array(o);r[0]=h;var s={};for(var l in t)hasOwnProperty.call(t,l)&&(s[l]=t[l]);s.originalType=e,s[c]="string"==typeof e?e:i,r[1]=s;for(var d=2;d<o;d++)r[d]=a[d];return n.createElement.apply(null,r)}return n.createElement.apply(null,a)}h.displayName="MDXCreateElement"},5162:(e,t,a)=>{a.d(t,{Z:()=>r});var n=a(7294),i=a(6010);const o={tabItem:"tabItem_Ymn6"};function r(e){let{children:t,hidden:a,className:r}=e;return n.createElement("div",{role:"tabpanel",className:(0,i.Z)(o.tabItem,r),hidden:a},t)}},4866:(e,t,a)=>{a.d(t,{Z:()=>w});var n=a(7462),i=a(7294),o=a(6010),r=a(2466),s=a(6550),l=a(1980),d=a(7392),p=a(12);function c(e){return function(e){return i.Children.map(e,(e=>{if((0,i.isValidElement)(e)&&"value"in e.props)return e;throw new Error(`Docusaurus error: Bad <Tabs> child <${"string"==typeof e.type?e.type:e.type.name}>: all children of the <Tabs> component should be <TabItem>, and every <TabItem> should have a unique "value" prop.`)}))}(e).map((e=>{let{props:{value:t,label:a,attributes:n,default:i}}=e;return{value:t,label:a,attributes:n,default:i}}))}function u(e){const{values:t,children:a}=e;return(0,i.useMemo)((()=>{const e=t??c(a);return function(e){const t=(0,d.l)(e,((e,t)=>e.value===t.value));if(t.length>0)throw new Error(`Docusaurus error: Duplicate values "${t.map((e=>e.value)).join(", ")}" found in <Tabs>. Every value needs to be unique.`)}(e),e}),[t,a])}function h(e){let{value:t,tabValues:a}=e;return a.some((e=>e.value===t))}function m(e){let{queryString:t=!1,groupId:a}=e;const n=(0,s.k6)(),o=function(e){let{queryString:t=!1,groupId:a}=e;if("string"==typeof t)return t;if(!1===t)return null;if(!0===t&&!a)throw new Error('Docusaurus error: The <Tabs> component groupId prop is required if queryString=true, because this value is used as the search param name. You can also provide an explicit value such as queryString="my-search-param".');return a??null}({queryString:t,groupId:a});return[(0,l._X)(o),(0,i.useCallback)((e=>{if(!o)return;const t=new URLSearchParams(n.location.search);t.set(o,e),n.replace({...n.location,search:t.toString()})}),[o,n])]}function g(e){const{defaultValue:t,queryString:a=!1,groupId:n}=e,o=u(e),[r,s]=(0,i.useState)((()=>function(e){let{defaultValue:t,tabValues:a}=e;if(0===a.length)throw new Error("Docusaurus error: the <Tabs> component requires at least one <TabItem> children component");if(t){if(!h({value:t,tabValues:a}))throw new Error(`Docusaurus error: The <Tabs> has a defaultValue "${t}" but none of its children has the corresponding value. Available values are: ${a.map((e=>e.value)).join(", ")}. If you intend to show no default tab, use defaultValue={null} instead.`);return t}const n=a.find((e=>e.default))??a[0];if(!n)throw new Error("Unexpected error: 0 tabValues");return n.value}({defaultValue:t,tabValues:o}))),[l,d]=m({queryString:a,groupId:n}),[c,g]=function(e){let{groupId:t}=e;const a=function(e){return e?`docusaurus.tab.${e}`:null}(t),[n,o]=(0,p.Nk)(a);return[n,(0,i.useCallback)((e=>{a&&o.set(e)}),[a,o])]}({groupId:n}),f=(()=>{const e=l??c;return h({value:e,tabValues:o})?e:null})();(0,i.useLayoutEffect)((()=>{f&&s(f)}),[f]);return{selectedValue:r,selectValue:(0,i.useCallback)((e=>{if(!h({value:e,tabValues:o}))throw new Error(`Can't select invalid tab value=${e}`);s(e),d(e),g(e)}),[d,g,o]),tabValues:o}}var f=a(2389);const b={tabList:"tabList__CuJ",tabItem:"tabItem_LNqP"};function k(e){let{className:t,block:a,selectedValue:s,selectValue:l,tabValues:d}=e;const p=[],{blockElementScrollPositionUntilNextRender:c}=(0,r.o5)(),u=e=>{const t=e.currentTarget,a=p.indexOf(t),n=d[a].value;n!==s&&(c(t),l(n))},h=e=>{let t=null;switch(e.key){case"Enter":u(e);break;case"ArrowRight":{const a=p.indexOf(e.currentTarget)+1;t=p[a]??p[0];break}case"ArrowLeft":{const a=p.indexOf(e.currentTarget)-1;t=p[a]??p[p.length-1];break}}t?.focus()};return i.createElement("ul",{role:"tablist","aria-orientation":"horizontal",className:(0,o.Z)("tabs",{"tabs--block":a},t)},d.map((e=>{let{value:t,label:a,attributes:r}=e;return i.createElement("li",(0,n.Z)({role:"tab",tabIndex:s===t?0:-1,"aria-selected":s===t,key:t,ref:e=>p.push(e),onKeyDown:h,onClick:u},r,{className:(0,o.Z)("tabs__item",b.tabItem,r?.className,{"tabs__item--active":s===t})}),a??t)})))}function y(e){let{lazy:t,children:a,selectedValue:n}=e;if(a=Array.isArray(a)?a:[a],t){const e=a.find((e=>e.props.value===n));return e?(0,i.cloneElement)(e,{className:"margin-top--md"}):null}return i.createElement("div",{className:"margin-top--md"},a.map(((e,t)=>(0,i.cloneElement)(e,{key:t,hidden:e.props.value!==n}))))}function v(e){const t=g(e);return i.createElement("div",{className:(0,o.Z)("tabs-container",b.tabList)},i.createElement(k,(0,n.Z)({},e,t)),i.createElement(y,(0,n.Z)({},e,t)))}function w(e){const t=(0,f.Z)();return i.createElement(v,(0,n.Z)({key:String(t)},e))}},7371:(e,t,a)=>{a.r(t),a.d(t,{assets:()=>c,contentTitle:()=>d,default:()=>g,frontMatter:()=>l,metadata:()=>p,toc:()=>u});var n=a(7462),i=(a(7294),a(3905)),o=a(4866),r=a(5162),s=a(9960);const l={id:"chipexo-tutorial",title:"ChIP-exo Tutorial",sidebar_label:"ChIP-exo"},d=void 0,p={unversionedId:"Tutorials/chipexo-tutorial",id:"Tutorials/chipexo-tutorial",title:"ChIP-exo Tutorial",description:"Generating two basic sequence-specific ChIP-exo plots: composite and heatmap",source:"@site/docs/Tutorials/chipexo-tutorial.md",sourceDirName:"Tutorials",slug:"/Tutorials/chipexo-tutorial",permalink:"/scriptmanager-docs/docs/Tutorials/chipexo-tutorial",draft:!1,tags:[],version:"current",lastUpdatedBy:"Olivia Wen-Mei Lang",lastUpdatedAt:1680027051,formattedLastUpdatedAt:"Mar 28, 2023",frontMatter:{id:"chipexo-tutorial",title:"ChIP-exo Tutorial",sidebar_label:"ChIP-exo"},sidebar:"someSidebar",previous:{title:"Color Selection Guide",permalink:"/scriptmanager-docs/docs/Guides/color-guide"},next:{title:"Genomic Features",permalink:"/scriptmanager-docs/docs/Tutorials/genomic-features-tutorial"}},c={},u=[{value:"Download ScriptManager (v0.14):",id:"download-scriptmanager-v014",level:2},{value:"Download Data",id:"download-data",level:2},{value:"BED File",id:"bed-file",level:3},{value:"BAM File",id:"bam-file",level:3},{value:"Generate the Plots",id:"generate-the-plots",level:2},{value:"1. Open ScriptManager",id:"1-open-scriptmanager",level:3},{value:"2. Generate BAI index file",id:"2-generate-bai-index-file",level:3},{value:"2.1. Navigate to BAM Manipulation \u27a1\ufe0f BAM-BAI Indexer",id:"21-navigate-to-bam-manipulation-\ufe0f-bam-bai-indexer",level:4},{value:"2.2. Generate BAI index files for each BAM file of interest by loading your BAM file and clicking &quot;Index.&quot;",id:"22-generate-bai-index-files-for-each-bam-file-of-interest-by-loading-your-bam-file-and-clicking-index",level:4},{value:"3. Resize the Reb1 motif-aligned BED file",id:"3-resize-the-reb1-motif-aligned-bed-file",level:3},{value:"3.1. Navigate to Coordinate File Manipulation \u27a1\ufe0f Expand BED File",id:"31-navigate-to-coordinate-file-manipulation-\ufe0f-expand-bed-file",level:4},{value:"3.2. For this tutorial, use the 250bp expansion and select &quot;Expand from Center&quot;.",id:"32-for-this-tutorial-use-the-250bp-expansion-and-select-expand-from-center",level:4},{value:"4. Generate the tag pileup",id:"4-generate-the-tag-pileup",level:3},{value:"4.1. Navigate to Sequence Read Analysis \u27a1\ufe0f Tag Pileup",id:"41-navigate-to-sequence-read-analysis-\ufe0f-tag-pileup",level:4},{value:"4.2. Load the BED and BAM files",id:"42-load-the-bed-and-bam-files",level:4},{value:"4.3. Select output directory &amp; make sure &quot;Output Matrix&quot;, &quot;CDT&quot;, and &quot;Output GZIP&quot; are selected",id:"43-select-output-directory--make-sure-output-matrix-cdt-and-output-gzip-are-selected",level:4},{value:"4.4. When ready, select \u2018Pile Tags\u2019 to execute",id:"44-when-ready-select-pile-tags-to-execute",level:4},{value:"4.5. Save composite results",id:"45-save-composite-results",level:4},{value:"5. Generate Heatmaps",id:"5-generate-heatmaps",level:3},{value:"5.1. Navigate to Figure Generation \u27a1\ufe0f Heat Map.",id:"51-navigate-to-figure-generation-\ufe0f-heat-map",level:4},{value:"5.2. Heatmap Generator can only generate one color at a time, so \u2018Sense\u2019 and \u2018Anti\u2019 files should be processed separately. The ChIP-exo standard for strand colors is \u2018Sense\u2019 == blue and \u2018Anti\u2019 == red.",id:"52-heatmap-generator-can-only-generate-one-color-at-a-time-so-sense-and-anti-files-should-be-processed-separately-the-chip-exo-standard-for-strand-colors-is-sense--blue-and-anti--red",level:4},{value:"5.3. Click &quot;Load Files&quot; and select the <code>_sense.cdt</code> output CDT files from running the TagPileup step.",id:"53-click-load-files-and-select-the-_sensecdt-output-cdt-files-from-running-the-tagpileup-step",level:4},{value:"5.4. Select &quot;blue&quot; for the color (per the lab standard)",id:"54-select-blue-for-the-color-per-the-lab-standard",level:4},{value:"5.5. The heatmaps for this dataset show the best contrast when using the default &quot;Percentile Threshold&quot; value (.95 or 95%).  Otherwise your heatmaps will come out too light or too dark to see the shape.",id:"55-the-heatmaps-for-this-dataset-show-the-best-contrast-when-using-the-default-percentile-threshold-value-95-or-95--otherwise-your-heatmaps-will-come-out-too-light-or-too-dark-to-see-the-shape",level:4},{value:"5.6. Click the &quot;Output Heatmap&quot; checkbox. The Heatmap generator does not save the produced PNG by default.",id:"56-click-the-output-heatmap-checkbox-the-heatmap-generator-does-not-save-the-produced-png-by-default",level:4},{value:"5.7. Click &quot;Generate&quot; to save your Sense PNG heatmap!",id:"57-click-generate-to-save-your-sense-png-heatmap",level:4},{value:"5.8. Remove the &quot;Sense&quot; file by selecting it to highlight it and then clicking the &quot;Remove Files button&quot;",id:"58-remove-the-sense-file-by-selecting-it-to-highlight-it-and-then-clicking-the-remove-files-button",level:4},{value:"5.9. Click &quot;Load Files&quot; and select the <code>_anti.cdt</code> output CDT files from running the TagPileup step.",id:"59-click-load-files-and-select-the-_anticdt-output-cdt-files-from-running-the-tagpileup-step",level:4},{value:"5.10. Select &quot;red&quot; for the color (per the lab standard)",id:"510-select-red-for-the-color-per-the-lab-standard",level:4},{value:"5.11. Since all the other parameters should be the same from you &quot;Sense&quot; run, you shouldn&#39;t need to re-select the &quot;Percentile Threshold&quot; or &quot;Output Heatmap&quot; options.",id:"511-since-all-the-other-parameters-should-be-the-same-from-you-sense-run-you-shouldnt-need-to-re-select-the-percentile-threshold-or-output-heatmap-options",level:4},{value:"5.12. Click &quot;Generate&quot; to save you Anti PNG heatmap!",id:"512-click-generate-to-save-you-anti-png-heatmap",level:4},{value:"6. Merge strand-separated heatmaps",id:"6-merge-strand-separated-heatmaps",level:3},{value:"6.1. Navigate to Figure Generation \u27a1\ufe0f Merge Heatmaps so we can merge our strand-separated heatmaps into a single PNG.",id:"61-navigate-to-figure-generation-\ufe0f-merge-heatmaps-so-we-can-merge-our-strand-separated-heatmaps-into-a-single-png",level:4},{value:"6.2. Click &quot;Load PNG Files&quot; to select the two files output by the HeatMap tool (sense and anti) in the last step",id:"62-click-load-png-files-to-select-the-two-files-output-by-the-heatmap-tool-sense-and-anti-in-the-last-step",level:4},{value:"6.3. Click &quot;Generate&quot; to merge the PNG files into the same heatmap",id:"63-click-generate-to-merge-the-png-files-into-the-same-heatmap",level:4},{value:"7. Label your merged heatmaps",id:"7-label-your-merged-heatmaps",level:3},{value:"7.1. Navigate to Figure Generation \u27a1\ufe0f Label Heatmaps so we can add axes to the heatmap",id:"71-navigate-to-figure-generation-\ufe0f-label-heatmaps-so-we-can-add-axes-to-the-heatmap",level:4},{value:"7.2. Optionally type text into the fields to annotate your axes (Or leave them empty to just generate an SVG border)",id:"72-optionally-type-text-into-the-fields-to-annotate-your-axes-or-leave-them-empty-to-just-generate-an-svg-border",level:4},{value:"7.3. Click &quot;Generate&quot; to create an SVG file that labels your PNG",id:"73-click-generate-to-create-an-svg-file-that-labels-your-png",level:4},{value:"General Comments",id:"general-comments",level:2},{value:"Command-Line shell script",id:"command-line-shell-script",level:2}],h={toc:u},m="wrapper";function g(e){let{components:t,...l}=e;return(0,i.kt)(m,(0,n.Z)({},h,l,{components:t,mdxType:"MDXLayout"}),(0,i.kt)("p",null,(0,i.kt)("em",{parentName:"p"},"Generating two basic sequence-specific ChIP-exo plots: composite and heatmap")),(0,i.kt)("p",null,(0,i.kt)("strong",{parentName:"p"},"Goal:")," This tutorial provides a guide to generating 2 basic plots (Composite plot and Heatmap) using the ScriptManager platform and data generated by the ",(0,i.kt)("a",{parentName:"p",href:"http://www.yeastepigenome.org/"},"Yeast Epigenome project"),"."),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(480).Z,style:{width:"20%"}}),(0,i.kt)("img",{src:a(5307).Z,style:{width:"70%"}})),(0,i.kt)("h2",{id:"download-scriptmanager-v014"},"Download ScriptManager (v0.14):"),(0,i.kt)("p",null,"The current version of ScriptManager is available for download ",(0,i.kt)("strong",{parentName:"p"},(0,i.kt)("a",{parentName:"strong",href:"https://github.com/CEGRcode/scriptmanager/releases/download/v0.14/ScriptManager-v0.14.jar"},"here")),". Make sure you have ",(0,i.kt)("a",{parentName:"p",href:"/docs/#java"},"Java installed"),"."),(0,i.kt)("p",null,"The file ",(0,i.kt)("inlineCode",{parentName:"p"},"ScriptManager-v0.14.jar")," should be placed someplace locally accessible. For example on Mac OS on the Desktop (Permissions will need to be accepted) or someplace in your home directory (",(0,i.kt)("inlineCode",{parentName:"p"},"Macintosh_HD:Users/userID/ScriptManager"),")"),(0,i.kt)("h2",{id:"download-data"},"Download Data"),(0,i.kt)("p",null,"You need one set of genomic coordinate regions to investigate (BED) and one file of sequencing data alignments (BAM) to complete this exercise. ",(0,i.kt)("a",{parentName:"p",href:"/docs/References/file-formats"},"Read more about the BED/BAM file formats here.")),(0,i.kt)("h3",{id:"bed-file"},"BED File"),(0,i.kt)("p",null,"This is the set of Reb1 binding sites from the ",(0,i.kt)("a",{parentName:"p",href:"https://pubmed.ncbi.nlm.nih.gov/29563167/"},"Rossi et al (2018)"),"."),(0,i.kt)(s.Z,{className:"button button--secondary",href:"https://github.com/CEGRcode/2018-Rossi_GenomeResearch/blob/master/Fig1_Reb1/A.Reb1_Rhee_primary_sites_975.bed",mdxType:"Link"},"Download sample BED file"),(0,i.kt)("br",null),(0,i.kt)("br",null),(0,i.kt)("admonition",{type:"caution"},(0,i.kt)("p",{parentName:"admonition"},"If your BED file downloads with a ",(0,i.kt)("inlineCode",{parentName:"p"},".txt")," extension, make sure to change the filename to a ",(0,i.kt)("inlineCode",{parentName:"p"},".bed")," extension. For this tutorial, the BED file is named ",(0,i.kt)("inlineCode",{parentName:"p"},"Reb1_Rhee_primary_sites_975.bed"),".")),(0,i.kt)("h3",{id:"bam-file"},"BAM File"),(0,i.kt)("p",null,"This is the set of Reb1 read alignments from the Yeast Epigenome Project (YEP). ",(0,i.kt)("a",{parentName:"p",href:"https://pubmed.ncbi.nlm.nih.gov/33692541/"},"See Rossi et al (2021)")," for more details."),(0,i.kt)(s.Z,{className:"button button--secondary",href:"ftp://data1.commons.psu.edu/pub/commons/eberly/pughlab/yeast-epigenome-project/12141_YEP.zip",mdxType:"Link"},"Download sample BAM file"),(0,i.kt)("p",null,"OR"),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(5553).Z,style:{width:"60%"}})),(0,i.kt)("ol",null,(0,i.kt)("li",{parentName:"ol"},"Navigate to ",(0,i.kt)("a",{parentName:"li",href:"http://www.yeastepigenome.org/"},"www.yeastepigenome.org")," and search for Reb1"),(0,i.kt)("li",{parentName:"ol"},'Select "META DATA"'),(0,i.kt)("li",{parentName:"ol"},'Select "Direct Download"'),(0,i.kt)("li",{parentName:"ol"},"Unzip the resulting file \u201812141_YEP.zip\u2019 and inspect the contents of the new ",(0,i.kt)("inlineCode",{parentName:"li"},"12141_YEP")," folder. It should contain a file called ",(0,i.kt)("inlineCode",{parentName:"li"},"12141_filtered.bam"),".")),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(7087).Z,style:{width:"50%"}}),(0,i.kt)("img",{src:a(756).Z,style:{width:"50%"}})),(0,i.kt)("h2",{id:"generate-the-plots"},"Generate the Plots"),(0,i.kt)("h3",{id:"1-open-scriptmanager"},"1. Open ScriptManager"),(0,i.kt)(o.Z,{mdxType:"Tabs"},(0,i.kt)(r.Z,{value:"macos",label:"MacOS",default:!0,mdxType:"TabItem"},(0,i.kt)("p",null,"Depending on your system permissions, you may need to be an administrator to open this for the first time. On Mac systems, this can be done by right-clicking the file and selecting \u2018Open\u2019 at the top."),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(3002).Z,style:{width:"40%",border:"solid 1px"}})),(0,i.kt)("admonition",{type:"caution"},(0,i.kt)("p",{parentName:"admonition"},"Some MacOS systems may not properly open the JAR by simply double-clicking on the JAR file so you may need to open your Terminal window and execute it from the command line by executing the jar file ",(0,i.kt)("em",{parentName:"p"},"without any arguments or flags"),":"),(0,i.kt)("pre",{parentName:"admonition"},(0,i.kt)("code",{parentName:"pre"},"java -jar /path/to/ScriptManager.jar\n")),(0,i.kt)("p",{parentName:"admonition"},"If you're not sure how to type the path to ScriptManager, you can type ",(0,i.kt)("inlineCode",{parentName:"p"},"java -jar ")," (end with space) and then drag ScriptManager from Finder into your Terminal window and then press enter."))),(0,i.kt)(r.Z,{value:"linux",label:"Linux",mdxType:"TabItem"},(0,i.kt)("p",null,"Double-click or right-click the ScriptManager JAR file to start the program."),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(4152).Z,style:{width:"40%",border:"solid 1px"}}))),(0,i.kt)(r.Z,{value:"windows",label:"Windows",mdxType:"TabItem"},(0,i.kt)("p",null,"Double-click or right-click the ScriptManager JAR file to start the program."),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(1745).Z,style:{width:"40%",border:"solid 1px"}})))),(0,i.kt)("p",null,"Once you see the main tool selection window, you're off to the races!"),(0,i.kt)("h3",{id:"2-generate-bai-index-file"},"2. Generate BAI index file"),(0,i.kt)("p",null,"A BAI index file is required for each BAM file of interest (i.e., the tag occupancy data you want to plot). This file allows for rapid access of the sorted and aligned sequence reads (BAM file)."),(0,i.kt)("admonition",{type:"tip"},(0,i.kt)("p",{parentName:"admonition"},"SAM/BAM standard is to keep BAI file in same directory as BAM file with the ScriptManager-generated filename."),(0,i.kt)("pre",{parentName:"admonition"},(0,i.kt)("code",{parentName:"pre",className:"language-bash"},"Reb1_YEP_12141.bam\nReb1_YEP_12141.bam.bai  # Need to generate this file to proceed.\n"))),(0,i.kt)("h4",{id:"21-navigate-to-bam-manipulation-\ufe0f-bam-bai-indexer"},"2.1. Navigate to BAM Manipulation \u27a1\ufe0f ",(0,i.kt)("a",{parentName:"h4",href:"/docs/bam-manipulation/bam-indexer"},(0,i.kt)("strong",{parentName:"a"},"BAM-BAI Indexer"))),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(3257).Z,style:{width:"60%"}})),(0,i.kt)("h4",{id:"22-generate-bai-index-files-for-each-bam-file-of-interest-by-loading-your-bam-file-and-clicking-index"},'2.2. Generate BAI index files for each BAM file of interest by loading your BAM file and clicking "Index."'),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(5117).Z,style:{width:"60%"}}),(0,i.kt)("i",{class:"fa-solid fa-arrow-right fa-lg"}),(0,i.kt)("img",{src:a(6608).Z,style:{width:"30%"}})),(0,i.kt)("admonition",{type:"note"},(0,i.kt)("p",{parentName:"admonition"},"The speed of this step scales with the size of the BAM file. Generally this step 30 sec for a 100 MB BAM file but may take 1-2 min for a multi-GB BAM file.")),(0,i.kt)("h3",{id:"3-resize-the-reb1-motif-aligned-bed-file"},"3. Resize the Reb1 motif-aligned BED file"),(0,i.kt)("p",null,"The BED file is the set of reference coordinates that your heatmap and composite plots will be aligned to, but you\u2019ll need to specify how far upstream and downstream you want your data to be plotted; i.e., \u201cSize of Expansion (bp). If you bed file is defined by more than a 1 bp interval AND you want to add to limits of that interval, then select  \u201cAdd to Border\u201d)."),(0,i.kt)("h4",{id:"31-navigate-to-coordinate-file-manipulation-\ufe0f-expand-bed-file"},"3.1. Navigate to Coordinate File Manipulation \u27a1\ufe0f ",(0,i.kt)("a",{parentName:"h4",href:"/docs/coordinate-manipulation/expand-bed"},(0,i.kt)("strong",{parentName:"a"},"Expand BED File"))),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(6386).Z,style:{width:"60%"}})),(0,i.kt)("h4",{id:"32-for-this-tutorial-use-the-250bp-expansion-and-select-expand-from-center"},'3.2. For this tutorial, use the 250bp expansion and select "Expand from Center".'),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(2081).Z,style:{width:"60%"}}),(0,i.kt)("i",{class:"fa-solid fa-arrow-right fa-lg"}),(0,i.kt)("img",{src:a(4567).Z,style:{width:"30%"}})),(0,i.kt)("admonition",{type:"tip"},(0,i.kt)("p",{parentName:"admonition"},"BED file coordinates often need to be resized for more informative tag pileups. For Reb1 (yeast), 250-500 bp windows are generally sufficient. Mammalian samples may require larger windows (500-2000 bp) based on the amount of indirect-crosslinking")),(0,i.kt)("h3",{id:"4-generate-the-tag-pileup"},"4. Generate the tag pileup"),(0,i.kt)("p",null,"Use TagPileup to pileup the BAM data within a set of BED coordinate windows to generate the composite plot and the matrix(CDT) files that will be used to generate the heatmaps."),(0,i.kt)("h4",{id:"41-navigate-to-sequence-read-analysis-\ufe0f-tag-pileup"},"4.1. Navigate to Sequence Read Analysis \u27a1\ufe0f ",(0,i.kt)("a",{parentName:"h4",href:"/docs/read-analysis/tag-pileup"},(0,i.kt)("strong",{parentName:"a"},"Tag Pileup"))),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(3823).Z,style:{width:"60%"}})),(0,i.kt)("h4",{id:"42-load-the-bed-and-bam-files"},"4.2. Load the BED and BAM files"),(0,i.kt)("h4",{id:"43-select-output-directory--make-sure-output-matrix-cdt-and-output-gzip-are-selected"},'4.3. Select output directory & make sure "Output Matrix", "CDT", and "Output GZIP" are selected'),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("p",{parentName:"li"},"Bioinformatic projects should be organized in a uniform and consistent manner as described below")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("p",{parentName:"li"},(0,i.kt)("a",{parentName:"p",href:"https://journals.plos.org/ploscompbiol/article/file?id=10.1371/journal.pcbi.1000424&type=printable"},"Paper on how to organize bioinformatics projects (Noble 2009)")))),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(50).Z,style:{width:"100%"}})),(0,i.kt)("h4",{id:"44-when-ready-select-pile-tags-to-execute"},"4.4. When ready, select \u2018Pile Tags\u2019 to execute"),(0,i.kt)("admonition",{type:"tip"},(0,i.kt)("p",{parentName:"admonition"},"The default parameters Tag Pileup is set to expect is a sequence-specific strand separated ChIP-exo dataset. Modifications to these parameters are needed for more specific analysis or when using data generated from other assays.")),(0,i.kt)("h4",{id:"45-save-composite-results"},"4.5. Save composite results"),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(9).Z,style:{width:"100%"}})),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("p",{parentName:"li"},"The displayed composite plot can be modified by right-clicking and selecting properties. Things such as axis labels, axis range, and colors can be modified here.")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("p",{parentName:"li"},"The final image can then be saved by right-clicking and selecting \u2018Save as\u2019. PNG is fine for most cases, but SVG is strongly recommended if this composite plot will be used in Adobe Illustrator later."))),(0,i.kt)("admonition",{type:"note"},(0,i.kt)("p",{parentName:"admonition"},"Besides the composite plot image, ScriptManager has saved the matrix ","*",".CDT files to your Output Directory together with the composite plot values file (If you didn't change the name it would be called ",(0,i.kt)("inlineCode",{parentName:"p"},"composite_average.out"),"). These CDT files will be used as the input for generating heatmaps in the next step.")),(0,i.kt)("h3",{id:"5-generate-heatmaps"},"5. Generate Heatmaps"),(0,i.kt)("h4",{id:"51-navigate-to-figure-generation-\ufe0f-heat-map"},"5.1. Navigate to Figure Generation \u27a1\ufe0f ",(0,i.kt)("a",{parentName:"h4",href:"/docs/figure-generation/heatmap"},(0,i.kt)("strong",{parentName:"a"},"Heat Map")),"."),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(9729).Z,style:{width:"60%"}})),(0,i.kt)("h4",{id:"52-heatmap-generator-can-only-generate-one-color-at-a-time-so-sense-and-anti-files-should-be-processed-separately-the-chip-exo-standard-for-strand-colors-is-sense--blue-and-anti--red"},"5.2. Heatmap Generator can only generate one color at a time, so \u2018Sense\u2019 and \u2018Anti\u2019 files should be processed separately. The ChIP-exo standard for strand colors is \u2018Sense\u2019 == blue and \u2018Anti\u2019 == red."),(0,i.kt)("p",null,"Start by generating the 'Sense' heatmap first."),(0,i.kt)("h4",{id:"53-click-load-files-and-select-the-_sensecdt-output-cdt-files-from-running-the-tagpileup-step"},'5.3. Click "Load Files" and select the ',(0,i.kt)("inlineCode",{parentName:"h4"},"_sense.cdt")," output CDT files from running the TagPileup step."),(0,i.kt)("h4",{id:"54-select-blue-for-the-color-per-the-lab-standard"},'5.4. Select "blue" for the color (per the lab standard)'),(0,i.kt)("h4",{id:"55-the-heatmaps-for-this-dataset-show-the-best-contrast-when-using-the-default-percentile-threshold-value-95-or-95--otherwise-your-heatmaps-will-come-out-too-light-or-too-dark-to-see-the-shape"},'5.5. The heatmaps for this dataset show the best contrast when using the default "Percentile Threshold" value (.95 or 95%).  Otherwise your heatmaps will come out too light or too dark to see the shape.'),(0,i.kt)("h4",{id:"56-click-the-output-heatmap-checkbox-the-heatmap-generator-does-not-save-the-produced-png-by-default"},'5.6. Click the "Output Heatmap" checkbox. The Heatmap generator does not save the produced PNG by default.'),(0,i.kt)("h4",{id:"57-click-generate-to-save-your-sense-png-heatmap"},'5.7. Click "Generate" to save your Sense PNG heatmap!'),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(3594).Z,style:{width:"40%"}}),(0,i.kt)("i",{class:"fa-solid fa-arrow-right fa-lg"}),(0,i.kt)("img",{src:a(5867).Z,style:{width:"40%"}})),(0,i.kt)("admonition",{type:"tip"},(0,i.kt)("p",{parentName:"admonition"},'"Percentile Threshold" is a useful for looking at the shape of ChIP-exo binding patterns while "Absolute Threshold" is useful for setting a shared contrast threshold across several samples and comparing signal density. For other data, you may need to play around with settings to find the right contrast for you data.')),(0,i.kt)("p",null,"Similarly generate the 'Anti' heatmap."),(0,i.kt)("h4",{id:"58-remove-the-sense-file-by-selecting-it-to-highlight-it-and-then-clicking-the-remove-files-button"},'5.8. Remove the "Sense" file by selecting it to highlight it and then clicking the "Remove Files button"'),(0,i.kt)("h4",{id:"59-click-load-files-and-select-the-_anticdt-output-cdt-files-from-running-the-tagpileup-step"},'5.9. Click "Load Files" and select the ',(0,i.kt)("inlineCode",{parentName:"h4"},"_anti.cdt")," output CDT files from running the TagPileup step."),(0,i.kt)("h4",{id:"510-select-red-for-the-color-per-the-lab-standard"},'5.10. Select "red" for the color (per the lab standard)'),(0,i.kt)("h4",{id:"511-since-all-the-other-parameters-should-be-the-same-from-you-sense-run-you-shouldnt-need-to-re-select-the-percentile-threshold-or-output-heatmap-options"},'5.11. Since all the other parameters should be the same from you "Sense" run, you shouldn\'t need to re-select the "Percentile Threshold" or "Output Heatmap" options.'),(0,i.kt)("h4",{id:"512-click-generate-to-save-you-anti-png-heatmap"},'5.12. Click "Generate" to save you Anti PNG heatmap!'),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(9432).Z,style:{width:"40%"}}),(0,i.kt)("i",{class:"fa-solid fa-arrow-right fa-lg"}),(0,i.kt)("img",{src:a(1276).Z,style:{width:"40%"}})),(0,i.kt)("h3",{id:"6-merge-strand-separated-heatmaps"},"6. Merge strand-separated heatmaps"),(0,i.kt)("h4",{id:"61-navigate-to-figure-generation-\ufe0f-merge-heatmaps-so-we-can-merge-our-strand-separated-heatmaps-into-a-single-png"},"6.1. Navigate to Figure Generation \u27a1\ufe0f ",(0,i.kt)("a",{parentName:"h4",href:"/docs/figure-generation/merge-heatmap"},(0,i.kt)("strong",{parentName:"a"},"Merge Heatmaps"))," so we can merge our strand-separated heatmaps into a single PNG."),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(2827).Z,style:{width:"60%"}})),(0,i.kt)("p",null,"The script will automatically match sense to anti heatmaps using the standardized naming conventions used by ScriptManager"),(0,i.kt)("h4",{id:"62-click-load-png-files-to-select-the-two-files-output-by-the-heatmap-tool-sense-and-anti-in-the-last-step"},'6.2. Click "Load PNG Files" to select the two files output by the HeatMap tool (sense and anti) in the last step'),(0,i.kt)("h4",{id:"63-click-generate-to-merge-the-png-files-into-the-same-heatmap"},'6.3. Click "Generate" to merge the PNG files into the same heatmap'),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(1842).Z,style:{width:"50%"}}),(0,i.kt)("i",{class:"fa-solid fa-arrow-right fa-lg"}),(0,i.kt)("img",{src:a(518).Z,style:{width:"40%"}})),(0,i.kt)("h3",{id:"7-label-your-merged-heatmaps"},"7. Label your merged heatmaps"),(0,i.kt)("h4",{id:"71-navigate-to-figure-generation-\ufe0f-label-heatmaps-so-we-can-add-axes-to-the-heatmap"},"7.1. Navigate to Figure Generation \u27a1\ufe0f ",(0,i.kt)("a",{parentName:"h4",href:"/docs/figure-generation/heatmap-labeler"},(0,i.kt)("strong",{parentName:"a"},"Label Heatmaps"))," so we can add axes to the heatmap"),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(5523).Z,style:{width:"60%"}})),(0,i.kt)("h4",{id:"72-optionally-type-text-into-the-fields-to-annotate-your-axes-or-leave-them-empty-to-just-generate-an-svg-border"},"7.2. Optionally type text into the fields to annotate your axes (Or leave them empty to just generate an SVG border)"),(0,i.kt)("div",{class:"tutorial-img-flow-container"},(0,i.kt)("img",{src:a(3688).Z,style:{width:"40%"}}),(0,i.kt)("i",{class:"fa-solid fa-arrow-right fa-lg"}),(0,i.kt)("img",{src:a(6055).Z,style:{width:"40%"}})),(0,i.kt)("h4",{id:"73-click-generate-to-create-an-svg-file-that-labels-your-png"},'7.3. Click "Generate" to create an SVG file that labels your PNG'),(0,i.kt)("p",null,"Open up your SVG file to see what it looks like! Use PowerPoint, Adobe Illustrator, or wherever you edit your publication figures to make any further edits!"),(0,i.kt)("h2",{id:"general-comments"},"General Comments"),(0,i.kt)("p",null,"Bioinformatic projects should be organized in a uniform and consistent manner so that you can easily find them in the future. Consider the organizational structure described in ",(0,i.kt)("a",{parentName:"p",href:"https://journals.plos.org/ploscompbiol/article/file?id=10.1371/journal.pcbi.1000424&type=printable"},(0,i.kt)("strong",{parentName:"a"},"Noble, 2009")),"."),(0,i.kt)("h2",{id:"command-line-shell-script"},"Command-Line shell script"),(0,i.kt)("p",null,"The following shell commands records the locations for a BED file, a BAM file, and the anticipated OUTPUT basename as environmental variables to derive the corresponding composite plot values and heatmaps. This can serve as a template for you to write out your own workflows as bash scripts that execute command-line style ScriptManager."),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-bash"},'SCRIPTMANAGER=/path/to/ScriptManager.jar\nBEDFILE=/path/to/Reb1_Rhee_primary_sites_975.bed\nBAMFILE=/path/to/12141_filtered.bam\nOUTPUT=/path/to/myoutput\n\nsamtools index $BAMFILE\njava -jar $SCRIPTMANAGER coordinate-manipulation expand-bed -c 250 $BEDFILE -o BED_250bp.bed\njava -jar $SCRIPTMANAGER read-analysis tag-pileup $BEDFILE $BAMFILE -o $OUTPUT\\_composite.out -M $OUTPUT\\_matrix\njava -jar $SCRIPTMANAGER figure-generation heatmap -p .95 --blue $OUTPUT\\_matrix_sense.cdt -o SENSE.png\njava -jar $SCRIPTMANAGER figure-generation heatmap -p .95 --red $OUTPUT\\_matrix_sense.cdt -o ANTI.png\njava -jar $SCRIPTMANAGER figure-generation merge-heatmap SENSE.png ANTI.png -o $OUTPUT\\_heatmap.png\njava -jar $SCRIPTMANAGER figure-generation label-heatmap $OUTPUT\\_heatmap.png \\\n  -x "Reb1" -y "Reb1_Rhee_primary_sites_975" -l -250 -m 0 -r +250 -o $OUTPUT\\_heatmap.svg\n\nrm BED_250bp.bed SENSE.png ANTI.png\n\n# Output files:\n#  - /path/to/myoutput_composite.out\n#  - /path/to/myoutput_matrix_sense.cdt\n#  - /path/to/myoutput_matrix_anti.cdt\n#  - /path/to/myoutput_heatmap.png\n')))}g.isMDXComponent=!0},5553:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/download-yep-home-f975840806150a454e5ec6533fd7fd53.png"},756:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/download-yep-reb1-metadata-ab2aeca9e2417d0c313723d71a34d85b.png"},7087:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/download-yep-reb1-3dd6db352f8dab08f27e1cad8092367f.png"},6608:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/gui-bamidx-complete-e925a25e686f6c6813848dd104f8cfd5.png"},5117:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/gui-bamidx-aa4e45d9e6ea63a6f3bb4a921bfd1ad5.png"},4567:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/gui-expandbed-complete-031750880fd32f9fc0d7002c35bd022d.png"},2081:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/gui-expandbed-0faadcd1af45e62bd9a96280c0094d50.png"},1276:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/gui-heatmap-anti-out-c15ecf9808b3239fc0b72c6cd268aa24.png"},9432:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/gui-heatmap-anti-efc608a8912c1548cac99e6b1fea024e.png"},5867:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/gui-heatmap-sense-out-e317b5a92372f2fd64ffe5ffc796d76a.png"},3594:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/gui-heatmap-sense-acaff40410d3935bec8e20cca749c86a.png"},6055:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/gui-labelhm-dialog-e203db28c23bc3209f9f55c6f1200491.png"},3688:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/gui-labelhm-8003bd7b41dc4bda146d53121c762e69.png"},518:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/gui-mergehm-out-4145e0ec8d9f72b6db23a6b1727be0f8.png"},1842:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/gui-mergehm-f255e0ce11a26571158ee46f4bfb3eb3.png"},9:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/gui-tagpileup-out-799264f9c9d6119ab8036c02dfbaa9d5.png"},50:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/gui-tagpileup-61c2817ab78015cd8bfe5b1e1377955c.png"},3257:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/maingui-bamidx-916174344aa9670cd3fe1bb3f0287a74.png"},6386:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/maingui-expandbed-c34fe4ec13cf11c4c61fa108f2a8d5a2.png"},9729:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/maingui-heatmap-9bd8cf5b3e862579d99b4ffb1b913ca5.png"},5523:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/maingui-labelhm-a88a99ec59698462932fe22fef870bcd.png"},2827:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/maingui-mergehm-a39596c9dc0067a0ff2596d266912ebb.png"},3823:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/maingui-tagpileup-a595f39070628d9d700325cc9d77895d.png"},5307:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/ChIP-exo_composite-1733680a72d93be79fcf1793799c21c8.png"},480:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/ChIP-exo_heatmap-b74eb063d0ee9acbc7574fb04f85839b.png"},4152:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/LinuxOpenSM-1fe13f9ab9c369dd24c0a67c7ee86f41.png"},3002:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/MacOpenSM-5f0508d59e500792ea199946818706f1.png"},1745:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/WindowsOpenSM-ac32d3855e9f0cf632ab34c5c9fc983d.png"}}]);