<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-Guides/tool-group">
<head>
<meta charset="UTF-8">
<meta name="generator" content="Docusaurus v2.3.1">
<title data-rh="true">Tool Groups | ScriptManager</title><meta data-rh="true" name="viewport" content="width=device-width,initial-scale=1"><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://CEGRcode.github.io/scriptmanager-docs/docs/Guides/tool-group"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Tool Groups | ScriptManager"><meta data-rh="true" name="description" content="BAM Format Converter"><meta data-rh="true" property="og:description" content="BAM Format Converter"><link data-rh="true" rel="icon" href="/scriptmanager-docs/img/SM_favicon.png"><link data-rh="true" rel="canonical" href="https://CEGRcode.github.io/scriptmanager-docs/docs/Guides/tool-group"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/scriptmanager-docs/docs/Guides/tool-group" hreflang="en"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/scriptmanager-docs/docs/Guides/tool-group" hreflang="x-default"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.3.0/css/all.min.css" integrity="sha512-SzlrxWUlpfuzQ+pcUCosxcglQRNAq/DZjVsC0lE40xsADsfeQoEypE+enwcOiGjk/bSuGGKHEyjSoQ1zVisanQ==" crossorigin="anonymous"><link rel="stylesheet" href="/scriptmanager-docs/assets/css/styles.93e36019.css">
<link rel="preload" href="/scriptmanager-docs/assets/js/runtime~main.015d9caf.js" as="script">
<link rel="preload" href="/scriptmanager-docs/assets/js/main.510075bb.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region" aria-label="Skip to main content"><a class="skipToContent_fXgn" href="#docusaurus_skipToContent_fallback">Skip to main content</a></div><nav aria-label="Main" class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Toggle navigation bar" aria-expanded="false" class="navbar__toggle clean-btn" type="button"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/scriptmanager-docs/"><div class="navbar__logo"><img src="/scriptmanager-docs/img/SM_favicon.png" alt="ScriptManager Logo" class="themedImage_ToTc themedImage--light_HNdA"><img src="/scriptmanager-docs/img/SM_favicon.png" alt="ScriptManager Logo" class="themedImage_ToTc themedImage--dark_i4oU"></div><b class="navbar__title text--truncate">ScriptManager</b></a><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/scriptmanager-docs/docs/">General</a><a class="navbar__item navbar__link" href="/scriptmanager-docs/docs/read-analysis/tag-pileup">Tools</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/CEGRcode/scriptmanager" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link"></a><div class="toggle_vylO colorModeToggle_DEke"><button class="clean-btn toggleButton_gllP toggleButtonDisabled_aARS" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)" aria-live="polite"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_pyhR"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_wfgR"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_ZlJk"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div id="docusaurus_skipToContent_fallback" class="main-wrapper mainWrapper_z2l0 docsWrapper_BCFX"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_sjWU" type="button"></button><div class="docPage__5DB"><aside class="theme-doc-sidebar-container docSidebarContainer_b6E3"><div class="sidebarViewport_Xe31"><div class="sidebar_njMd"><nav aria-label="Docs sidebar" class="menu thin-scrollbar menu_SIkG"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--sublist-caret menu__link--active" aria-expanded="true" href="/scriptmanager-docs/docs/">Guides</a></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/scriptmanager-docs/docs/">Getting Started</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/scriptmanager-docs/docs/Guides/command-line">Command Line</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/scriptmanager-docs/docs/Guides/tool-group">Tool Groups</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/scriptmanager-docs/docs/Guides/color-guide">Color Selection Guide</a></li></ul></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--sublist-caret" aria-expanded="false" href="/scriptmanager-docs/docs/Tutorials/chipexo-tutorial">Tutorials</a></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--sublist-caret" aria-expanded="false" href="/scriptmanager-docs/docs/References/file-formats">References</a></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--sublist-caret" aria-expanded="false" href="/scriptmanager-docs/docs/Contributing/developer-guidelines">Contributing</a></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--sublist-caret" aria-expanded="false" href="/scriptmanager-docs/docs/Community/code-of-conduct">Community</a></div></li></ul></nav></div></div></aside><main class="docMainContainer_gTbr"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_VOVn"><div class="docItemContainer_Djhp"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Z_bl" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a aria-label="Home page" class="breadcrumbs__link" href="/scriptmanager-docs/"><svg viewBox="0 0 24 24" class="breadcrumbHomeIcon_YNFT"><path d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z" fill="currentColor"></path></svg></a></li><li class="breadcrumbs__item"><span class="breadcrumbs__link">Guides</span><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">Tool Groups</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_ETCw theme-doc-toc-mobile tocMobile_ITEo"><button type="button" class="clean-btn tocCollapsibleButton_TO0P">On this page</button></div><div class="theme-doc-markdown markdown"><header><h1>Tool Groups</h1></header><h2 class="anchor anchorWithStickyNavbar_LWe7" id="bam-format-converter">BAM Format Converter<a href="#bam-format-converter" class="hash-link" aria-label="Direct link to BAM Format Converter" title="Direct link to BAM Format Converter">​</a></h2><p><em>Description of BAM-Format-Converter tool</em></p><table><thead><tr><th>Tool Name</th><th>Input</th><th>Output</th><th>Description</th></tr></thead><tbody><tr><td><a href="/scriptmanager-docs/docs/bam-format-converter/bam-to-scidx"><code>bam-to-scidx</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#scidx">scIDX</a></td><td>Convert BAM file to scIDX file.</td></tr><tr><td><a href="/scriptmanager-docs/docs/bam-format-converter/bam-to-gff"><code>bam-to-gff</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#gff">GFF</a></td><td>Convert BAM file to GFF file.</td></tr><tr><td><a href="/scriptmanager-docs/docs/bam-format-converter/bam-to-bed"><code>bam-to-bed</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bed">BED</a></td><td>Convert BAM file to BED file.</td></tr><tr><td><a href="/scriptmanager-docs/docs/bam-format-converter/bam-to-bedgraph"><code>bam-to-bedgraph</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bedgraph">bedGraph</a></td><td>Convert BAM file to bedGraph file.</td></tr></tbody></table><h2 class="anchor anchorWithStickyNavbar_LWe7" id="bam-manipulation">BAM Manipulation<a href="#bam-manipulation" class="hash-link" aria-label="Direct link to BAM Manipulation" title="Direct link to BAM Manipulation">​</a></h2><p><em>General description of BAM Manipulation</em></p><table><thead><tr><th>Tool Name</th><th>Input</th><th>Output</th><th>Description</th></tr></thead><tbody><tr><td><a href="/scriptmanager-docs/docs/bam-manipulation/bam-indexer"><code>bam-indexer</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAI</a></td><td>Generates BAI Index for input BAM files. Output BAI is in the same directory as input BAM file.</td></tr><tr><td><a href="/scriptmanager-docs/docs/bam-manipulation/sort-bam"><code>sort-bam</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td>Sort BAM files in order to efficiently extract and manipulate.\nRAM intensive process. If program freezes, increase JAVA heap size.</td></tr><tr><td><a href="/scriptmanager-docs/docs/bam-manipulation/remove-duplicates"><code>remove-duplicates</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td>Removes duplicate reads in Paired-End sequencing given identical 5&#x27; read locations. RAM intensive process. If program freezes, increase JAVA heap size.</td></tr><tr><td><a href="/scriptmanager-docs/docs/bam-manipulation/merge-bam"><code>merge-bam</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a>[<!-- -->..<!-- -->]</td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td>Merges Multiple BAM files into single BAM file. Sorting is performed automatically. RAM intensive process. If program freezes, increase JAVA heap size.</td></tr><tr><td><a href="/scriptmanager-docs/docs/bam-manipulation/filter-pip-seq"><code>filter-pip-seq</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td>Filter BAM file by -1 nucleotide. Requires genome FASTA file. Note this program does not index the resulting BAM file and user must use appropriate samtools command to generate BAI.</td></tr></tbody></table><h2 class="anchor anchorWithStickyNavbar_LWe7" id="bam-statistics">BAM Statistics<a href="#bam-statistics" class="hash-link" aria-label="Direct link to BAM Statistics" title="Direct link to BAM Statistics">​</a></h2><table><thead><tr><th>Tool Name</th><th>Input</th><th>Output</th><th>Description</th></tr></thead><tbody><tr><td><a href="/scriptmanager-docs/docs/bam-statistics/se-stat">se-stat</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#txt">TXT</a></td><td>Genome-Genome correlations for replicate comparisons given multiple sorted and indexed (BAI) BAM files.</td></tr><tr><td><a href="/scriptmanager-docs/docs/bam-statistics/pe-stat">pe-stat</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#txt">TXT</a></td><td>Generates Insert-size Histogram statistic (GEO requirement) and outputs BAM Header including alignment statistics and parameters given a sorted and indexed (BAI) paired-end BAM File.</td></tr><tr><td><a href="/scriptmanager-docs/docs/bam-statistics/bam-correlation">bam-correlation</a>bam-correlation)</td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#txt">TXT</a></td><td>Output BAM Header including alignment statistics and parameters given any indexed (BAI) BAM File.</td></tr></tbody></table><h2 class="anchor anchorWithStickyNavbar_LWe7" id="coordinate-manipulation">Coordinate Manipulation<a href="#coordinate-manipulation" class="hash-link" aria-label="Direct link to Coordinate Manipulation" title="Direct link to Coordinate Manipulation">​</a></h2><p><em>Coordinate Manipulation tools description</em></p><table><thead><tr><th>Tool Name</th><th>Input</th><th>Output</th><th>Description</th></tr></thead><tbody><tr><td><a href="/scriptmanager-docs/docs/coordinate-manipulation/expand-bed">expand-bed</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bed">BED</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bed">BED</a></td><td>Expands input BED file by adding positions to the border or around the center.</td></tr><tr><td><a href="/scriptmanager-docs/docs/coordinate-manipulation/expand-gff">expand-gff</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#gff">GFF</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#gff">GFF</a></td><td>Expands input GFF file by adding positions to the border or around the center.</td></tr><tr><td><a href="/scriptmanager-docs/docs/coordinate-manipulation/bed-to-gff">bed-to-gff</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bed">BED</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#gff">GFF</a></td><td>Converts BED file format to GFF file format.</td></tr><tr><td><a href="/scriptmanager-docs/docs/coordinate-manipulation/gff-to-bed">gff-to-bed</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#gff">GFF</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bed">BED</a></td><td>Converts GFF file format to BED file format.</td></tr><tr><td><a href="/scriptmanager-docs/docs/coordinate-manipulation/sort-bed">sort-bed</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bed">BED</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bed">BED</a></td><td>Sort a CDT file and its corresponding BED file by the total score in the CDT file across the specified interval.</td></tr><tr><td><a href="/scriptmanager-docs/docs/coordinate-manipulation/sort-gff">sort-gff</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#gff">GFF</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#gff">GFF</a></td><td>Sort a CDT file and its corresponding GFF file by the total score in the CDT file across the specified interval.</td></tr></tbody></table><h2 class="anchor anchorWithStickyNavbar_LWe7" id="figure-generation">Figure Generation<a href="#figure-generation" class="hash-link" aria-label="Direct link to Figure Generation" title="Direct link to Figure Generation">​</a></h2><table><thead><tr><th>Tool Name</th><th>Input</th><th>Output</th><th>Description</th></tr></thead><tbody><tr><td><a href="/scriptmanager-docs/docs/figure-generation/heatmap">heatmap</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#cdt">CDT</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#png">PNG</a></td><td>Generate heatmap using CDT files.</td></tr><tr><td><a href="/scriptmanager-docs/docs/figure-generation/merge-heatmap">merge-heatmap</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#png">PNG</a> + <a href="/scriptmanager-docs/docs/References/file-formats#png">PNG</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#png">PNG</a></td><td>Merge Sense and Antisense png heatmaps.</td></tr><tr><td><a href="/scriptmanager-docs/docs/figure-generation/four-color">four-color</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#fasta">FASTA</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#png">PNG</a></td><td>Generate 4Color sequence plot given FASTA file and user-defined RGB colors.</td></tr><tr><td><a href="/scriptmanager-docs/docs/figure-generation/composite-plot">composite-plot</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#cdt">CDT</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#png">PNG</a></td><td>Generate a Composite Plot PNG from composite data like the output in TagPileup.</td></tr></tbody></table><h2 class="anchor anchorWithStickyNavbar_LWe7" id="file-utilities">File Utilities<a href="#file-utilities" class="hash-link" aria-label="Direct link to File Utilities" title="Direct link to File Utilities">​</a></h2><p><em>File Utilities tools description</em></p><table><thead><tr><th>Tool Name</th><th>Input</th><th>Output</th><th>Description</th></tr></thead><tbody><tr><td><a href="/scriptmanager-docs/docs/file-utilities/md5checksum">md5checksum</a></td><td>*</td><td><a href="/scriptmanager-docs/docs/References/file-formats#txt">TXT</a></td><td>Calculate MD5 checksum for an input file.</td></tr></tbody></table><h2 class="anchor anchorWithStickyNavbar_LWe7" id="peak-analysis">Peak Analysis<a href="#peak-analysis" class="hash-link" aria-label="Direct link to Peak Analysis" title="Direct link to Peak Analysis">​</a></h2><p><em>Peak Analysis tools description</em></p><table><thead><tr><th>Tool Name</th><th>Input</th><th>Output</th><th>Description</th></tr></thead><tbody><tr><td><a href="/scriptmanager-docs/docs/peak-analysis/peak-align-ref"><code>peak-align-ref</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bed">BED(ref)</a> + <a href="/scriptmanager-docs/docs/References/file-formats#bed">BED(peak)</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#cdt">CDT</a></td><td>Align BED peaks to Reference BED file creating CDT files for heatmap generation.</td></tr><tr><td><a href="/scriptmanager-docs/docs/peak-analysis/filter-bed"><code>filter-bed</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bed">BED</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bed">BED(cluster)</a> + <a href="/scriptmanager-docs/docs/References/file-formats#bed">BED(filter)</a></td><td>Filter BED file using user-specified exclusion zone using the score column to determine which peak to retain.</td></tr><tr><td><a href="/scriptmanager-docs/docs/peak-analysis/tile-genome"><code>tile-genome</code></a></td><td>genomeName</td><td>[<a href="/scriptmanager-docs/docs/References/file-formats#bed">BED</a>|<a href="/scriptmanager-docs/docs/References/file-formats#gff">GFF</a>]</td><td>Generate a coordinate file that tiles (non-overlapping) across an entire genome.</td></tr><tr><td><a href="/scriptmanager-docs/docs/peak-analysis/rand-coord"><code>rand-coord</code></a></td><td>genomeName</td><td>[<a href="/scriptmanager-docs/docs/References/file-formats#bed">BED</a>|<a href="/scriptmanager-docs/docs/References/file-formats#gff">GFF</a>]</td><td>Generate a coordinate file that tiles (non-overlapping) across an entire genome.</td></tr><tr><td><a href="/scriptmanager-docs/docs/peak-analysis/signal-dup"><code>signal-dup</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#gff">GFF</a> + <a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#txt">TXT</a></td><td>Calculate duplication statistics at user-specified regions.</td></tr></tbody></table><h2 class="anchor anchorWithStickyNavbar_LWe7" id="read-analysis">Read Analysis<a href="#read-analysis" class="hash-link" aria-label="Direct link to Read Analysis" title="Direct link to Read Analysis">​</a></h2><p>These Read Analysis tools are mostly tools that work with the <!-- -->[CDT or other tab-delimited Matrix file formats][file-format]<!-- -->.</p><p>Most commonly used  is the TagPileup tool.</p><p>Others manipulate files to calculate normalization factors, normalize the data, or other manipulations and transformations of read info.</p><table><thead><tr><th>Tool Name</th><th>Input</th><th>Output</th><th>Description</th></tr></thead><tbody><tr><td><a href="/scriptmanager-docs/docs/read-analysis/tag-pileup"><code>tag-pileup</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bed">BED</a> + <a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#cdt">CDT</a></td><td>Pileup 5&#x27; ends of aligned tags given BED and BAM files according to user-defined parameters.</td></tr><tr><td><a href="/scriptmanager-docs/docs/read-analysis/scaling-factor"><code>scaling-factor</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bam">BAM</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#txt">TXT</a></td><td>Calculate the factor as either total tag normalization or normalization of ChIP-seq data with control. (PMID:22883957)</td></tr><tr><td><a href="/scriptmanager-docs/docs/read-analysis/scale-matrix"><code>scale-matrix</code></a></td><td>[<a href="/scriptmanager-docs/docs/References/file-formats#cdt">CDT</a>|<a href="/scriptmanager-docs/docs/References/file-formats#matrix-format-custom-for-these-tools">TAB</a>]</td><td><a href="/scriptmanager-docs/docs/References/file-formats#matrix-format-custom-for-these-tools">TAB</a>[<!-- -->..<!-- -->]</td><td>Apply a user-specified scaling factor to tab-delimited matrix data.</td></tr><tr><td><a href="/scriptmanager-docs/docs/read-analysis/aggregate-data"><code>aggregate-data</code></a></td><td>[<a href="/scriptmanager-docs/docs/References/file-formats#cdt">CDT</a>|<a href="/scriptmanager-docs/docs/References/file-formats#matrix-format-custom-for-these-tools">TAB</a>]<!-- -->[<!-- -->..<!-- -->]</td><td><a href="/scriptmanager-docs/docs/References/file-formats#matrix-format-custom-for-these-tools">TAB</a>[<!-- -->..<!-- -->]</td><td>The AggregateData tool is used to process a bunch of matrix files into one matrix file.</td></tr></tbody></table><h2 class="anchor anchorWithStickyNavbar_LWe7" id="sequence-analysis">Sequence Analysis<a href="#sequence-analysis" class="hash-link" aria-label="Direct link to Sequence Analysis" title="Direct link to Sequence Analysis">​</a></h2><p><em>Sequence Analysis tools description</em></p><table><thead><tr><th>Tool Name</th><th>Input</th><th>Output</th><th>Description</th></tr></thead><tbody><tr><td><a href="/scriptmanager-docs/docs/sequence-analysis/fasta-extract"><code>fasta-extract</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bed">BED</a> + <a href="/scriptmanager-docs/docs/References/file-formats#fasta">FASTA(genome)</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#fasta">FASTA</a></td><td>Generate FASTA file from indexed Genome FASTA file and BED file. Script will generate FAI index if not present in Genome FASTA folder.</td></tr><tr><td><a href="/scriptmanager-docs/docs/sequence-analysis/randomize-fasta"><code>randomize-fasta</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#fasta">FASTA</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#fasta">FASTA</a></td><td>Randomizes FASTA sequence for each input entry</td></tr><tr><td><a href="/scriptmanager-docs/docs/sequence-analysis/search-motif"><code>search-motif</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#fasta">FASTA</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bed">BED</a></td><td>Search for an IUPAC DNA sequence motif in FASTA files with mismatches allowed.</td></tr><tr><td><a href="/scriptmanager-docs/docs/sequence-analysis/dna-shape-bed"><code>dna-shape-bed</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#bed">BED</a> + <a href="/scriptmanager-docs/docs/References/file-formats#fasta">FASTA(genome)</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#cdt">CDT</a></td><td>Calculate intrinsic DNA shape parameters given BED file and Genome FASTA file. Based on Roh&#x27;s lab <a href="https://pubmed.ncbi.nlm.nih.gov/23703209" target="_blank" rel="noopener noreferrer">DNAshape server data</a>.</td></tr><tr><td><a href="/scriptmanager-docs/docs/sequence-analysis/dna-shape-fasta"><code>dna-shape-fasta</code></a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#fasta">FASTA</a></td><td><a href="/scriptmanager-docs/docs/References/file-formats#cdt">CDT</a></td><td>Calculate intrinsic DNA shape parameters given input FASTA files. Based on Roh&#x27;s lab <a href="https://pubmed.ncbi.nlm.nih.gov/23703209" target="_blank" rel="noopener noreferrer">DNAshape server data</a>.</td></tr></tbody></table></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"></div><div class="col lastUpdated_vwxv"><span class="theme-last-updated">Last updated<!-- --> on <b><time datetime="2023-01-27T16:48:18.000Z">Jan 27, 2023</time></b> by <b>Olivia Wen-Mei Lang</b></span></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--prev" href="/scriptmanager-docs/docs/Guides/command-line"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Command Line</div></a><a class="pagination-nav__link pagination-nav__link--next" href="/scriptmanager-docs/docs/Guides/color-guide"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Color Selection Guide</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_bqdL thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#bam-format-converter" class="table-of-contents__link toc-highlight">BAM Format Converter</a></li><li><a href="#bam-manipulation" class="table-of-contents__link toc-highlight">BAM Manipulation</a></li><li><a href="#bam-statistics" class="table-of-contents__link toc-highlight">BAM Statistics</a></li><li><a href="#coordinate-manipulation" class="table-of-contents__link toc-highlight">Coordinate Manipulation</a></li><li><a href="#figure-generation" class="table-of-contents__link toc-highlight">Figure Generation</a></li><li><a href="#file-utilities" class="table-of-contents__link toc-highlight">File Utilities</a></li><li><a href="#peak-analysis" class="table-of-contents__link toc-highlight">Peak Analysis</a></li><li><a href="#read-analysis" class="table-of-contents__link toc-highlight">Read Analysis</a></li><li><a href="#sequence-analysis" class="table-of-contents__link toc-highlight">Sequence Analysis</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/scriptmanager-docs/docs/References/tool-index">Tool Index (A-Z)</a></li><li class="footer__item"><a class="footer__link-item" href="/scriptmanager-docs/docs/References/file-formats">File Formats</a></li></ul></div><div class="col footer__col"><div class="footer__title">Community</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://pughlab.mbg.cornell.edu" target="_blank" rel="noopener noreferrer" class="footer__link-item">Pugh Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://williamkmlai.github.io" target="_blank" rel="noopener noreferrer" class="footer__link-item">Lai Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div><div class="col footer__col"><div class="footer__title">Other Tools We Develop</div><ul class="footer__items clean-list"><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/scriptmanager-docs" target="_blank" rel="noopener noreferrer" class="footer__link-item">ScriptManager<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://github.com/seqcode/pegr" target="_blank" rel="noopener noreferrer" class="footer__link-item">PEGR<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/stencil" target="_blank" rel="noopener noreferrer" class="footer__link-item">STENCIL<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/GenoPipe-docs" target="_blank" rel="noopener noreferrer" class="footer__link-item">GenoPipe<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="margin-bottom--sm"><img src="/scriptmanager-docs/img/pughlab_logo.svg" alt="Pugh Lab Logo" class="themedImage_ToTc themedImage--light_HNdA footer__logo"><img src="/scriptmanager-docs/img/pughlab_logo.svg" alt="Pugh Lab Logo" class="themedImage_ToTc themedImage--dark_i4oU footer__logo"></div><div class="footer__copyright">Copyright © 2023 ScriptManager. Built with Docusaurus.</div></div></div></footer></div>
<script src="/scriptmanager-docs/assets/js/runtime~main.015d9caf.js"></script>
<script src="/scriptmanager-docs/assets/js/main.510075bb.js"></script>
</body>
</html>