<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-Tutorials/chipseq-se-tutorial">
<head>
<meta charset="UTF-8">
<meta name="generator" content="Docusaurus v2.3.1">
<title data-rh="true">ChIP-seq Tutorial (Single-end data) | ScriptManager</title><meta data-rh="true" name="viewport" content="width=device-width,initial-scale=1"><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://CEGRcode.github.io/scriptmanager-docs/docs/Tutorials/chipseq-se-tutorial"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="ChIP-seq Tutorial (Single-end data) | ScriptManager"><meta data-rh="true" name="description" content="Generating two basic ChIP-seq plots from single-ended data: composite and heatmap"><meta data-rh="true" property="og:description" content="Generating two basic ChIP-seq plots from single-ended data: composite and heatmap"><link data-rh="true" rel="icon" href="/scriptmanager-docs/img/SM_favicon.png"><link data-rh="true" rel="canonical" href="https://CEGRcode.github.io/scriptmanager-docs/docs/Tutorials/chipseq-se-tutorial"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/scriptmanager-docs/docs/Tutorials/chipseq-se-tutorial" hreflang="en"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/scriptmanager-docs/docs/Tutorials/chipseq-se-tutorial" hreflang="x-default"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.3.0/css/all.min.css" integrity="sha512-SzlrxWUlpfuzQ+pcUCosxcglQRNAq/DZjVsC0lE40xsADsfeQoEypE+enwcOiGjk/bSuGGKHEyjSoQ1zVisanQ==" crossorigin="anonymous"><link rel="stylesheet" href="/scriptmanager-docs/assets/css/styles.93e36019.css">
<link rel="preload" href="/scriptmanager-docs/assets/js/runtime~main.015d9caf.js" as="script">
<link rel="preload" href="/scriptmanager-docs/assets/js/main.510075bb.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region" aria-label="Skip to main content"><a class="skipToContent_fXgn" href="#docusaurus_skipToContent_fallback">Skip to main content</a></div><nav aria-label="Main" class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Toggle navigation bar" aria-expanded="false" class="navbar__toggle clean-btn" type="button"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/scriptmanager-docs/"><div class="navbar__logo"><img src="/scriptmanager-docs/img/SM_favicon.png" alt="ScriptManager Logo" class="themedImage_ToTc themedImage--light_HNdA"><img src="/scriptmanager-docs/img/SM_favicon.png" alt="ScriptManager Logo" class="themedImage_ToTc themedImage--dark_i4oU"></div><b class="navbar__title text--truncate">ScriptManager</b></a><a class="navbar__item navbar__link" href="/scriptmanager-docs/docs/">General</a><a class="navbar__item navbar__link" href="/scriptmanager-docs/docs/read-analysis/tag-pileup">Tools</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/CEGRcode/scriptmanager" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link"></a><div class="toggle_vylO colorModeToggle_DEke"><button class="clean-btn toggleButton_gllP toggleButtonDisabled_aARS" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)" aria-live="polite"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_pyhR"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_wfgR"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_ZlJk"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div id="docusaurus_skipToContent_fallback" class="main-wrapper mainWrapper_z2l0 docsWrapper_BCFX"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_sjWU" type="button"></button><div class="docPage__5DB"><main class="docMainContainer_gTbr docMainContainerEnhanced_Uz_u"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_VOVn"><div class="docItemContainer_Djhp"><article><div class="tocCollapsible_ETCw theme-doc-toc-mobile tocMobile_ITEo"><button type="button" class="clean-btn tocCollapsibleButton_TO0P">On this page</button></div><div class="theme-doc-markdown markdown"><header><h1>ChIP-seq Tutorial (Single-end data)</h1></header><p><em>Generating two basic ChIP-seq plots from single-ended data: composite and heatmap</em></p><p><strong>Goal:</strong> This tutorial provides a guide to generating 2 basic plots (Composite plot and Heatmap) using the ScriptManager platform and <u>single-end</u> data generated by the <a href="http://www.encodeproject.org/" target="_blank" rel="noopener noreferrer">ENCODE project</a>. This specific example will look at the binding of p300 within open chromatin regions assayed by ATAC-seq.</p><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/ChIP-exo_heatmap-b74eb063d0ee9acbc7574fb04f85839b.png" style="width:20%" class="img_ev3q"><div><center><img loading="lazy" src="https://www.encodeproject.org/static/img/encode-logo-small-2x.png" align="center" class="img_ev3q"></center><br><center><img loading="lazy" src="/scriptmanager-docs/assets/images/ChIP-exo_composite-1733680a72d93be79fcf1793799c21c8.png" style="width:80%" class="img_ev3q"></center></div></div><h2 class="anchor anchorWithStickyNavbar_LWe7" id="download-scriptmanager-v014">Download ScriptManager (v0.14):<a href="#download-scriptmanager-v014" class="hash-link" aria-label="Direct link to Download ScriptManager (v0.14):" title="Direct link to Download ScriptManager (v0.14):">​</a></h2><p>The current version of ScriptManager is available for download <strong><a href="https://github.com/CEGRcode/scriptmanager/releases/download/v0.14/ScriptManager-v0.14.jar" target="_blank" rel="noopener noreferrer">here</a></strong>. Make sure you have <a href="/scriptmanager-docs/docs/#java">Java installed</a>.</p><p>The file <code>ScriptManager-v0.14.jar</code> should be placed someplace locally accessible. For example on Mac OS on the Desktop (Permissions will need to be accepted) or someplace in your home directory (<code>Macintosh_HD:Users/userID/ScriptManager</code>)</p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="download-data">Download Data<a href="#download-data" class="hash-link" aria-label="Direct link to Download Data" title="Direct link to Download Data">​</a></h2><p>You need one set of genomic coordinate regions to investigate (BED) and one file of sequencing data alignments (BAM) to complete this exercise. <a href="/scriptmanager-docs/docs/References/file-formats">Read more about the BED/BAM file formats here.</a></p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="bed-file">BED File<a href="#bed-file" class="hash-link" aria-label="Direct link to BED File" title="Direct link to BED File">​</a></h3><p>These are open chromatin regions in HepG2 as assayed by ATAC-seq. <a href="https://www.encodeproject.org/experiments/ENCSR042AWH/" target="_blank" rel="noopener noreferrer">See ENCODE</a> for more details.</p><a href="https://www.encodeproject.org/files/ENCFF913MQB/@@download/ENCFF913MQB.bed.gz" target="_blank" rel="noopener noreferrer" class="button button--secondary">Download `ENCFF913MQB.bed.gz` BED file</a><br><br><div class="theme-admonition theme-admonition-note alert alert--secondary admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 14 16"><path fill-rule="evenodd" d="M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"></path></svg></span>note</div><div class="admonitionContent_S0QG"><p>It is okay if your file is gzipped (<code>.gz</code>). ScriptManager&#x27;s tools support gzipped tab-delimited formats.</p></div></div><h3 class="anchor anchorWithStickyNavbar_LWe7" id="bam-file">BAM File<a href="#bam-file" class="hash-link" aria-label="Direct link to BAM File" title="Direct link to BAM File">​</a></h3><p>This is the ChIP-seq alignment information for the EP300 factor in HepG2 cells. <a href="https://www.encodeproject.org/experiments/ENCSR271XMW/" target="_blank" rel="noopener noreferrer">See ENCODE</a> for more details.</p><a href="https://www.encodeproject.org/files/ENCFF779LBX/@@download/ENCFF779LBX.bam" target="_blank" rel="noopener noreferrer" class="button button--secondary">Download `ENCFF779LBX.bam` BAM file</a><h2 class="anchor anchorWithStickyNavbar_LWe7" id="generate-the-plots">Generate the Plots<a href="#generate-the-plots" class="hash-link" aria-label="Direct link to Generate the Plots" title="Direct link to Generate the Plots">​</a></h2><h3 class="anchor anchorWithStickyNavbar_LWe7" id="1-open-scriptmanager">1. Open ScriptManager<a href="#1-open-scriptmanager" class="hash-link" aria-label="Direct link to 1. Open ScriptManager" title="Direct link to 1. Open ScriptManager">​</a></h3><div class="tabs-container tabList__CuJ"><ul role="tablist" aria-orientation="horizontal" class="tabs"><li role="tab" tabindex="0" aria-selected="true" class="tabs__item tabItem_LNqP tabs__item--active">MacOS</li><li role="tab" tabindex="-1" aria-selected="false" class="tabs__item tabItem_LNqP">Linux</li><li role="tab" tabindex="-1" aria-selected="false" class="tabs__item tabItem_LNqP">Windows</li></ul><div class="margin-top--md"><div role="tabpanel" class="tabItem_Ymn6"><p>Depending on your system permissions, you may need to be an administrator to open this for the first time. On Mac systems, this can be done by right-clicking the file and selecting ‘Open’ at the top.</p><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/MacOpenSM-5f0508d59e500792ea199946818706f1.png" style="width:40%;border:solid 1px" class="img_ev3q"></div><div class="theme-admonition theme-admonition-caution alert alert--warning admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"></path></svg></span>caution</div><div class="admonitionContent_S0QG"><p>Some MacOS systems may not properly open the JAR by simply double-clicking on the JAR file so you may need to open your Terminal window and execute it from the command line by executing the jar file <em>without any arguments or flags</em>:</p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">java -jar /path/to/ScriptManager.jar</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>If you&#x27;re not sure how to type the path to ScriptManager, you can type <code>java -jar </code> (end with space) and then drag ScriptManager from Finder into your Terminal window and then press enter.</p></div></div></div><div role="tabpanel" class="tabItem_Ymn6" hidden=""><p>Double-click or right-click the ScriptManager JAR file to start the program.</p><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/LinuxOpenSM-1fe13f9ab9c369dd24c0a67c7ee86f41.png" style="width:40%;border:solid 1px" class="img_ev3q"></div></div><div role="tabpanel" class="tabItem_Ymn6" hidden=""><p>Double-click or right-click the ScriptManager JAR file to start the program.</p><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/WindowsOpenSM-ac32d3855e9f0cf632ab34c5c9fc983d.png" style="width:40%;border:solid 1px" class="img_ev3q"></div></div></div></div><p>Once you see the main tool selection window, you&#x27;re off to the races!</p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="2-generate-bai-index-file">2. Generate BAI index file<a href="#2-generate-bai-index-file" class="hash-link" aria-label="Direct link to 2. Generate BAI index file" title="Direct link to 2. Generate BAI index file">​</a></h3><p>A BAI index file is required for each BAM file of interest (i.e., the tag occupancy data you want to plot). This file allows for rapid access of the sorted and aligned sequence reads (BAM file).</p><div class="theme-admonition theme-admonition-tip alert alert--success admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>tip</div><div class="admonitionContent_S0QG"><p>SAM/BAM standard is to keep BAI file in same directory as BAM file with the ScriptManager-generated filename.</p><div class="language-bash codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-bash codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">ENCFF779LBX.bam</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">ENCFF779LBX.bam.bai  </span><span class="token comment" style="color:#999988;font-style:italic"># Need to generate this file to proceed.</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div></div></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="21-navigate-to-bam-manipulation-️-bam-bai-indexer">2.1. Navigate to BAM Manipulation ➡️ <a href="/scriptmanager-docs/docs/bam-manipulation/bam-indexer"><strong>BAM-BAI Indexer</strong></a><a href="#21-navigate-to-bam-manipulation-️-bam-bai-indexer" class="hash-link" aria-label="Direct link to 21-navigate-to-bam-manipulation-️-bam-bai-indexer" title="Direct link to 21-navigate-to-bam-manipulation-️-bam-bai-indexer">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/maingui-bamidx-916174344aa9670cd3fe1bb3f0287a74.png" style="width:60%" class="img_ev3q"></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="22-generate-bai-index-files-for-each-bam-file-of-interest-by-loading-your-bam-file-and-clicking-index">2.2. Generate BAI index files for each BAM file of interest by loading your BAM file and clicking &quot;Index.&quot;<a href="#22-generate-bai-index-files-for-each-bam-file-of-interest-by-loading-your-bam-file-and-clicking-index" class="hash-link" aria-label="Direct link to 2.2. Generate BAI index files for each BAM file of interest by loading your BAM file and clicking &quot;Index.&quot;" title="Direct link to 2.2. Generate BAI index files for each BAM file of interest by loading your BAM file and clicking &quot;Index.&quot;">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-bamidx-aa4e45d9e6ea63a6f3bb4a921bfd1ad5.png" style="width:60%" class="img_ev3q"><i class="fa-solid fa-arrow-right fa-lg"></i><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-bamidx-complete-e925a25e686f6c6813848dd104f8cfd5.png" style="width:30%" class="img_ev3q"></div><div class="theme-admonition theme-admonition-note alert alert--secondary admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 14 16"><path fill-rule="evenodd" d="M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"></path></svg></span>note</div><div class="admonitionContent_S0QG"><p>The speed of this step scales with the size of the BAM file. Generally this step 30 sec for a 100 MB BAM file but may take 1-2 min for a multi-GB BAM file.</p></div></div><h3 class="anchor anchorWithStickyNavbar_LWe7" id="3-resize-the-reb1-motif-aligned-bed-file">3. Resize the Reb1 motif-aligned BED file<a href="#3-resize-the-reb1-motif-aligned-bed-file" class="hash-link" aria-label="Direct link to 3. Resize the Reb1 motif-aligned BED file" title="Direct link to 3. Resize the Reb1 motif-aligned BED file">​</a></h3><p>The BED file is the set of reference coordinates that your heatmap and composite plots will be aligned to, but you’ll need to specify how far upstream and downstream you want your data to be plotted; i.e., “Size of Expansion (bp). If you bed file is defined by more than a 1 bp interval AND you want to add to limits of that interval, then select  “Add to Border”).</p><h4 class="anchor anchorWithStickyNavbar_LWe7" id="31-navigate-to-coordinate-file-manipulation-️-expand-bed-file">3.1. Navigate to Coordinate File Manipulation ➡️ <a href="/scriptmanager-docs/docs/coordinate-manipulation/expand-bed"><strong>Expand BED File</strong></a><a href="#31-navigate-to-coordinate-file-manipulation-️-expand-bed-file" class="hash-link" aria-label="Direct link to 31-navigate-to-coordinate-file-manipulation-️-expand-bed-file" title="Direct link to 31-navigate-to-coordinate-file-manipulation-️-expand-bed-file">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/maingui-expandbed-c34fe4ec13cf11c4c61fa108f2a8d5a2.png" style="width:60%" class="img_ev3q"></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="32-for-this-tutorial-use-the-250bp-expansion-and-select-expand-from-center">3.2. For this tutorial, use the 250bp expansion and select &quot;Expand from Center&quot;.<a href="#32-for-this-tutorial-use-the-250bp-expansion-and-select-expand-from-center" class="hash-link" aria-label="Direct link to 3.2. For this tutorial, use the 250bp expansion and select &quot;Expand from Center&quot;." title="Direct link to 3.2. For this tutorial, use the 250bp expansion and select &quot;Expand from Center&quot;.">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-expandbed-0faadcd1af45e62bd9a96280c0094d50.png" style="width:60%" class="img_ev3q"><i class="fa-solid fa-arrow-right fa-lg"></i><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-expandbed-complete-031750880fd32f9fc0d7002c35bd022d.png" style="width:30%" class="img_ev3q"></div><div class="theme-admonition theme-admonition-tip alert alert--success admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>tip</div><div class="admonitionContent_S0QG"><p>BED file coordinates often need to be resized for more informative tag pileups. For Reb1 (yeast), 250-500 bp windows are generally sufficient. Mammalian samples may require larger windows (500-2000 bp) based on the amount of indirect-crosslinking</p></div></div><h3 class="anchor anchorWithStickyNavbar_LWe7" id="4-generate-the-tag-pileup">4. Generate the tag pileup<a href="#4-generate-the-tag-pileup" class="hash-link" aria-label="Direct link to 4. Generate the tag pileup" title="Direct link to 4. Generate the tag pileup">​</a></h3><p>Use TagPileup to pileup the BAM data within a set of BED coordinate windows to generate the composite plot and the matrix(CDT) files that will be used to generate the heatmaps.</p><h4 class="anchor anchorWithStickyNavbar_LWe7" id="41-navigate-to-sequence-read-analysis-️-tag-pileup">4.1. Navigate to Sequence Read Analysis ➡️ <a href="/scriptmanager-docs/docs/read-analysis/tag-pileup"><strong>Tag Pileup</strong></a><a href="#41-navigate-to-sequence-read-analysis-️-tag-pileup" class="hash-link" aria-label="Direct link to 41-navigate-to-sequence-read-analysis-️-tag-pileup" title="Direct link to 41-navigate-to-sequence-read-analysis-️-tag-pileup">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/maingui-tagpileup-a595f39070628d9d700325cc9d77895d.png" style="width:60%" class="img_ev3q"></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="42-load-the-bed-and-bam-files">4.2. Load the BED and BAM files<a href="#42-load-the-bed-and-bam-files" class="hash-link" aria-label="Direct link to 4.2. Load the BED and BAM files" title="Direct link to 4.2. Load the BED and BAM files">​</a></h4><h4 class="anchor anchorWithStickyNavbar_LWe7" id="43-select-output-directory--make-sure-output-matrix-cdt-and-output-gzip-are-selected">4.3. Select output directory &amp; make sure &quot;Output Matrix&quot;, &quot;CDT&quot;, and &quot;Output GZIP&quot; are selected<a href="#43-select-output-directory--make-sure-output-matrix-cdt-and-output-gzip-are-selected" class="hash-link" aria-label="Direct link to 4.3. Select output directory &amp; make sure &quot;Output Matrix&quot;, &quot;CDT&quot;, and &quot;Output GZIP&quot; are selected" title="Direct link to 4.3. Select output directory &amp; make sure &quot;Output Matrix&quot;, &quot;CDT&quot;, and &quot;Output GZIP&quot; are selected">​</a></h4><ul><li><p>Bioinformatic projects should be organized in a uniform and consistent manner as described below</p></li><li><p><a href="https://journals.plos.org/ploscompbiol/article/file?id=10.1371/journal.pcbi.1000424&amp;type=printable" target="_blank" rel="noopener noreferrer">Paper on how to organize bioinformatics projects (Noble 2009)</a></p></li></ul><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-tagpileup-61c2817ab78015cd8bfe5b1e1377955c.png" style="width:100%" class="img_ev3q"></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="44-when-ready-select-pile-tags-to-execute">4.4. When ready, select ‘Pile Tags’ to execute<a href="#44-when-ready-select-pile-tags-to-execute" class="hash-link" aria-label="Direct link to 4.4. When ready, select ‘Pile Tags’ to execute" title="Direct link to 4.4. When ready, select ‘Pile Tags’ to execute">​</a></h4><div class="theme-admonition theme-admonition-tip alert alert--success admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>tip</div><div class="admonitionContent_S0QG"><p>The default parameters Tag Pileup is set to expect is a sequence-specific strand separated ChIP-exo dataset. Modifications to these parameters are needed for more specific analysis or when using data generated from other assays.</p></div></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="45-save-composite-results">4.5. Save composite results<a href="#45-save-composite-results" class="hash-link" aria-label="Direct link to 4.5. Save composite results" title="Direct link to 4.5. Save composite results">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-tagpileup-out-799264f9c9d6119ab8036c02dfbaa9d5.png" style="width:100%" class="img_ev3q"></div><ul><li><p>The displayed composite plot can be modified by right-clicking and selecting properties. Things such as axis labels, axis range, and colors can be modified here.</p></li><li><p>The final image can then be saved by right-clicking and selecting ‘Save as’. PNG is fine for most cases, but SVG is strongly recommended if this composite plot will be used in Adobe Illustrator later.</p></li></ul><div class="theme-admonition theme-admonition-note alert alert--secondary admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 14 16"><path fill-rule="evenodd" d="M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"></path></svg></span>note</div><div class="admonitionContent_S0QG"><p>Besides the composite plot image, ScriptManager has saved the matrix <!-- -->*<!-- -->.CDT files to your Output Directory together with the composite plot values file (If you didn&#x27;t change the name it would be called <code>composite_average.out</code>). These CDT files will be used as the input for generating heatmaps in the next step.</p></div></div><h3 class="anchor anchorWithStickyNavbar_LWe7" id="5-generate-heatmaps">5. Generate Heatmaps<a href="#5-generate-heatmaps" class="hash-link" aria-label="Direct link to 5. Generate Heatmaps" title="Direct link to 5. Generate Heatmaps">​</a></h3><h4 class="anchor anchorWithStickyNavbar_LWe7" id="51-navigate-to-figure-generation-️-heat-map">5.1. Navigate to Figure Generation ➡️ <a href="/scriptmanager-docs/docs/figure-generation/heatmap"><strong>Heat Map</strong></a>.<a href="#51-navigate-to-figure-generation-️-heat-map" class="hash-link" aria-label="Direct link to 51-navigate-to-figure-generation-️-heat-map" title="Direct link to 51-navigate-to-figure-generation-️-heat-map">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/maingui-heatmap-9bd8cf5b3e862579d99b4ffb1b913ca5.png" style="width:60%" class="img_ev3q"></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="52-heatmap-generator-can-only-generate-one-color-at-a-time-so-sense-and-anti-files-should-be-processed-separately-the-chip-exo-standard-for-strand-colors-is-sense--blue-and-anti--red">5.2. Heatmap Generator can only generate one color at a time, so ‘Sense’ and ‘Anti’ files should be processed separately. The ChIP-exo standard for strand colors is ‘Sense’ == blue and ‘Anti’ == red.<a href="#52-heatmap-generator-can-only-generate-one-color-at-a-time-so-sense-and-anti-files-should-be-processed-separately-the-chip-exo-standard-for-strand-colors-is-sense--blue-and-anti--red" class="hash-link" aria-label="Direct link to 5.2. Heatmap Generator can only generate one color at a time, so ‘Sense’ and ‘Anti’ files should be processed separately. The ChIP-exo standard for strand colors is ‘Sense’ == blue and ‘Anti’ == red." title="Direct link to 5.2. Heatmap Generator can only generate one color at a time, so ‘Sense’ and ‘Anti’ files should be processed separately. The ChIP-exo standard for strand colors is ‘Sense’ == blue and ‘Anti’ == red.">​</a></h4><p>Start by generating the &#x27;Sense&#x27; heatmap first.</p><h4 class="anchor anchorWithStickyNavbar_LWe7" id="53-click-load-files-and-select-the-_sensecdt-output-cdt-files-from-running-the-tagpileup-step">5.3. Click &quot;Load Files&quot; and select the <code>_sense.cdt</code> output CDT files from running the TagPileup step.<a href="#53-click-load-files-and-select-the-_sensecdt-output-cdt-files-from-running-the-tagpileup-step" class="hash-link" aria-label="Direct link to 53-click-load-files-and-select-the-_sensecdt-output-cdt-files-from-running-the-tagpileup-step" title="Direct link to 53-click-load-files-and-select-the-_sensecdt-output-cdt-files-from-running-the-tagpileup-step">​</a></h4><h4 class="anchor anchorWithStickyNavbar_LWe7" id="54-select-blue-for-the-color-per-the-lab-standard">5.4. Select &quot;blue&quot; for the color (per the lab standard)<a href="#54-select-blue-for-the-color-per-the-lab-standard" class="hash-link" aria-label="Direct link to 5.4. Select &quot;blue&quot; for the color (per the lab standard)" title="Direct link to 5.4. Select &quot;blue&quot; for the color (per the lab standard)">​</a></h4><h4 class="anchor anchorWithStickyNavbar_LWe7" id="55-the-heatmaps-for-this-dataset-show-the-best-contrast-when-using-the-default-percentile-threshold-value-95-or-95--otherwise-your-heatmaps-will-come-out-too-light-or-too-dark-to-see-the-shape">5.5. The heatmaps for this dataset show the best contrast when using the default &quot;Percentile Threshold&quot; value (.95 or 95%).  Otherwise your heatmaps will come out too light or too dark to see the shape.<a href="#55-the-heatmaps-for-this-dataset-show-the-best-contrast-when-using-the-default-percentile-threshold-value-95-or-95--otherwise-your-heatmaps-will-come-out-too-light-or-too-dark-to-see-the-shape" class="hash-link" aria-label="Direct link to 5.5. The heatmaps for this dataset show the best contrast when using the default &quot;Percentile Threshold&quot; value (.95 or 95%).  Otherwise your heatmaps will come out too light or too dark to see the shape." title="Direct link to 5.5. The heatmaps for this dataset show the best contrast when using the default &quot;Percentile Threshold&quot; value (.95 or 95%).  Otherwise your heatmaps will come out too light or too dark to see the shape.">​</a></h4><h4 class="anchor anchorWithStickyNavbar_LWe7" id="56-click-the-output-heatmap-checkbox-the-heatmap-generator-does-not-save-the-produced-png-by-default">5.6. Click the &quot;Output Heatmap&quot; checkbox. The Heatmap generator does not save the produced PNG by default.<a href="#56-click-the-output-heatmap-checkbox-the-heatmap-generator-does-not-save-the-produced-png-by-default" class="hash-link" aria-label="Direct link to 5.6. Click the &quot;Output Heatmap&quot; checkbox. The Heatmap generator does not save the produced PNG by default." title="Direct link to 5.6. Click the &quot;Output Heatmap&quot; checkbox. The Heatmap generator does not save the produced PNG by default.">​</a></h4><h4 class="anchor anchorWithStickyNavbar_LWe7" id="57-click-generate-to-save-your-sense-png-heatmap">5.7. Click &quot;Generate&quot; to save your Sense PNG heatmap!<a href="#57-click-generate-to-save-your-sense-png-heatmap" class="hash-link" aria-label="Direct link to 5.7. Click &quot;Generate&quot; to save your Sense PNG heatmap!" title="Direct link to 5.7. Click &quot;Generate&quot; to save your Sense PNG heatmap!">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-heatmap-sense-acaff40410d3935bec8e20cca749c86a.png" style="width:40%" class="img_ev3q"><i class="fa-solid fa-arrow-right fa-lg"></i><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-heatmap-sense-out-e317b5a92372f2fd64ffe5ffc796d76a.png" style="width:40%" class="img_ev3q"></div><div class="theme-admonition theme-admonition-tip alert alert--success admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>tip</div><div class="admonitionContent_S0QG"><p>&quot;Percentile Threshold&quot; is a useful for looking at the shape of ChIP-exo binding patterns while &quot;Absolute Threshold&quot; is useful for setting a shared contrast threshold across several samples and comparing signal density. For other data, you may need to play around with settings to find the right contrast for you data.</p></div></div><p>Similarly generate the &#x27;Anti&#x27; heatmap.</p><h4 class="anchor anchorWithStickyNavbar_LWe7" id="58-remove-the-sense-file-by-selecting-it-to-highlight-it-and-then-clicking-the-remove-files-button">5.8. Remove the &quot;Sense&quot; file by selecting it to highlight it and then clicking the &quot;Remove Files button&quot;<a href="#58-remove-the-sense-file-by-selecting-it-to-highlight-it-and-then-clicking-the-remove-files-button" class="hash-link" aria-label="Direct link to 5.8. Remove the &quot;Sense&quot; file by selecting it to highlight it and then clicking the &quot;Remove Files button&quot;" title="Direct link to 5.8. Remove the &quot;Sense&quot; file by selecting it to highlight it and then clicking the &quot;Remove Files button&quot;">​</a></h4><h4 class="anchor anchorWithStickyNavbar_LWe7" id="59-click-load-files-and-select-the-_anticdt-output-cdt-files-from-running-the-tagpileup-step">5.9. Click &quot;Load Files&quot; and select the <code>_anti.cdt</code> output CDT files from running the TagPileup step.<a href="#59-click-load-files-and-select-the-_anticdt-output-cdt-files-from-running-the-tagpileup-step" class="hash-link" aria-label="Direct link to 59-click-load-files-and-select-the-_anticdt-output-cdt-files-from-running-the-tagpileup-step" title="Direct link to 59-click-load-files-and-select-the-_anticdt-output-cdt-files-from-running-the-tagpileup-step">​</a></h4><h4 class="anchor anchorWithStickyNavbar_LWe7" id="510-select-red-for-the-color-per-the-lab-standard">5.10. Select &quot;red&quot; for the color (per the lab standard)<a href="#510-select-red-for-the-color-per-the-lab-standard" class="hash-link" aria-label="Direct link to 5.10. Select &quot;red&quot; for the color (per the lab standard)" title="Direct link to 5.10. Select &quot;red&quot; for the color (per the lab standard)">​</a></h4><h4 class="anchor anchorWithStickyNavbar_LWe7" id="511-since-all-the-other-parameters-should-be-the-same-from-you-sense-run-you-shouldnt-need-to-re-select-the-percentile-threshold-or-output-heatmap-options">5.11. Since all the other parameters should be the same from you &quot;Sense&quot; run, you shouldn&#x27;t need to re-select the &quot;Percentile Threshold&quot; or &quot;Output Heatmap&quot; options.<a href="#511-since-all-the-other-parameters-should-be-the-same-from-you-sense-run-you-shouldnt-need-to-re-select-the-percentile-threshold-or-output-heatmap-options" class="hash-link" aria-label="Direct link to 5.11. Since all the other parameters should be the same from you &quot;Sense&quot; run, you shouldn&#x27;t need to re-select the &quot;Percentile Threshold&quot; or &quot;Output Heatmap&quot; options." title="Direct link to 5.11. Since all the other parameters should be the same from you &quot;Sense&quot; run, you shouldn&#x27;t need to re-select the &quot;Percentile Threshold&quot; or &quot;Output Heatmap&quot; options.">​</a></h4><h4 class="anchor anchorWithStickyNavbar_LWe7" id="512-click-generate-to-save-you-anti-png-heatmap">5.12. Click &quot;Generate&quot; to save you Anti PNG heatmap!<a href="#512-click-generate-to-save-you-anti-png-heatmap" class="hash-link" aria-label="Direct link to 5.12. Click &quot;Generate&quot; to save you Anti PNG heatmap!" title="Direct link to 5.12. Click &quot;Generate&quot; to save you Anti PNG heatmap!">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-heatmap-anti-efc608a8912c1548cac99e6b1fea024e.png" style="width:40%" class="img_ev3q"><i class="fa-solid fa-arrow-right fa-lg"></i><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-heatmap-anti-out-c15ecf9808b3239fc0b72c6cd268aa24.png" style="width:40%" class="img_ev3q"></div><h3 class="anchor anchorWithStickyNavbar_LWe7" id="6-merge-strange-separated-heatmaps">6. Merge strange-separated heatmaps<a href="#6-merge-strange-separated-heatmaps" class="hash-link" aria-label="Direct link to 6. Merge strange-separated heatmaps" title="Direct link to 6. Merge strange-separated heatmaps">​</a></h3><h4 class="anchor anchorWithStickyNavbar_LWe7" id="61-navigate-to-figure-generation-️-merge-heatmaps-so-we-can-merge-our-strand-separated-heatmaps-into-a-single-png">6.1. Navigate to Figure Generation ➡️ <a href="/scriptmanager-docs/docs/figure-generation/merge-heatmap"><strong>Merge Heatmaps</strong></a> so we can merge our strand-separated heatmaps into a single PNG.<a href="#61-navigate-to-figure-generation-️-merge-heatmaps-so-we-can-merge-our-strand-separated-heatmaps-into-a-single-png" class="hash-link" aria-label="Direct link to 61-navigate-to-figure-generation-️-merge-heatmaps-so-we-can-merge-our-strand-separated-heatmaps-into-a-single-png" title="Direct link to 61-navigate-to-figure-generation-️-merge-heatmaps-so-we-can-merge-our-strand-separated-heatmaps-into-a-single-png">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/maingui-mergehm-a39596c9dc0067a0ff2596d266912ebb.png" style="width:60%" class="img_ev3q"></div><p>The script will automatically match sense to anti heatmaps using the standardized naming conventions used by ScriptManager</p><h4 class="anchor anchorWithStickyNavbar_LWe7" id="62-click-load-png-files-to-select-the-two-files-output-by-the-heatmap-tool-sense-and-anti-in-the-last-step">6.2. Click &quot;Load PNG Files&quot; to select the two files output by the HeatMap tool (sense and anti) in the last step<a href="#62-click-load-png-files-to-select-the-two-files-output-by-the-heatmap-tool-sense-and-anti-in-the-last-step" class="hash-link" aria-label="Direct link to 6.2. Click &quot;Load PNG Files&quot; to select the two files output by the HeatMap tool (sense and anti) in the last step" title="Direct link to 6.2. Click &quot;Load PNG Files&quot; to select the two files output by the HeatMap tool (sense and anti) in the last step">​</a></h4><h4 class="anchor anchorWithStickyNavbar_LWe7" id="63-click-generate-to-merge-the-png-files-into-the-same-heatmap">6.3. Click &quot;Generate&quot; to merge the PNG files into the same heatmap<a href="#63-click-generate-to-merge-the-png-files-into-the-same-heatmap" class="hash-link" aria-label="Direct link to 6.3. Click &quot;Generate&quot; to merge the PNG files into the same heatmap" title="Direct link to 6.3. Click &quot;Generate&quot; to merge the PNG files into the same heatmap">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-mergehm-f255e0ce11a26571158ee46f4bfb3eb3.png" style="width:50%" class="img_ev3q"><i class="fa-solid fa-arrow-right fa-lg"></i><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-mergehm-out-4145e0ec8d9f72b6db23a6b1727be0f8.png" style="width:40%" class="img_ev3q"></div><h3 class="anchor anchorWithStickyNavbar_LWe7" id="7-label-your-merged-heatmaps">7. Label your merged heatmaps<a href="#7-label-your-merged-heatmaps" class="hash-link" aria-label="Direct link to 7. Label your merged heatmaps" title="Direct link to 7. Label your merged heatmaps">​</a></h3><h4 class="anchor anchorWithStickyNavbar_LWe7" id="71-navigate-to-figure-generation-️-label-heatmaps-so-we-can-add-axes-to-the-heatmap">7.1. Navigate to Figure Generation ➡️ <a href="/scriptmanager-docs/docs/figure-generation/heatmap-labeler"><strong>Label Heatmaps</strong></a> so we can add axes to the heatmap<a href="#71-navigate-to-figure-generation-️-label-heatmaps-so-we-can-add-axes-to-the-heatmap" class="hash-link" aria-label="Direct link to 71-navigate-to-figure-generation-️-label-heatmaps-so-we-can-add-axes-to-the-heatmap" title="Direct link to 71-navigate-to-figure-generation-️-label-heatmaps-so-we-can-add-axes-to-the-heatmap">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/maingui-labelhm-a88a99ec59698462932fe22fef870bcd.png" style="width:60%" class="img_ev3q"></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="72-optionally-type-text-into-the-fields-to-annotate-your-axes-or-leave-them-empty-to-just-generate-an-svg-border">7.2. Optionally type text into the fields to annotate your axes (Or leave them empty to just generate an SVG border)<a href="#72-optionally-type-text-into-the-fields-to-annotate-your-axes-or-leave-them-empty-to-just-generate-an-svg-border" class="hash-link" aria-label="Direct link to 7.2. Optionally type text into the fields to annotate your axes (Or leave them empty to just generate an SVG border)" title="Direct link to 7.2. Optionally type text into the fields to annotate your axes (Or leave them empty to just generate an SVG border)">​</a></h4><div class="tutorial-img-flow-container"><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-labelhm-8003bd7b41dc4bda146d53121c762e69.png" style="width:40%" class="img_ev3q"><i class="fa-solid fa-arrow-right fa-lg"></i><img loading="lazy" src="/scriptmanager-docs/assets/images/gui-labelhm-dialog-e203db28c23bc3209f9f55c6f1200491.png" style="width:40%" class="img_ev3q"></div><h4 class="anchor anchorWithStickyNavbar_LWe7" id="73-click-generate-to-create-an-svg-file-that-labels-your-png">7.3. Click &quot;Generate&quot; to create an SVG file that labels your PNG<a href="#73-click-generate-to-create-an-svg-file-that-labels-your-png" class="hash-link" aria-label="Direct link to 7.3. Click &quot;Generate&quot; to create an SVG file that labels your PNG" title="Direct link to 7.3. Click &quot;Generate&quot; to create an SVG file that labels your PNG">​</a></h4><p>Open up your SVG file to see what it looks like! Use PowerPoint, Adobe Illustrator, or wherever you edit your publication figures to make any further edits!</p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="general-comments">General Comments<a href="#general-comments" class="hash-link" aria-label="Direct link to General Comments" title="Direct link to General Comments">​</a></h2><p>Bioinformatic projects should be organized in a uniform and consistent manner so that you can easily find them in the future. Consider the organizational structure described in <a href="https://journals.plos.org/ploscompbiol/article/file?id=10.1371/journal.pcbi.1000424&amp;type=printable" target="_blank" rel="noopener noreferrer"><strong>Noble, 2009</strong></a>.</p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="bonus-exercises">Bonus exercises!<a href="#bonus-exercises" class="hash-link" aria-label="Direct link to Bonus exercises!" title="Direct link to Bonus exercises!">​</a></h3><ul><li>Navigate the <!-- -->[YEP datasets][yep-data]<!-- --> to find BAM files of other factors and pile them up against the Reb1 binding site.<ul><li>Do any of these factors appear to bind the same places as Reb1?</li><li>Make sure you pile up the MasterNoTag control as the background sample to make sure that your patterns aren&#x27;t explained by background!</li></ul></li><li>Look for other sacCer3 reference BED files (TSS, TES, +1 Nucleosome)<ul><li>Do you find any factors in the YEP data that bind to any of these sites?</li><li>Do all sites get bound? Is there a type of site that is more bound than others?</li><li>Are there different binding patterns? Does your factor bind upstream or downstream or right on top of Reb1?</li></ul></li></ul><h2 class="anchor anchorWithStickyNavbar_LWe7" id="command-line-shell-script">Command-Line shell script<a href="#command-line-shell-script" class="hash-link" aria-label="Direct link to Command-Line shell script" title="Direct link to Command-Line shell script">​</a></h2><p>The following shell commands records the locations for a BED file, a BAM file, and the anticipated OUTPUT basename as environmental variables to derive the corresponding composite plot values and heatmaps. This can serve as a template for you to write out your own workflows as bash scripts that execute command-line style ScriptManager.</p><div class="language-bash codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-bash codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token assign-left variable" style="color:#36acaa">SCRIPTMANAGER</span><span class="token operator" style="color:#393A34">=</span><span class="token plain">/path/to/ScriptManager.jar</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token assign-left variable" style="color:#36acaa">BEDFILE</span><span class="token operator" style="color:#393A34">=</span><span class="token plain">/path/to/ENCFF913MQB.bed.gz</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token assign-left variable" style="color:#36acaa">BAMFILE</span><span class="token operator" style="color:#393A34">=</span><span class="token plain">/path/to/ENCFF779LBX.bam</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token assign-left variable" style="color:#36acaa">OUTPUT</span><span class="token operator" style="color:#393A34">=</span><span class="token plain">/path/to/myoutput</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">samtools index </span><span class="token variable" style="color:#36acaa">$BAMFILE</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">java -jar </span><span class="token variable" style="color:#36acaa">$SCRIPTMANAGER</span><span class="token plain"> coordinate-manipulation expand-bed -c </span><span class="token number" style="color:#36acaa">250</span><span class="token plain"> </span><span class="token variable" style="color:#36acaa">$BEDFILE</span><span class="token plain"> -o BED_250bp.bed</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">java -jar </span><span class="token variable" style="color:#36acaa">$SCRIPTMANAGER</span><span class="token plain"> read-analysis tag-pileup </span><span class="token variable" style="color:#36acaa">$BEDFILE</span><span class="token plain"> </span><span class="token variable" style="color:#36acaa">$BAMFILE</span><span class="token plain"> -o </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token punctuation" style="color:#393A34">\</span><span class="token plain">_composite.out -M </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token punctuation" style="color:#393A34">\</span><span class="token plain">_matrix</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">java -jar </span><span class="token variable" style="color:#36acaa">$SCRIPTMANAGER</span><span class="token plain"> figure-generation heatmap -p .95 --blue </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token punctuation" style="color:#393A34">\</span><span class="token plain">_matrix_sense.cdt -o SENSE.png</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">java -jar </span><span class="token variable" style="color:#36acaa">$SCRIPTMANAGER</span><span class="token plain"> figure-generation heatmap -p .95 --red </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token punctuation" style="color:#393A34">\</span><span class="token plain">_matrix_sense.cdt -o ANTI.png</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">java -jar </span><span class="token variable" style="color:#36acaa">$SCRIPTMANAGER</span><span class="token plain"> figure-generation merge-heatmap SENSE.png ANTI.png -o </span><span class="token variable" style="color:#36acaa">$OUTPUT</span><span class="token punctuation" style="color:#393A34">\</span><span class="token plain">_heatmap.png</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">rm</span><span class="token plain"> BED_250bp.bed SENSE.png ANTI.png</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Output files:</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic">#  - /path/to/myoutput_composite.out</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic">#  - /path/to/myoutput_matrix_sense.cdt</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic">#  - /path/to/myoutput_matrix_anti.cdt</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic">#  - /path/to/myoutput_heatmap.png</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div></div></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"></nav></div></div><div class="col col--3"><div class="tableOfContents_bqdL thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#download-scriptmanager-v014" class="table-of-contents__link toc-highlight">Download ScriptManager (v0.14):</a></li><li><a href="#download-data" class="table-of-contents__link toc-highlight">Download Data</a><ul><li><a href="#bed-file" class="table-of-contents__link toc-highlight">BED File</a></li><li><a href="#bam-file" class="table-of-contents__link toc-highlight">BAM File</a></li></ul></li><li><a href="#generate-the-plots" class="table-of-contents__link toc-highlight">Generate the Plots</a><ul><li><a href="#1-open-scriptmanager" class="table-of-contents__link toc-highlight">1. Open ScriptManager</a></li><li><a href="#2-generate-bai-index-file" class="table-of-contents__link toc-highlight">2. Generate BAI index file</a></li><li><a href="#3-resize-the-reb1-motif-aligned-bed-file" class="table-of-contents__link toc-highlight">3. Resize the Reb1 motif-aligned BED file</a></li><li><a href="#4-generate-the-tag-pileup" class="table-of-contents__link toc-highlight">4. Generate the tag pileup</a></li><li><a href="#5-generate-heatmaps" class="table-of-contents__link toc-highlight">5. Generate Heatmaps</a></li><li><a href="#6-merge-strange-separated-heatmaps" class="table-of-contents__link toc-highlight">6. Merge strange-separated heatmaps</a></li><li><a href="#7-label-your-merged-heatmaps" class="table-of-contents__link toc-highlight">7. Label your merged heatmaps</a></li></ul></li><li><a href="#general-comments" class="table-of-contents__link toc-highlight">General Comments</a><ul><li><a href="#bonus-exercises" class="table-of-contents__link toc-highlight">Bonus exercises!</a></li></ul></li><li><a href="#command-line-shell-script" class="table-of-contents__link toc-highlight">Command-Line shell script</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/scriptmanager-docs/docs/References/tool-index">Tool Index (A-Z)</a></li><li class="footer__item"><a class="footer__link-item" href="/scriptmanager-docs/docs/References/file-formats">File Formats</a></li></ul></div><div class="col footer__col"><div class="footer__title">Community</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://pughlab.mbg.cornell.edu" target="_blank" rel="noopener noreferrer" class="footer__link-item">Pugh Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://williamkmlai.github.io" target="_blank" rel="noopener noreferrer" class="footer__link-item">Lai Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div><div class="col footer__col"><div class="footer__title">Other Tools We Develop</div><ul class="footer__items clean-list"><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/scriptmanager-docs" target="_blank" rel="noopener noreferrer" class="footer__link-item">ScriptManager<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://github.com/seqcode/pegr" target="_blank" rel="noopener noreferrer" class="footer__link-item">PEGR<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/stencil" target="_blank" rel="noopener noreferrer" class="footer__link-item">STENCIL<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/GenoPipe-docs" target="_blank" rel="noopener noreferrer" class="footer__link-item">GenoPipe<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="margin-bottom--sm"><img src="/scriptmanager-docs/img/pughlab_logo.svg" alt="Pugh Lab Logo" class="themedImage_ToTc themedImage--light_HNdA footer__logo"><img src="/scriptmanager-docs/img/pughlab_logo.svg" alt="Pugh Lab Logo" class="themedImage_ToTc themedImage--dark_i4oU footer__logo"></div><div class="footer__copyright">Copyright © 2023 ScriptManager. Built with Docusaurus.</div></div></div></footer></div>
<script src="/scriptmanager-docs/assets/js/runtime~main.015d9caf.js"></script>
<script src="/scriptmanager-docs/assets/js/main.510075bb.js"></script>
</body>
</html>